
# first convert plink file to txt

AA_predxican_prepare.r
```R
# AA
args <- as.numeric(commandArgs(TRUE))
chr=args[1]


library(snpStats)


pathAA="/net/mulan/home/shanglu/GENOA/data/AA"
pathAAbed = "/net/mulan/home/shanglu/GENOA/data/AA/gene_bed"
pathresult_AA = "/net/mulan/home/shanglu/GENOA/analysis/conditional/output/AA"
path_AA_result = "/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping/result"
path_AA_output="/net/mulan/home/shanglu/GENOA/data/AA/output"
path_AAeqtl = "/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping"

load(paste0(pathAA, "/gene_AA_anno_order_correct.RData"))

gene_AA_anno_order_chr = gene_AA_anno_order[gene_AA_anno_order$chr==chr,]

chr.num = read.table(paste0(pathAA,"/Gene_file/chr",chr,".num"))$V1
print(paste0("chr ",chr," read"))

load(paste0(path_AAeqtl,"/AA_full_table_chr_",chr,".RData"))

for( j in 1:chr.num){


	i_index = chr
	j_index = j
  
  snps_index = which(AA_full_table$GENE %in% gene_AA_anno_order_chr$GENE[j])
  info = AA_full_table[snps_index,]
  
  save(info, file = paste0(pathAAbed,"/chr_",i_index,"/chr_",i_index,"_gene_",j_index,"_snpinfo.RData"))
  
  fam <- paste0(pathAAbed,"/chr_",i_index,"/chr_",i_index,"_gene_",j_index,".fam")
  bim <- paste0(pathAAbed,"/chr_",i_index,"/chr_",i_index,"_gene_",j_index,".bim")
  bed <- paste0(pathAAbed,"/chr_",i_index,"/chr_",i_index,"_gene_",j_index,".bed")
  
  if(file.exists(fam)){
  snpin_index = match( as.character(info$rs),as.character(read.table(bim)$V2))
 
  
  m = read.plink(bed, bim, fam)
  write.SnpMatrix(m$genotypes[,snpin_index], paste0(pathAAbed,"/chr_",i_index,"/chr_",i_index,"_gene_",j_index,".txt"),quote=F,col.names = F,row.names = F)
  
  a = read.table(paste0(pathAAbed,"/chr_",i_index,"/chr_",i_index,"_gene_",j_index,".txt"),header=F)
  a = 2-a
  for(k in 1:ncol(a)){
    a[is.na(a[,k]), k] <- mean(a[,k], na.rm = TRUE)
  }
  colnames(a) = as.character(info$rs)
  save(a, file = paste0(pathAAbed,"/chr_",i_index,"/chr_",i_index,"_gene_",j_index,".RData"))

	}
  }


```

EA_predxican_prepare.r
```
#EA
args <- as.numeric(commandArgs(TRUE))
chr=args[1]
print(i)

library(snpStats)


pathEA="/net/mulan/home/shanglu/GENOA/data/EA"
pathEAbed = "/net/mulan/home/shanglu/GENOA/data/EA/gene_bed"
pathresult_EA = "/net/mulan/home/shanglu/GENOA/analysis/conditional/output/EA"
path_EA_result = "/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping/result"
path_EA_output="/net/mulan/home/shanglu/GENOA/data/EA/output"
path_EAeqtl = "/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping"

load(paste0(pathEA, "/gene_EA_anno_order_correct.RData"))

gene_EA_anno_order_chr = gene_EA_anno_order[gene_EA_anno_order$chr==chr,]

chr.num = read.table(paste0(pathEA,"/Gene_file/chr",chr,".num"))$V1
print(paste0("chr ",chr," read"))

load(paste0(path_EAeqtl,"/EA_full_table_chr_",chr,".RData"))

for( j in 1:chr.num){


	i_index = chr
	j_index = j
  
  snps_index = which(EA_full_table$GENE %in% gene_EA_anno_order_chr$GENE[j])
  info = EA_full_table[snps_index,]
  
  save(info, file = paste0(pathEAbed,"/chr_",i_index,"/chr_",i_index,"_gene_",j_index,"_snpinfo.RData"))
  
  fam <- paste0(pathEAbed,"/chr_",i_index,"/chr_",i_index,"_gene_",j_index,".fam")
  bim <- paste0(pathEAbed,"/chr_",i_index,"/chr_",i_index,"_gene_",j_index,".bim")
  bed <- paste0(pathEAbed,"/chr_",i_index,"/chr_",i_index,"_gene_",j_index,".bed")
  
  if(file.exists(fam)){
  snpin_index = match( as.character(info$rs),as.character(read.table(bim)$V2))
 
  
  m = read.plink(bed, bim, fam)
  write.SnpMatrix(m$genotypes[,snpin_index], paste0(pathEAbed,"/chr_",i_index,"/chr_",i_index,"_gene_",j_index,".txt"),quote=F,col.names = F,row.names = F)
  
  a = read.table(paste0(pathEAbed,"/chr_",i_index,"/chr_",i_index,"_gene_",j_index,".txt"),header=F)
  a = 2-a
  for(k in 1:ncol(a)){
    a[is.na(a[,k]), k] <- mean(a[,k], na.rm = TRUE)
  }
  colnames(a) = as.character(info$rs)
  save(a, file = paste0(pathEAbed,"/chr_",i_index,"/chr_",i_index,"_gene_",j_index,".RData"))

	}
  }



  

```


```
#!/bin/bash
#SBATCH --job-name=AAtwas
#SBATCH --array=1-22
#SBATCH --cpus-per-task=1
#SBATCH --output=/net/mulan/home/shanglu/GENOA/data/AA/out/AAtwas%a.out
#SBATCH --error=/net/mulan/home/shanglu/GENOA/data/AA/err/AAtwas%a.err
#SBATCH --workdir=/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping/code
#SBATCH --partition=mulan

bash

let k=0

for i in `seq 1 22`
do
let k=${k}+1
if [ ${k} -eq ${SLURM_ARRAY_TASK_ID} ]; then
R CMD BATCH --slave "--args ${i} " ./AA_predxican_prepare.r  /net/mulan/home/shanglu/GENOA/data/AA/out/AA_predxican_prepare${i}.out
fi
done



```


```
#!/bin/bash
#SBATCH --job-name=EAtwas
#SBATCH --array=1-22
#SBATCH --cpus-per-task=1
#SBATCH --output=/net/mulan/home/shanglu/GENOA/data/EA/out/EAtwas%a.out
#SBATCH --error=/net/mulan/home/shanglu/GENOA/data/EA/err/EAtwas%a.err
#SBATCH --workdir=/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping/code
#SBATCH --partition=mulan

bash

let k=0

for i in `seq 1 22`
do
let k=${k}+1
if [ ${k} -eq ${SLURM_ARRAY_TASK_ID} ]; then
R CMD BATCH --slave "--args ${i} " ./EA_predxican_prepare.r  /net/mulan/home/shanglu/GENOA/data/EA/out/EA_predxican_prepare${i}.out
fi
done



```



# elnet net: codes from Predixcan

```R

path_AA="/net/mulan/home/shanglu/GENOA/data/AA"

suppressMessages(library(dplyr))
suppressMessages(library(glmnet))
suppressMessages((library(reshape2)))
suppressMessages(library(methods))

"%&%" <- function(a,b) paste(a,b, sep = "")


# generate_fold_ids
generate_fold_ids <- function(n_samples, n_folds=10) {
  n <- ceiling(n_samples / n_folds)
  fold_ids <- rep(1:n_folds, n)
  sample(fold_ids[1:n_samples])
}

# calc_R2
calc_R2 <- function(y, y_pred) {
  tss <- sum(y**2)
  rss <- sum((y - y_pred)**2)
  1 - rss/tss
}

# calc_corr
calc_corr <- function(y, y_pred) {
  sum(y*y_pred) / (sqrt(sum(y**2)) * sqrt(sum(y_pred**2)))
}

# nested_cv_elastic_net_perf
nested_cv_elastic_net_perf <- function(x, y, n_samples, n_train_test_folds, n_k_folds, alpha) {
  # Gets performance estimates for k-fold cross-validated elastic-net models.
  # Splits data into n_train_test_folds disjoint folds, roughly equal in size,
  # and for each fold, calculates a n_k_folds cross-validated elastic net model. Lambda parameter is
  # cross validated. Then get performance measures for how the model predicts on the hold-out
  # fold. Get the coefficient of determination, R^2, and a p-value, where the null hypothesis
  # is there is no correlation between prediction and observed.
  #
  # The mean and standard deviation of R^2 over all folds is then reported, and the p-values
  # are combined using Fisher's method.
  R2_folds <- rep(0, n_train_test_folds)
  corr_folds <- rep(0, n_train_test_folds)
  zscore_folds <- rep(0, n_train_test_folds)
  pval_folds <- rep(0, n_train_test_folds)
  # Outer-loop split into training and test set.
  train_test_fold_ids <- generate_fold_ids(n_samples, n_folds=n_train_test_folds)
  for (test_fold in 1:n_train_test_folds) {
    train_idxs <- which(train_test_fold_ids != test_fold)
    test_idxs <- which(train_test_fold_ids == test_fold)
    x_train <- x[train_idxs, ]
    y_train <- y[train_idxs]
    x_test <- x[test_idxs, ]
    y_test <- y[test_idxs]
    # Inner-loop - split up training set for cross-validation to choose lambda.
    cv_fold_ids <- generate_fold_ids(length(y_train), n_k_folds)
    y_pred <- tryCatch({
      # Fit model with training data.
      fit <- cv.glmnet(x_train, y_train, nfolds = n_k_folds, alpha = alpha, type.measure='mse', foldid = cv_fold_ids)
      # Predict test data using model that had minimal mean-squared error in cross validation.
      predict(fit, x_test, s = 'lambda.min')},
      # if the elastic-net model did not converge, predict the mean of the y_train (same as all non-intercept coef=0)
      error = function(cond) rep(mean(y_train), length(y_test)))
    R2_folds[test_fold] <- calc_R2(y_test, y_pred)
    # Get p-value for correlation test between predicted y and actual y.
    # If there was no model, y_pred will have var=0, so cor.test will yield NA.
    # In that case, give a random number from uniform distribution, which is what would
    # usually happen under the null.
    corr_folds[test_fold] <- ifelse(sd(y_pred) != 0, cor(y_pred, y_test), 0)
    zscore_folds[test_fold] <- atanh(corr_folds[test_fold])*sqrt(length(y_test) - 3) # Fisher transformation
    pval_folds[test_fold] <- ifelse(sd(y_pred) != 0, cor.test(y_pred, y_test)$p.value, runif(1))
  }
  R2_avg <- mean(R2_folds)
  R2_sd <- sd(R2_folds)
  rho_avg <- mean(corr_folds)
  rho_se <- sd(corr_folds)
  rho_avg_squared <- rho_avg**2
  # Stouffer's method for combining z scores.
  zscore_est <- sum(zscore_folds) / sqrt(n_train_test_folds)
  zscore_pval <- 2*pnorm(abs(zscore_est), lower.tail = FALSE)
  # Fisher's method for combining p-values: https://en.wikipedia.org/wiki/Fisher%27s_method
  pval_est <- pchisq(-2 * sum(log(pval_folds)), 2*n_train_test_folds, lower.tail = F)
  list(R2_avg=R2_avg, R2_sd=R2_sd, pval_est=pval_est, rho_avg=rho_avg, rho_se=rho_se, rho_zscore=zscore_est, rho_avg_squared=rho_avg_squared, zscore_pval=zscore_pval)
}


do_covariance <- function(gene_id, cis_gt) {
  geno_cov <- cov(cis_gt)
  geno_cov[lower.tri(geno_cov)] <- NA
  cov_df <- melt(geno_cov, varnames = c("rsid1", "rsid2"), na.rm = TRUE) %>%
              mutate(gene=gene_id) %>%
              select(GENE=gene, RSID1=rsid1, RSID2=rsid2, VALUE=value) %>%
              arrange(GENE, RSID1, RSID2)
  cov_df
}

do_covariance <- function(gene_id, cis_gt, rsids, varIDs) {
  model_gt <- cis_gt[,varIDs, drop=FALSE]
  colnames(model_gt) <- rsids
  geno_cov <- cov(model_gt)
  geno_cov[lower.tri(geno_cov)] <- NA
  cov_df <- melt(geno_cov, varnames = c("rsid1", "rsid2"), na.rm = TRUE) %>%
              mutate(gene=gene_id) %>%
              select(GENE=gene, RSID1=rsid1, RSID2=rsid2, VALUE=value) %>%
              arrange(GENE, RSID1, RSID2)
  cov_df
}



i_index=1
j_index=435

main <- function(i_index, j_index, n_folds=10, n_train_test_folds=5,seed=NA, alpha=0.5, null_testing=FALSE) {
		 
# get_gene_annotation
load(paste0(path_AA,"/gene_AA_anno_order_correct.RData"))
# get_filtered_snp_annot
load(paste0(path_AA,"/gene_bed/chr_",i_index,"/chr_",i_index,"_gene_",j_index,"_snpinfo.RData"))
# get_gene_expression
expr_df = unlist(read.table(paste0(path_AA,"/gene_bed/chr_",i_index,"/chr_",i_index,"_gene_",j_index,".fam"))$V7)
# get_gene_coords
gene_index = which(gene_AA_anno_order$GENE %in% info$GENE[1])
gene_coords = c(gene_AA_anno_order$low[gene_index],gene_AA_anno_order$up[gene_index])
# get_cis_genotype
load(paste0(path_AA,"/gene_bed/chr_",i_index,"/chr_",i_index,"_gene_",j_index,".RData"))
cis_gt = as.matrix(a)
n_samples = dim(cis_gt)[1]
n_genes = 1

gene_id = info$GENE[1]
gene_name = info$GENE[1]

snp_annot = info	
snp_annot$varID = as.character(snp_annot$rs)
snp_annot$rsid = as.character(snp_annot$rs)
snp_annot$refAllele = as.character(snp_annot$allele1)
snp_annot$effectAllele = as.character(snp_annot$allele0)

  # Set seed----
  seed <- ifelse(is.na(seed), sample(1:1000000, 1), seed)
  set.seed(seed)

  # Prepare output data----
  model_summary_file <- paste0('/net/mulan/home/shanglu/GENOA/analysis/prediction/predixcan/AA/chr_',i_index, "/AA" , '_chr_' ,i_index,'_gene_', j_index, '_model_summaries.txt')
  model_summary_cols <- c('gene_id', 'gene_name', 'alpha', 'n_snps_in_window', 'n_snps_in_model', 'lambda_min_mse',
                          'test_R2_avg', 'test_R2_sd', 'cv_R2_avg', 'cv_R2_sd', 'in_sample_R2',
                          'nested_cv_fisher_pval', 'rho_avg', 'rho_se', 'rho_zscore', 'rho_avg_squared', 'zscore_pval',
                          'cv_rho_avg', 'cv_rho_se', 'cv_rho_avg_squared', 'cv_zscore_est', 'cv_zscore_pval', 'cv_pval_est')
  write(model_summary_cols, file = model_summary_file, ncol = 24, sep = '\t')
  
  weights_file <- paste0('/net/mulan/home/shanglu/GENOA/analysis/prediction/predixcan/AA/chr_',i_index,'/AA_chr_',i_index,'_gene_',j_index,'.txt')
  
  weights_col <- c('gene_id', 'rsid', 'varID', 'ref', 'alt', 'beta')
  write(weights_col, file = weights_file, ncol = 6, sep = '\t')
  
 tiss_chr_summ_f <- paste0('/net/mulan/home/shanglu/GENOA/analysis/prediction/predixcan/AA/chr_',i_index,'/AA_chr_',i_index,'_gene_',j_index,'_summary.txt')
  tiss_chr_summ_col <- c('n_samples', 'chrom', 'cv_seed', 'n_genes')
  tiss_chr_summ <- data.frame(n_samples, i_index, seed, n_genes)
  colnames(tiss_chr_summ) <- tiss_chr_summ_col
  write.table(tiss_chr_summ, file = tiss_chr_summ_f, quote = FALSE, row.names = FALSE, sep = '\t')
  
  covariance_file <- paste0('/net/mulan/home/shanglu/GENOA/analysis/prediction/predixcan/AA/chr_',i_index,'/AA_chr_',i_index,'_gene_',j_index,'_covariance.txt')
  covariance_col <- c('GENE', 'RSID1', 'RSID2', 'VALUE')
  write(covariance_col, file = covariance_file, ncol = 4, sep = ' ')
  
 

  # Attempt to build model for each gene----
    gene_name <- info$GENE[1]
    gene = gene_name
    coords <- gene_coords
    if (dim(info)[1]==0) {
      # No snps within window for gene.
      model_summary <- c(gene, gene_name, alpha, 0, 0, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA)
      write(model_summary, file = model_summary_file, append = TRUE, ncol = 24, sep = '\t')
      next
    }
    model_summary <- c(gene, gene_name, alpha, ncol(cis_gt), 0, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA)
    if (ncol(cis_gt) >= 2) {
      expression_vec <- expr_df
      adj_expression <- expression_vec # covariates already adjusted in our model
      if (null_testing)
        adj_expression <- sample(adj_expression)
      perf_measures <- nested_cv_elastic_net_perf(cis_gt, adj_expression, n_samples, n_train_test_folds, n_folds, alpha)
      R2_avg <- perf_measures$R2_avg
      R2_sd <- perf_measures$R2_sd
      pval_est <- perf_measures$pval_est
      rho_avg <- perf_measures$rho_avg
      rho_se <- perf_measures$rho_se
      rho_zscore <- perf_measures$rho_zscore
      rho_avg_squared <- perf_measures$rho_avg_squared
      zscore_pval <- perf_measures$zscore_pval
      # Fit on all data
      cv_fold_ids <- generate_fold_ids(length(adj_expression), n_folds)
      fit <- tryCatch(cv.glmnet(cis_gt, adj_expression, nfolds = n_folds, alpha = 0.5, type.measure='mse', foldid = cv_fold_ids, keep = TRUE),
                      error = function(cond) {message('Error'); message(geterrmessage()); list()})
      if (length(fit) > 0) {
        cv_R2_folds <- rep(0, n_folds)
        cv_corr_folds <- rep(0, n_folds)
        cv_zscore_folds <- rep(0, n_folds)
        cv_pval_folds <- rep(0, n_folds)
        best_lam_ind <- which.min(fit$cvm)
        for (j in 1:n_folds) {
          fold_idxs <- which(cv_fold_ids == j)
          adj_expr_fold_pred <- fit$fit.preval[fold_idxs, best_lam_ind]
          cv_R2_folds[j] <- calc_R2(adj_expression[fold_idxs], adj_expr_fold_pred)
          cv_corr_folds[j] <- ifelse(sd(adj_expr_fold_pred) != 0, cor(adj_expr_fold_pred, adj_expression[fold_idxs]), 0)
          cv_zscore_folds[j] <- atanh(cv_corr_folds[j])*sqrt(length(adj_expression[fold_idxs]) - 3) # Fisher transformation
          cv_pval_folds[j] <- ifelse(sd(adj_expr_fold_pred) != 0, cor.test(adj_expr_fold_pred, adj_expression[fold_idxs])$p.value, runif(1))
        }
        cv_R2_avg <- mean(cv_R2_folds)
        cv_R2_sd <- sd(cv_R2_folds)
        adj_expr_pred <- predict(fit, as.matrix(cis_gt), s = 'lambda.min')
        training_R2 <- calc_R2(adj_expression, adj_expr_pred)
        
        cv_rho_avg <- mean(cv_corr_folds)
        cv_rho_se <- sd(cv_corr_folds)
        cv_rho_avg_squared <- cv_rho_avg**2
        # Stouffer's method for combining z scores.
        cv_zscore_est <- sum(cv_zscore_folds) / sqrt(n_folds)
        cv_zscore_pval <- 2*pnorm(abs(cv_zscore_est), lower.tail = FALSE)
        cv_pval_est <- pchisq(-2 * sum(log(cv_pval_folds)), 2*n_folds, lower.tail = F)
        
        if (fit$nzero[best_lam_ind] > 0) {
          
          weights <- fit$glmnet.fit$beta[which(fit$glmnet.fit$beta[,best_lam_ind] != 0), best_lam_ind]
          weighted_snps <- names(fit$glmnet.fit$beta[,best_lam_ind])[which(fit$glmnet.fit$beta[,best_lam_ind] != 0)]
          weighted_snps_info <- snp_annot %>% filter(varID %in% weighted_snps) %>% select(rsid, varID, refAllele, effectAllele)
          weighted_snps_info$gene <- gene
          weighted_snps_info <- weighted_snps_info %>%
            merge(data.frame(weights = weights, varID=weighted_snps), by = 'varID') %>%
            select(gene, rsid, varID, refAllele, effectAllele, weights)
          write.table(weighted_snps_info, file = weights_file, append = TRUE, quote = FALSE, col.names = FALSE, row.names = FALSE, sep = '\t')
          covariance_df <- do_covariance(gene, cis_gt, weighted_snps_info$rsid, weighted_snps_info$varID)
          write.table(covariance_df, file = covariance_file, append = TRUE, quote = FALSE, col.names = FALSE, row.names = FALSE, sep = " ")
          model_summary <- c(gene, gene_name, alpha, ncol(cis_gt), fit$nzero[best_lam_ind], fit$lambda[best_lam_ind], R2_avg, R2_sd, cv_R2_avg, cv_R2_sd, training_R2, pval_est,
                             rho_avg, rho_se, rho_zscore, rho_avg_squared, zscore_pval, cv_rho_avg, cv_rho_se, cv_rho_avg_squared, cv_zscore_est, cv_zscore_pval, cv_pval_est)
        } else {
          model_summary <- c(gene, gene_name, alpha, ncol(cis_gt), 0, fit$lambda[best_lam_ind], R2_avg, R2_sd,
                             cv_R2_avg, cv_R2_sd, training_R2, pval_est, rho_avg, rho_se, rho_zscore, rho_avg_squared, zscore_pval,
                             cv_rho_avg, cv_rho_se, cv_rho_avg_squared, cv_zscore_est, cv_zscore_pval, cv_pval_est)
        }
      } else {
        model_summary <- c(gene, gene_name, alpha, ncol(cis_gt), 0, NA, R2_avg, R2_sd, NA, NA, NA, pval_est, rho_avg, rho_se, rho_zscore, rho_avg_squared, zscore_pval,
                           NA, NA, NA, NA, NA, NA)
      }
    }
    write(model_summary, file = model_summary_file, append = TRUE, ncol = 24, sep = '\t')

}


```

# elastic net: codes from MESA

# AA_elastic_net_fromMESA.r

```R
argv <- commandArgs(trailingOnly = TRUE)
my_i_index <- argv[1]
my_j_index <- argv[2]
my_alpha = argv[3]

path_AA="/net/mulan/home/shanglu/GENOA/data/AA"

suppressMessages(library(dplyr))
suppressMessages(library(glmnet))
suppressMessages((library(reshape2)))
suppressMessages(library(methods))

"%&%" <- function(a,b) paste(a,b, sep = "")


# generate_fold_ids
generate_fold_ids <- function(n_samples, n_folds=10) {
  n <- ceiling(n_samples / n_folds)
  fold_ids <- rep(1:n_folds, n)
  sample(fold_ids[1:n_samples])
}

# calc_R2
calc_R2 <- function(y, y_pred) {
  tss <- sum(y**2)
  rss <- sum((y - y_pred)**2)
  1 - rss/tss
}

# calc_corr
calc_corr <- function(y, y_pred) {
  sum(y*y_pred) / (sqrt(sum(y**2)) * sqrt(sum(y_pred**2)))
}

do_elastic_net <- function(cis_gt, expr_adj, n_folds, cv_fold_ids, n_times, alpha) {
  #tryCatch({
    cis_gt <- as.matrix(cis_gt)
    fit <- cv.glmnet(cis_gt, expr_adj, nfolds = n_folds, alpha = alpha, keep = TRUE, type.measure='mse', foldid = cv_fold_ids[,1], parallel = FALSE)
    lambda_seq <- fit$lambda
    cvms <- matrix(nrow=length(lambda_seq), ncol=n_times)
    fits <- list()
    fits[[1]] <- fit
    cvms <- matrix(nrow = 100, ncol = n_times)
    cvms[1:length(fit$cvm),1] <- fit$cvm
    for (i in 2:(n_times)) {
      fit <- cv.glmnet(cis_gt, expr_adj, lambda = lambda_seq, nfolds = n_folds, alpha = alpha, keep = FALSE, foldid = cv_fold_ids[,i], parallel = FALSE)
      fits[[i]] <- fit
      cvms[1:length(fit$cvm),i] <- fit$cvm
    }
    avg_cvm <- rowMeans(cvms)
    best_lam_ind <- which.min(avg_cvm)
    best_lambda <- lambda_seq[best_lam_ind]
    out <- list(cv_fit = fits[[1]], min_avg_cvm = min(avg_cvm, na.rm = T), best_lam_ind = best_lam_ind, best_lambda = best_lambda)
  #  out
  #},error = function(cond) {
  #  message('Error')
  #  message(geterrmessage())
  #  out <- list()
  #  out
  #}
  #)
  out
}


evaluate_performance <- function(cis_gt, expr_adj, fit, best_lam_ind, best_lambda, cv_fold_ids, n_folds) {
  n_nonzero <- fit$nzero[best_lam_ind]
  if (n_nonzero > 0) {
    R2 <- rep(0, n_folds)
    for (j in (1:n_folds)) {
      fold_idxs <- which(cv_fold_ids[,1] == j)
      tss <- sum(expr_adj[fold_idxs]**2)
      rss <- sum((expr_adj[fold_idxs] - fit$fit.preval[fold_idxs, best_lam_ind])**2)
      R2[j] <- 1 - (rss/tss)
    }
    best_fit <- fit$glmnet.fit
    expr_adj_pred <- predict(best_fit, as.matrix(cis_gt), s = best_lambda)
    tss <- sum(expr_adj**2)
    rss <- sum((expr_adj - expr_adj_pred)**2)
    
    n_samp <- length(expr_adj)
    #f_stat <- ((tss - rss) / n_nonzero) / (rss / (n_samp - ncol(cis_gt) - 1))
    #p_val <- pf(f_stat, n_samp - 1, n_samp - ncol(cis_gt) - 1, lower.tail = FALSE)
    weights <- best_fit$beta[which(best_fit$beta[,best_lam_ind] != 0), best_lam_ind]
    weighted_snps <- names(best_fit$beta[,best_lam_ind])[which(best_fit$beta[,best_lam_ind] != 0)]
    R2_mean <- mean(R2)
    R2_sd <- sd(R2)
    inR2 <- 1 - (rss/tss)
    # Old way
    pred_perf <- summary(lm(expr_adj ~ fit$fit.preval[,best_lam_ind]))
    pred_perf_rsq <- pred_perf$r.squared
    
    pred_perf_pval <- pred_perf$coef[2,4]
    #one_sided_pval <- cor.test(expr_adj, fit$fit.preval[,best_lam_ind], alternative = 'greater')$p.value
    out <- list(weights = weights, n_weights = n_nonzero, weighted_snps = weighted_snps, R2_mean = R2_mean, R2_sd = R2_sd,
                inR2 = inR2, pred_perf_rsq = pred_perf_rsq, pred_perf_pval = pred_perf_pval)
  } else {
    out <- list(weights = NA, n_weights = n_nonzero, weighted_snps = NA, R2_mean = NA, R2_sd = NA,
                inR2 = NA, pred_perf_rsq = NA, pred_perf_pval = NA)
  }
  out
}


do_covariance <- function(gene_id, cis_gt, rsids, varIDs, out_file) {
  model_gt <- cis_gt[,varIDs, drop=FALSE]
  geno_cov <- cov(model_gt)
  #print(dim(geno_cov))
  cov_df <- data.frame(gene=character(),rsid1=character(),rsid2=character(), covariance=double())
  for (i in 1:length(rsids)) {
    for (j in i:length(rsids)) {
      #print(c(i, j))
      cov_df <- tryCatch(rbind(cov_df, data.frame(gene=gene_id,rsid1=rsids[i], rsid2=rsids[j], covariance=geno_cov[i,j])),
                         error = function(cond) browser())
    }
  }
  write.table(cov_df, file = out_file, append = TRUE, quote = FALSE, col.names = FALSE, row.names = FALSE, sep = " ")
}



main <- function(i_index,j_index,  n_times=3, n_k_folds=10,seed=NA,  alpha=1) {
  # Read in data----
  # get_gene_annotation
load(paste0(path_AA,"/gene_AA_anno_order_correct.RData"))
# get_filtered_snp_annot
load(paste0(path_AA,"/gene_bed/chr_",i_index,"/chr_",i_index,"_gene_",j_index,"_snpinfo.RData"))
# get_gene_expression
expr_df = unlist(read.table(paste0(path_AA,"/gene_bed/chr_",i_index,"/chr_",i_index,"_gene_",j_index,".fam"))$V7)
# get_gene_coords
gene_index = which(gene_AA_anno_order$GENE %in% info$GENE[1])
gene_coords = c(gene_AA_anno_order$low[gene_index],gene_AA_anno_order$up[gene_index])
# get_cis_genotype
load(paste0(path_AA,"/gene_bed/chr_",i_index,"/chr_",i_index,"_gene_",j_index,".RData"))
cis_gt = as.matrix(a)
n_samples = dim(cis_gt)[1]
n_genes = 1

gene_id = info$GENE[1]
gene_name = info$GENE[1]

snp_annot = info	
snp_annot$varID = as.character(snp_annot$rs)
snp_annot$rsid = as.character(snp_annot$rs)
snp_annot$refAllele = as.character(snp_annot$allele1)
snp_annot$effectAllele = as.character(snp_annot$allele0)

  
  # Set seed and cross-validation fold ids----
  seed <- ifelse(is.na(seed), sample(1:1000000, 1), seed)
  set.seed(seed)
  cv_fold_ids <- matrix(nrow = n_samples, ncol = n_times)
  for (j in 1:n_times)
    cv_fold_ids[,j] <- sample(1:n_k_folds, n_samples, replace = TRUE)
  
  # Prepare output data----
  model_summary_file <- paste0('/net/mulan/home/shanglu/GENOA/analysis/prediction/predixcan/AA/chr_',i_index, "/AA" , '_chr_' ,i_index,'_gene_', j_index, '_alpha_',alpha,'_model_summaries.txt')
  model_summary_cols <- c('gene_id', 'gene_name', 'alpha', 'cv_mse', 'lambda_iteration', 'lambda_min', 'n_snps_in_model',
                          'cv_R2_avg', 'cv_R2_sd', 'in_sample_R2', 'pred_perf_R2', 'pred_perf_pval')
  write(model_summary_cols, file = model_summary_file, ncol = 12, sep = '\t')
  
  weights_file <- paste0('/net/mulan/home/shanglu/GENOA/analysis/prediction/predixcan/AA/chr_',i_index,'/AA_chr_',i_index,'_gene_',j_index,'_alpha_',alpha,'.txt')
  weights_col <- c('gene_id', 'rsid', 'varID', 'ref', 'alt', 'beta')
  write(weights_col, file = weights_file, ncol = 6, sep = '\t')
  
  tiss_chr_summ_f <- paste0('/net/mulan/home/shanglu/GENOA/analysis/prediction/predixcan/AA/chr_',i_index,'/AA_chr_',i_index,'_gene_',j_index,'_alpha_',alpha,'_summary.txt')
  tiss_chr_summ_col <- c('n_samples', 'chrom', 'cv_seed', 'n_genes')
  tiss_chr_summ <- data.frame(n_samples, i_index, seed, n_genes)
  colnames(tiss_chr_summ) <- tiss_chr_summ_col
  write.table(tiss_chr_summ, file = tiss_chr_summ_f, quote = FALSE, row.names = FALSE, sep = '\t')
  
  covariance_file <- paste0('/net/mulan/home/shanglu/GENOA/analysis/prediction/predixcan/AA/chr_',i_index,'/AA_chr_',i_index,'_gene_',j_index,'_alpha_',alpha,'_covariance.txt')
  covariance_col <- c('GENE', 'RSID1', 'RSID2', 'VALUE')
  write(covariance_col, file = covariance_file, ncol = 4, sep = ' ')
  
  # Attempt to build model for each gene----
    gene_name <- info$GENE[1]
    gene = gene_name
    coords <- gene_coords
    model_summary <- c(gene, gene_name, alpha, NA, NA, NA, 0, NA, NA, NA, NA, NA)

    if (ncol(cis_gt) >= 2) {
      expression_vec <- expr_df
      adj_expression <- expression_vec
      elnet_out <- do_elastic_net(cis_gt, adj_expression, n_k_folds, cv_fold_ids, n_times, alpha)
      if (length(elnet_out) > 0) {
        eval <- evaluate_performance(cis_gt, adj_expression, elnet_out$cv_fit, elnet_out$best_lam_ind, elnet_out$best_lambda, cv_fold_ids, n_k_folds)
        model_summary <- c(gene, gene_name, alpha, elnet_out$min_avg_cvm, elnet_out$best_lam_ind,
                           elnet_out$best_lambda, eval$n_weights, eval$R2_mean, eval$R2_sd, eval$inR2,
                           eval$pred_perf_rsq, eval$pred_perf_pval)
          #return(list(eval = eval, model_summary = model_summary, elnet_out = elnet_out, snp_annot = snp_annot))
        if (eval$n_weights > 0) {
          weighted_snps_info <- snp_annot %>% filter(varID %in% eval$weighted_snps) %>% select(rsid, varID, refAllele, effectAllele)
          if (nrow(weighted_snps_info) == 0)
            browser()
          weighted_snps_info$gene <- gene
          weighted_snps_info <- weighted_snps_info %>% merge(data.frame(weights = eval$weights, varID=eval$weighted_snps), by = 'varID') %>%
                                  select(gene, rsid, varID, refAllele, effectAllele, weights)
          write.table(weighted_snps_info, file = weights_file, append = TRUE, quote = FALSE, col.names = FALSE, row.names = FALSE, sep = '\t')
          do_covariance(gene, cis_gt, weighted_snps_info$rsid, weighted_snps_info$varID, covariance_file)
        }
      }
    }
    write(model_summary, file = model_summary_file, append = TRUE, ncol = 12, sep = '\t')
  }



main(my_i_index,my_j_index,  n_times=3, n_k_folds=10,seed=NA,  alpha=my_alpha)

```

# EA_elastic_net_fromMESA.r
```R
argv <- commandArgs(trailingOnly = TRUE)
my_i_index <- argv[1]
my_j_index <- argv[2]
my_alpha = argv[3]


path_EA="/net/mulan/home/shanglu/GENOA/data/EA"

suppressMessages(library(dplyr))
suppressMessages(library(glmnet))
suppressMessages((library(reshape2)))
suppressMessages(library(methods))

"%&%" <- function(a,b) paste(a,b, sep = "")


# generate_fold_ids
generate_fold_ids <- function(n_samples, n_folds=10) {
  n <- ceiling(n_samples / n_folds)
  fold_ids <- rep(1:n_folds, n)
  sample(fold_ids[1:n_samples])
}

# calc_R2
calc_R2 <- function(y, y_pred) {
  tss <- sum(y**2)
  rss <- sum((y - y_pred)**2)
  1 - rss/tss
}

# calc_corr
calc_corr <- function(y, y_pred) {
  sum(y*y_pred) / (sqrt(sum(y**2)) * sqrt(sum(y_pred**2)))
}

do_elastic_net <- function(cis_gt, expr_adj, n_folds, cv_fold_ids, n_times, alpha) {
  #tryCatch({
    cis_gt <- as.matrix(cis_gt)
    fit <- cv.glmnet(cis_gt, expr_adj, nfolds = n_folds, alpha = alpha, keep = TRUE, type.measure='mse', foldid = cv_fold_ids[,1], parallel = FALSE)
    lambda_seq <- fit$lambda
    cvms <- matrix(nrow=length(lambda_seq), ncol=n_times)
    fits <- list()
    fits[[1]] <- fit
    cvms <- matrix(nrow = 100, ncol = n_times)
    cvms[1:length(fit$cvm),1] <- fit$cvm
    for (i in 2:(n_times)) {
      fit <- cv.glmnet(cis_gt, expr_adj, lambda = lambda_seq, nfolds = n_folds, alpha = alpha, keep = FALSE, foldid = cv_fold_ids[,i], parallel = FALSE)
      fits[[i]] <- fit
      cvms[1:length(fit$cvm),i] <- fit$cvm
    }
    avg_cvm <- rowMeans(cvms)
    best_lam_ind <- which.min(avg_cvm)
    best_lambda <- lambda_seq[best_lam_ind]
    out <- list(cv_fit = fits[[1]], min_avg_cvm = min(avg_cvm, na.rm = T), best_lam_ind = best_lam_ind, best_lambda = best_lambda)
  #  out
  #},error = function(cond) {
  #  message('Error')
  #  message(geterrmessage())
  #  out <- list()
  #  out
  #}
  #)
  out
}


evaluate_performance <- function(cis_gt, expr_adj, fit, best_lam_ind, best_lambda, cv_fold_ids, n_folds) {
  n_nonzero <- fit$nzero[best_lam_ind]
  if (n_nonzero > 0) {
    R2 <- rep(0, n_folds)
    for (j in (1:n_folds)) {
      fold_idxs <- which(cv_fold_ids[,1] == j)
      tss <- sum(expr_adj[fold_idxs]**2)
      rss <- sum((expr_adj[fold_idxs] - fit$fit.preval[fold_idxs, best_lam_ind])**2)
      R2[j] <- 1 - (rss/tss)
    }
    best_fit <- fit$glmnet.fit
    expr_adj_pred <- predict(best_fit, as.matrix(cis_gt), s = best_lambda)
    tss <- sum(expr_adj**2)
    rss <- sum((expr_adj - expr_adj_pred)**2)
    
    n_samp <- length(expr_adj)
    #f_stat <- ((tss - rss) / n_nonzero) / (rss / (n_samp - ncol(cis_gt) - 1))
    #p_val <- pf(f_stat, n_samp - 1, n_samp - ncol(cis_gt) - 1, lower.tail = FALSE)
    weights <- best_fit$beta[which(best_fit$beta[,best_lam_ind] != 0), best_lam_ind]
    weighted_snps <- names(best_fit$beta[,best_lam_ind])[which(best_fit$beta[,best_lam_ind] != 0)]
    R2_mean <- mean(R2)
    R2_sd <- sd(R2)
    inR2 <- 1 - (rss/tss)
    # Old way
    pred_perf <- summary(lm(expr_adj ~ fit$fit.preval[,best_lam_ind]))
    pred_perf_rsq <- pred_perf$r.squared
    
    pred_perf_pval <- pred_perf$coef[2,4]
    #one_sided_pval <- cor.test(expr_adj, fit$fit.preval[,best_lam_ind], alternative = 'greater')$p.value
    out <- list(weights = weights, n_weights = n_nonzero, weighted_snps = weighted_snps, R2_mean = R2_mean, R2_sd = R2_sd,
                inR2 = inR2, pred_perf_rsq = pred_perf_rsq, pred_perf_pval = pred_perf_pval)
  } else {
    out <- list(weights = NA, n_weights = n_nonzero, weighted_snps = NA, R2_mean = NA, R2_sd = NA,
                inR2 = NA, pred_perf_rsq = NA, pred_perf_pval = NA)
  }
  out
}


do_covariance <- function(gene_id, cis_gt, rsids, varIDs, out_file) {
  model_gt <- cis_gt[,varIDs, drop=FALSE]
  geno_cov <- cov(model_gt)
  #print(dim(geno_cov))
  cov_df <- data.frame(gene=character(),rsid1=character(),rsid2=character(), covariance=double())
  for (i in 1:length(rsids)) {
    for (j in i:length(rsids)) {
      #print(c(i, j))
      cov_df <- tryCatch(rbind(cov_df, data.frame(gene=gene_id,rsid1=rsids[i], rsid2=rsids[j], covariance=geno_cov[i,j])),
                         error = function(cond) browser())
    }
  }
  write.table(cov_df, file = out_file, append = TRUE, quote = FALSE, col.names = FALSE, row.names = FALSE, sep = " ")
}



main <- function(i_index,j_index,  n_times=3, n_k_folds=10,seed=NA,  alpha=1) {
  # Read in data----
  # get_gene_annotation
load(paste0(path_EA,"/gene_EA_anno_order_correct.RData"))
# get_filtered_snp_annot
load(paste0(path_EA,"/gene_bed/chr_",i_index,"/chr_",i_index,"_gene_",j_index,"_snpinfo.RData"))
# get_gene_expression
expr_df = unlist(read.table(paste0(path_EA,"/gene_bed/chr_",i_index,"/chr_",i_index,"_gene_",j_index,".fam"))$V7)
# get_gene_coords
gene_index = which(gene_EA_anno_order$GENE %in% info$GENE[1])
gene_coords = c(gene_EA_anno_order$low[gene_index],gene_EA_anno_order$up[gene_index])
# get_cis_genotype
load(paste0(path_EA,"/gene_bed/chr_",i_index,"/chr_",i_index,"_gene_",j_index,".RData"))
cis_gt = as.matrix(a)
n_samples = dim(cis_gt)[1]
n_genes = 1

gene_id = info$GENE[1]
gene_name = info$GENE[1]

snp_annot = info	
snp_annot$varID = as.character(snp_annot$rs)
snp_annot$rsid = as.character(snp_annot$rs)
snp_annot$refAllele = as.character(snp_annot$allele1)
snp_annot$effectAllele = as.character(snp_annot$allele0)

  
  # Set seed and cross-validation fold ids----
  seed <- ifelse(is.na(seed), sample(1:1000000, 1), seed)
  set.seed(seed)
  cv_fold_ids <- matrix(nrow = n_samples, ncol = n_times)
  for (j in 1:n_times)
    cv_fold_ids[,j] <- sample(1:n_k_folds, n_samples, replace = TRUE)
  
  # Prepare output data----
  model_summary_file <- paste0('/net/mulan/home/shanglu/GENOA/analysis/prediction/predixcan/EA/chr_',i_index, "/EA" , '_chr_' ,i_index,'_gene_', j_index, '_alpha_',alpha,'_model_summaries.txt')
  model_summary_cols <- c('gene_id', 'gene_name', 'alpha', 'cv_mse', 'lambda_iteration', 'lambda_min', 'n_snps_in_model',
                          'cv_R2_avg', 'cv_R2_sd', 'in_sample_R2', 'pred_perf_R2', 'pred_perf_pval')
  write(model_summary_cols, file = model_summary_file, ncol = 12, sep = '\t')
  
  weights_file <- paste0('/net/mulan/home/shanglu/GENOA/analysis/prediction/predixcan/EA/chr_',i_index,'/EA_chr_',i_index,'_gene_',j_index,'_alpha_',alpha,'.txt')
  weights_col <- c('gene_id', 'rsid', 'varID', 'ref', 'alt', 'beta')
  write(weights_col, file = weights_file, ncol = 6, sep = '\t')
  
  tiss_chr_summ_f <- paste0('/net/mulan/home/shanglu/GENOA/analysis/prediction/predixcan/EA/chr_',i_index,'/EA_chr_',i_index,'_gene_',j_index,'_alpha_',alpha,'_summary.txt')
  tiss_chr_summ_col <- c('n_samples', 'chrom', 'cv_seed', 'n_genes')
  tiss_chr_summ <- data.frame(n_samples, i_index, seed, n_genes)
  colnames(tiss_chr_summ) <- tiss_chr_summ_col
  write.table(tiss_chr_summ, file = tiss_chr_summ_f, quote = FALSE, row.names = FALSE, sep = '\t')
  
  covariance_file <- paste0('/net/mulan/home/shanglu/GENOA/analysis/prediction/predixcan/EA/chr_',i_index,'/EA_chr_',i_index,'_gene_',j_index,'_alpha_',alpha,'_covariance.txt')
  covariance_col <- c('GENE', 'RSID1', 'RSID2', 'VALUE')
  write(covariance_col, file = covariance_file, ncol = 4, sep = ' ')
  
  # Attempt to build model for each gene----
    gene_name <- info$GENE[1]
    gene = gene_name
    coords <- gene_coords
    model_summary <- c(gene, gene_name, alpha, NA, NA, NA, 0, NA, NA, NA, NA, NA)

    if (ncol(cis_gt) >= 2) {
      expression_vec <- expr_df
      adj_expression <- expression_vec
      elnet_out <- do_elastic_net(cis_gt, adj_expression, n_k_folds, cv_fold_ids, n_times, alpha)
      if (length(elnet_out) > 0) {
        eval <- evaluate_performance(cis_gt, adj_expression, elnet_out$cv_fit, elnet_out$best_lam_ind, elnet_out$best_lambda, cv_fold_ids, n_k_folds)
        model_summary <- c(gene, gene_name, alpha, elnet_out$min_avg_cvm, elnet_out$best_lam_ind,
                           elnet_out$best_lambda, eval$n_weights, eval$R2_mean, eval$R2_sd, eval$inR2,
                           eval$pred_perf_rsq, eval$pred_perf_pval)
          #return(list(eval = eval, model_summary = model_summary, elnet_out = elnet_out, snp_annot = snp_annot))
        if (eval$n_weights > 0) {
          weighted_snps_info <- snp_annot %>% filter(varID %in% eval$weighted_snps) %>% select(rsid, varID, refAllele, effectAllele)
          if (nrow(weighted_snps_info) == 0)
            browser()
          weighted_snps_info$gene <- gene
          weighted_snps_info <- weighted_snps_info %>% merge(data.frame(weights = eval$weights, varID=eval$weighted_snps), by = 'varID') %>%
                                  select(gene, rsid, varID, refAllele, effectAllele, weights)
          write.table(weighted_snps_info, file = weights_file, append = TRUE, quote = FALSE, col.names = FALSE, row.names = FALSE, sep = '\t')
          do_covariance(gene, cis_gt, weighted_snps_info$rsid, weighted_snps_info$varID, covariance_file)
        }
      }
    }
    write(model_summary, file = model_summary_file, append = TRUE, ncol = 12, sep = '\t')
  }



main(my_i_index,my_j_index,  n_times=3, n_k_folds=10,seed=NA,  alpha=my_alpha)

```



# AA_pred_run.sh
```
#!/bin/bash
#SBATCH --job-name=AApred
#SBATCH --array=1-66
#SBATCH --cpus-per-task=1
#SBATCH --output=/net/mulan/home/shanglu/GENOA/data/AA/out/AApred%a.out
#SBATCH --error=/net/mulan/home/shanglu/GENOA/data/AA/err/AApred%a.err
#SBATCH --workdir=/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping/code
#SBATCH --partition=mulan

bash

let k=0

pathGfile=/net/mulan/home/shanglu/GENOA/data/AA/Gene_file

for i in `seq 1 22`
do
gnum=`cat ${pathGfile}/chr${i}.num`
let k=${k}+1
if [ ${k} -eq ${SLURM_ARRAY_TASK_ID} ]; then
for j in `seq 1 ${gnum}`
do
alpha=0.05
Rscript ./AA_elastic_net_fromMESA.r $i $j $alpha
done
fi
let k=${k}+1
if [ ${k} -eq ${SLURM_ARRAY_TASK_ID} ]; then
for j in `seq 1 ${gnum}`
do
alpha=1
Rscript ./AA_elastic_net_fromMESA.r $i $j $alpha
done
fi
let k=${k}+1
if [ ${k} -eq ${SLURM_ARRAY_TASK_ID} ]; then
for j in `seq 1 ${gnum}`
do
alpha=0.5
Rscript ./AA_elastic_net_fromMESA.r $i $j $alpha
done
fi
done 
```



# EA_pred_run.sh
```
#!/bin/bash
#SBATCH --job-name=AApred
#SBATCH --array=1-10000%250
#SBATCH --cpus-per-task=1
#SBATCH --output=/net/mulan/home/shanglu/GENOA/data/AA/out/AApred%a.out
#SBATCH --error=/net/mulan/home/shanglu/GENOA/data/AA/err/AApred%a.err
#SBATCH --workdir=/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping/code
#SBATCH --partition=mulan

bash

let k=0

pathGfile=/net/mulan/home/shanglu/GENOA/data/AA/Gene_file

for i in `seq 1 9`
do
gnum=`cat ${pathGfile}/chr${i}.num`
for j in `seq 1 ${gnum}`
do
let k=${k}+1
if [ ${k} -eq ${SLURM_ARRAY_TASK_ID} ]; then
alpha=0.05
Rscript ./AA_elastic_net_fromMESA.r $i $j $alpha
alpha=1
Rscript ./AA_elastic_net_fromMESA.r $i $j $alpha
alpha=0.5
Rscript ./AA_elastic_net_fromMESA.r $i $j $alpha
fi
done
done



#!/bin/bash
#SBATCH --job-name=AApred
#SBATCH --array=1-10000%250
#SBATCH --cpus-per-task=1
#SBATCH --output=/net/mulan/home/shanglu/GENOA/data/AA/out/AApred%a.out
#SBATCH --error=/net/mulan/home/shanglu/GENOA/data/AA/err/AApred%a.err
#SBATCH --workdir=/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping/code
#SBATCH --partition=mulan

bash

let k=0

pathGfile=/net/mulan/home/shanglu/GENOA/data/AA/Gene_file

for i in `seq 10 22`
do
gnum=`cat ${pathGfile}/chr${i}.num`
for j in `seq 1 ${gnum}`
do
let k=${k}+1
if [ ${k} -eq ${SLURM_ARRAY_TASK_ID} ]; then
alpha=0.05
Rscript ./AA_elastic_net_fromMESA.r $i $j $alpha
alpha=1
Rscript ./AA_elastic_net_fromMESA.r $i $j $alpha
alpha=0.5
Rscript ./AA_elastic_net_fromMESA.r $i $j $alpha
fi
done
done
```



















