
We have:

1) GEMMA lmm weights in GENOA (AA & EA)
2) GEMMA bslmm weights in GENOA (AA & EA)
3) elastic net weights in GENOA (AA & EA)
4) MatrixeQTL weights in MESA (AFA & CAU & HIS)
5) elastic net weights in MESA (AFA & CAU & HIS)

we need to compare GENOA and MESA in elastic net, or lmm, or bslmm 

1) GEMMA lmm vs MatrixeQTL weights, use same gene-SNP pairs
2) GEMMA bslmm vs MatrixeQTL weights, use same gene-SNP pairs
3) elastic net, use 100kb region SNPs in both GENOA and MESA
4) elastic net, use all avaliable SNPs in both GENOA and MESA

want to compare:
1) genes with rho >0, rho_square
2) all gene, rho_square


```R

path_data_AA = "/net/mulan/home/shanglu/GENOA/data/AA"
path_geu = "/net/mulan/home/shanglu/GENOA/data/Geuvadis"
path_pred_GENOA = "/net/mulan/home/shanglu/GENOA/analysis/prediction/predixcan"
path_pred_MESA = "/net/mulan/home/shanglu/GENOA/data/MESA"

#----------------------------------------------------------------
# in mesa, restrict snps to 100kb region
  # first map rsid to position, and annotate gene location
  # then take subset
  
  
library(dplyr)
library(RSQLite)
"%&%" <- function(a,b) paste(a,b,sep='')

driver <- dbDriver("SQLite")

tissues <- c('AFA','HIS','CAU')

for (tissue in tissues) {
  print(tissue)
  unfiltered_db <- paste0("/net/mulan/home/shanglu/GENOA/data/MESA/",tissue,"_imputed_10_peer_3_pcs_2.db")
  in_conn <- dbConnect(driver, unfiltered_db)
  model_summaries <- dbGetQuery(in_conn, 'select * from model_summaries')
  weights <- dbGetQuery(in_conn, 'select * from weights')
  save(weights, file = paste0(tissue,"_MESA_weights.RData"))
  save(model_summaries, file = paste0(tissue,"_MESA_model_summaries.RData"))
}

#------------------------------------------------------
# AA elastic net result collect
#------------------------------------------------------

args <- as.numeric(commandArgs(TRUE))
i=args[1]
print(i)

path_AA="/net/mulan/home/shanglu/GENOA/data/AA"
path_AA_result="/net/mulan/home/shanglu/GENOA/analysis/prediction/predixcan"
path_AA_output="/net/mulan/home/shanglu/GENOA/analysis/prediction/predixcan/AA/"
load(paste0(path_AA,"/gene_AA_anno_protein.RData"))

gene_anno_sub_AA_chr = gene_AA_anno[gene_AA_anno$chr==i,]

gene_id = NULL
rsid = NULL
varID = NULL
ref = NULL
alt = NULL
beta = NULL
AA_elnet_table = data.frame(gene_id,rsid ,varID,ref,alt,beta)

weight = list()
for(j in 1:dim(gene_anno_sub_AA_chr)[1]){
	print(j)
	fn = paste0(path_AA_output,"/chr_",i,"/AA_chr_",i,"_gene_",j,"_alpha_0.5.txt")
	if(file.exists(fn)){
	weight[[j]] = read.table(fn,header=T)
	AA_elnet_table = rbind(AA_elnet_table, weight[[j]])
	}
}

save(AA_elnet_table, file = paste0(path_AA_result,"/AA_elnet_weight_chr_",i,".RData"))


#------------------------------------------------------
# EA elastic net result collect
#------------------------------------------------------
args <- as.numeric(commandArgs(TRUE))
i=args[1]
print(i)

path_EA="/net/mulan/home/shanglu/GENOA/data/EA"
path_EA_result="/net/mulan/home/shanglu/GENOA/analysis/prediction/predixcan"
path_EA_output="/net/mulan/home/shanglu/GENOA/analysis/prediction/predixcan/EA/"
load(paste0(path_EA,"/gene_EA_anno_protein.RData"))

gene_anno_sub_EA_chr = gene_EA_anno[gene_EA_anno$chr==i,]

gene_id = NULL
rsid = NULL
varID = NULL
ref = NULL
alt = NULL
beta = NULL
EA_elnet_table = data.frame(gene_id,rsid ,varID,ref,alt,beta)

weight = list()
for(j in 1:dim(gene_anno_sub_EA_chr)[1]){
	print(j)
	fn = paste0(path_EA_output,"/chr_",i,"/EA_chr_",i,"_gene_",j,"_alpha_0.5.txt")
	if(file.exists(fn)){
	weight[[j]] = read.table(fn,header=T)
	EA_elnet_table = rbind(EA_elnet_table, weight[[j]])
	}
}

save(EA_elnet_table, file = paste0(path_EA_result,"/EA_elnet_weight_chr_",i,".RData"))




#----------------------------------------------------------------

# gene annotation:
load(paste0(path_data_AA,"/gene_AA_anno_order_correct.RData"))

#----------------------------------------------------------------
# snp annotation: 
  # could use geuvadis annotation
i_index = 1
bim = read.table(paste0(path_geu,"/chr",i_index,".bim"))
bim$V2 = as.character(bim$V2)

#----------------------------------------------------------------

##############################
# use +/-100kb region in MESA
##############################

# load MESA unfiltered weights

pop = c("AFA", "CAU", "HIS")
for(popnum in 1:3){

print(popnum)

load(paste0(path_pred_MESA,"/",pop[popnum],"_MESA_weights.RData"))

gene_name = as.character(weights$gene)
gene_name01 = strsplit(gene_name,"[.]")
gene_name02 = c()
for(i in 1:length(gene_name01)){gene_name02[i]=paste0(gene_name01[[i]][1])}
weights$gene = gene_name02

MESA_snploc = as.character(weights$varID)
MESA_snploc01 = strsplit(MESA_snploc,"[_]")
MESA_snplocs = c()
for(i in 1:length(MESA_snploc01)){MESA_snplocs[i]=paste0(MESA_snploc01[[i]][2])}
MESA_snplocs = as.numeric(MESA_snplocs)

weights$MESA_snplocs = MESA_snplocs

ind = match( weights$gene,gene_AA_anno_order$GENE)

weights$low = gene_AA_anno_order$low[ind]
weights$up = gene_AA_anno_order$up[ind]
weights$chr = gene_AA_anno_order$chr[ind]

weights$left = weights$low-100000
weights$right = weights$up+100000

weights$inside = (weights$MESA_snplocs>=weights$left)*(weights$MESA_snplocs<=weights$right)
weights = na.omit(weights)

weights_100kb = weights[weights$inside==1,]

save(weights_100kb, file = paste0(pop[popnum],"_weights_100kb.RData"))
}

# AFA
> dim(weights_100kb)
[1] 64319    13
# CAU
> dim(weights_100kb)
[1] 85437    13
# HIS
> dim(weights_100kb)
[1] 75275    13

##############################
# use 1Mb region in MESA
##############################

pop = c("AFA", "CAU", "HIS")
for(popnum in 1:3){

print(popnum)

load(paste0(path_pred_MESA,"/",pop[popnum],"_MESA_weights.RData"))

gene_name = as.character(weights$gene)
gene_name01 = strsplit(gene_name,"[.]")
gene_name02 = c()
for(i in 1:length(gene_name01)){gene_name02[i]=paste0(gene_name01[[i]][1])}
weights$gene = gene_name02

MESA_snploc = as.character(weights$varID)
MESA_snploc01 = strsplit(MESA_snploc,"[_]")
MESA_snplocs = c()
for(i in 1:length(MESA_snploc01)){MESA_snplocs[i]=paste0(MESA_snploc01[[i]][2])}
MESA_snplocs = as.numeric(MESA_snplocs)

weights$MESA_snplocs = MESA_snplocs

ind = match( weights$gene,gene_AA_anno_order$GENE)

weights$low = gene_AA_anno_order$low[ind]
weights$up = gene_AA_anno_order$up[ind]
weights$chr = gene_AA_anno_order$chr[ind]

weights_1Mb = na.omit(weights)
save(weights_1Mb, file = paste0(pop[popnum],"_weights_1Mb.RData"))
}

# AFA
> dim(weights_1Mb)
[1] 183915     10
# CAU
> dim(weights_1Mb)
[1] 201194     10
# HIS
> dim(weights_1Mb)
[1] 203752     10


##############################
# use +/-100kb region in GENOA
##############################

path_predixcan = "/net/mulan/home/shanglu/GENOA/analysis/prediction/predixcan"
path_elnet = "/net/mulan/home/shanglu/GENOA/analysis/prediction/elnet"
path_pred = "/net/mulan/home/shanglu/GENOA/analysis/prediction"
path = "/net/mulan/home/shanglu/GENOA/data/Geuvadis"
gene_expr_geuvadis = read.table(paste0(path,"/Geuvadis_gene_expression.txt"),header=T)
gene_exp_ping = read.table(paste0(path,"/Geuvadis_gene_expression_ping.txt"),header=T)

gene_name = as.character(gene_exp_ping$TargetID)
gene_name01 = strsplit(gene_name,"[.]")
gene_name02 = c()
for(i in 1:length(gene_name01)){gene_name02[i]=paste0(gene_name01[[i]][1])}
gene_name02=unlist(gene_name02)
gene_exp_ping$ENSG = gene_name02


```


```R

#---------------------------
# for AA
#---------------------------


for(i in 1:22){
	print(i)  
  load(paste0(path_predixcan,"/AA_elnet_weight_chr_",i,".RData"))
  AA_elnet_table$GENE = as.character(AA_elnet_table$gene_id)
  AA_elnet_table$SNP = as.character(AA_elnet_table$rsid)
  AA_elnet_table$allele1 = as.character(AA_elnet_table$ref)
  AA_elnet_table$weight = as.character(AA_elnet_table$beta)
  AA_chr = AA_elnet_table
	genes_aa = unique(as.character(AA_chr$GENE))
	genes_geu = unique(as.character(gene_exp_ping$ENSG)[which(gene_exp_ping$Chr==i)])
	common_gene = intersect(genes_aa,genes_geu)
	AA_keep = AA_chr[which(as.character(AA_chr$GENE) %in% common_gene),]
	bim = paste0(path,"/chr",i,".bim")
	Geu_snp = read.table(bim)
	common_snp = intersect(as.character(Geu_snp$V2),as.character(AA_keep$SNP))
	ind_geu = which(as.character(Geu_snp$V2) %in% common_snp)
	ind_aa = which(as.character(AA_keep$SNP) %in% common_snp)
	AA_to_use = AA_keep[ind_aa,]
	geu_to_use = Geu_snp[ind_geu,]
	index = match( as.character(AA_to_use$SNP),as.character(Geu_snp$V2))
	AA_to_use$index_in_geu = index
	AA_to_use$minor_geu = as.character(Geu_snp$V5)[index]
	AA_to_use$major_geu = as.character(Geu_snp$V6)[index]
	save(AA_to_use, file = paste0(path_pred,"/AA/elnet_AA_to_use_chr_",i,".RData"))
	print(dim(AA_to_use))
}

#---------------------------
# for EA
#---------------------------


for(i in 1:22){
	print(i)  
  load(paste0(path_predixcan,"/EA_elnet_weight_chr_",i,".RData"))
  EA_elnet_table$GENE = as.character(EA_elnet_table$gene_id)
  EA_elnet_table$SNP = as.character(EA_elnet_table$rsid)
  EA_elnet_table$allele1 = as.character(EA_elnet_table$ref)
  EA_elnet_table$weight = as.character(EA_elnet_table$beta)
  EA_chr = EA_elnet_table
	genes_EA = unique(as.character(EA_chr$GENE))
	genes_geu = unique(as.character(gene_exp_ping$ENSG)[which(gene_exp_ping$Chr==i)])
	common_gene = intersect(genes_EA,genes_geu)
	EA_keep = EA_chr[which(as.character(EA_chr$GENE) %in% common_gene),]
	bim = paste0(path,"/chr",i,".bim")
	Geu_snp = read.table(bim)
	common_snp = intersect(as.character(Geu_snp$V2),as.character(EA_keep$SNP))
	ind_geu = which(as.character(Geu_snp$V2) %in% common_snp)
	ind_EA = which(as.character(EA_keep$SNP) %in% common_snp)
	EA_to_use = EA_keep[ind_EA,]
	geu_to_use = Geu_snp[ind_geu,]
	index = match( as.character(EA_to_use$SNP),as.character(Geu_snp$V2))
	EA_to_use$index_in_geu = index
	EA_to_use$minor_geu = as.character(Geu_snp$V5)[index]
	EA_to_use$major_geu = as.character(Geu_snp$V6)[index]
	save(EA_to_use, file = paste0(path_pred,"/EA/elnet_EA_to_use_chr_",i,".RData"))
	print(dim(EA_to_use))
}

#---------------------------
# for AFA
#---------------------------

path_MESA = "/net/mulan/home/shanglu/GENOA/data/MESA"
pop = c("AFA", "CAU", "HIS")

popnum = 1
load(paste0(path_MESA,"/",pop[popnum],"_weights_100kb.RData"))


for(i in 1:22){
	print(i)  
  AFA_elnet_table = weights_100kb[weights_100kb$chr==i,]
  AFA_elnet_table$GENE = as.character(AFA_elnet_table$gene)
  AFA_elnet_table$SNP = as.character(AFA_elnet_table$rsid)
  AFA_elnet_table$allele1 = as.character(AFA_elnet_table$eff_allele)
  AFA_chr = AFA_elnet_table
	genes_AFA = unique(as.character(AFA_chr$GENE))
	genes_geu = unique(as.character(gene_exp_ping$ENSG)[which(gene_exp_ping$Chr==i)])
	common_gene = intersect(genes_AFA,genes_geu)
	AFA_keep = AFA_chr[which(as.character(AFA_chr$GENE) %in% common_gene),]
	bim = paste0(path,"/chr",i,".bim")
	Geu_snp = read.table(bim)
	common_snp = intersect(as.character(Geu_snp$V2),as.character(AFA_keep$SNP))
	ind_geu = which(as.character(Geu_snp$V2) %in% common_snp)
	ind_AFA = which(as.character(AFA_keep$SNP) %in% common_snp)
	AFA_to_use = AFA_keep[ind_AFA,]
	geu_to_use = Geu_snp[ind_geu,]
	index = match( as.character(AFA_to_use$SNP),as.character(Geu_snp$V2))
	AFA_to_use$index_in_geu = index
	AFA_to_use$minor_geu = as.character(Geu_snp$V5)[index]
	AFA_to_use$major_geu = as.character(Geu_snp$V6)[index]
	save(AFA_to_use, file = paste0(path_pred,"/AFA/elnet_100kb_AFA_to_use_chr_",i,".RData"))
	print(dim(AFA_to_use))
}


#---------------------------
# for CAU
#---------------------------

path_MESA = "/net/mulan/home/shanglu/GENOA/data/MESA"
pop = c("AFA", "CAU", "HIS")

popnum = 2
load(paste0(path_MESA,"/",pop[popnum],"_weights_100kb.RData"))


for(i in 1:22){
	print(i)  
  CAU_elnet_table = weights_100kb[weights_100kb$chr==i,]
  CAU_elnet_table$GENE = as.character(CAU_elnet_table$gene)
  CAU_elnet_table$SNP = as.character(CAU_elnet_table$rsid)
  CAU_elnet_table$allele1 = as.character(CAU_elnet_table$eff_allele)
  CAU_chr = CAU_elnet_table
	genes_CAU = unique(as.character(CAU_chr$GENE))
	genes_geu = unique(as.character(gene_exp_ping$ENSG)[which(gene_exp_ping$Chr==i)])
	common_gene = intersect(genes_CAU,genes_geu)
	CAU_keep = CAU_chr[which(as.character(CAU_chr$GENE) %in% common_gene),]
	bim = paste0(path,"/chr",i,".bim")
	Geu_snp = read.table(bim)
	common_snp = intersect(as.character(Geu_snp$V2),as.character(CAU_keep$SNP))
	ind_geu = which(as.character(Geu_snp$V2) %in% common_snp)
	ind_CAU = which(as.character(CAU_keep$SNP) %in% common_snp)
	CAU_to_use = CAU_keep[ind_CAU,]
	geu_to_use = Geu_snp[ind_geu,]
	index = match( as.character(CAU_to_use$SNP),as.character(Geu_snp$V2))
	CAU_to_use$index_in_geu = index
	CAU_to_use$minor_geu = as.character(Geu_snp$V5)[index]
	CAU_to_use$major_geu = as.character(Geu_snp$V6)[index]
	save(CAU_to_use, file = paste0(path_pred,"/CAU/elnet_100kb_CAU_to_use_chr_",i,".RData"))
	print(dim(CAU_to_use))
}


#---------------------------
# for HIS
#---------------------------

path_MESA = "/net/mulan/home/shanglu/GENOA/data/MESA"
pop = c("AFA", "CAU", "HIS")

popnum = 3
load(paste0(path_MESA,"/",pop[popnum],"_weights_100kb.RData"))


for(i in 1:22){
	print(i)  
  HIS_elnet_table = weights_100kb[weights_100kb$chr==i,]
  HIS_elnet_table$GENE = as.character(HIS_elnet_table$gene)
  HIS_elnet_table$SNP = as.character(HIS_elnet_table$rsid)
  HIS_elnet_table$allele1 = as.character(HIS_elnet_table$eff_allele)
  HIS_chr = HIS_elnet_table
	genes_HIS = unique(as.character(HIS_chr$GENE))
	genes_geu = unique(as.character(gene_exp_ping$ENSG)[which(gene_exp_ping$Chr==i)])
	common_gene = intersect(genes_HIS,genes_geu)
	HIS_keep = HIS_chr[which(as.character(HIS_chr$GENE) %in% common_gene),]
	bim = paste0(path,"/chr",i,".bim")
	Geu_snp = read.table(bim)
	common_snp = intersect(as.character(Geu_snp$V2),as.character(HIS_keep$SNP))
	ind_geu = which(as.character(Geu_snp$V2) %in% common_snp)
	ind_HIS = which(as.character(HIS_keep$SNP) %in% common_snp)
	HIS_to_use = HIS_keep[ind_HIS,]
	geu_to_use = Geu_snp[ind_geu,]
	index = match( as.character(HIS_to_use$SNP),as.character(Geu_snp$V2))
	HIS_to_use$index_in_geu = index
	HIS_to_use$minor_geu = as.character(Geu_snp$V5)[index]
	HIS_to_use$major_geu = as.character(Geu_snp$V6)[index]
	save(HIS_to_use, file = paste0(path_pred,"/HIS/elnet_100kb_HIS_to_use_chr_",i,".RData"))
	print(dim(HIS_to_use))
}



#---------------------------
# for AFA 1Mb
#---------------------------

path_MESA = "/net/mulan/home/shanglu/GENOA/data/MESA"
pop = c("AFA", "CAU", "HIS")

popnum = 1
load(paste0(path_MESA,"/",pop[popnum],"_weights_1Mb.RData"))


for(i in 1:22){
	print(i)  
  AFA_elnet_table = weights_1Mb[weights_1Mb$chr==i,]
  AFA_elnet_table$GENE = as.character(AFA_elnet_table$gene)
  AFA_elnet_table$SNP = as.character(AFA_elnet_table$rsid)
  AFA_elnet_table$allele1 = as.character(AFA_elnet_table$eff_allele)
  AFA_chr = AFA_elnet_table
	genes_AFA = unique(as.character(AFA_chr$GENE))
	genes_geu = unique(as.character(gene_exp_ping$ENSG)[which(gene_exp_ping$Chr==i)])
	common_gene = intersect(genes_AFA,genes_geu)
	AFA_keep = AFA_chr[which(as.character(AFA_chr$GENE) %in% common_gene),]
	bim = paste0(path,"/chr",i,".bim")
	Geu_snp = read.table(bim)
	common_snp = intersect(as.character(Geu_snp$V2),as.character(AFA_keep$SNP))
	ind_geu = which(as.character(Geu_snp$V2) %in% common_snp)
	ind_AFA = which(as.character(AFA_keep$SNP) %in% common_snp)
	AFA_to_use = AFA_keep[ind_AFA,]
	geu_to_use = Geu_snp[ind_geu,]
	index = match( as.character(AFA_to_use$SNP),as.character(Geu_snp$V2))
	AFA_to_use$index_in_geu = index
	AFA_to_use$minor_geu = as.character(Geu_snp$V5)[index]
	AFA_to_use$major_geu = as.character(Geu_snp$V6)[index]
	save(AFA_to_use, file = paste0(path_pred,"/AFA/elnet_1Mb_AFA_to_use_chr_",i,".RData"))
	print(dim(AFA_to_use))
}


#---------------------------
# for CAU 1Mb
#---------------------------

path_MESA = "/net/mulan/home/shanglu/GENOA/data/MESA"
pop = c("AFA", "CAU", "HIS")

popnum = 2
load(paste0(path_MESA,"/",pop[popnum],"_weights_1Mb.RData"))


for(i in 1:22){
	print(i)  
  CAU_elnet_table = weights_1Mb[weights_1Mb$chr==i,]
  CAU_elnet_table$GENE = as.character(CAU_elnet_table$gene)
  CAU_elnet_table$SNP = as.character(CAU_elnet_table$rsid)
  CAU_elnet_table$allele1 = as.character(CAU_elnet_table$eff_allele)
  CAU_chr = CAU_elnet_table
	genes_CAU = unique(as.character(CAU_chr$GENE))
	genes_geu = unique(as.character(gene_exp_ping$ENSG)[which(gene_exp_ping$Chr==i)])
	common_gene = intersect(genes_CAU,genes_geu)
	CAU_keep = CAU_chr[which(as.character(CAU_chr$GENE) %in% common_gene),]
	bim = paste0(path,"/chr",i,".bim")
	Geu_snp = read.table(bim)
	common_snp = intersect(as.character(Geu_snp$V2),as.character(CAU_keep$SNP))
	ind_geu = which(as.character(Geu_snp$V2) %in% common_snp)
	ind_CAU = which(as.character(CAU_keep$SNP) %in% common_snp)
	CAU_to_use = CAU_keep[ind_CAU,]
	geu_to_use = Geu_snp[ind_geu,]
	index = match( as.character(CAU_to_use$SNP),as.character(Geu_snp$V2))
	CAU_to_use$index_in_geu = index
	CAU_to_use$minor_geu = as.character(Geu_snp$V5)[index]
	CAU_to_use$major_geu = as.character(Geu_snp$V6)[index]
	save(CAU_to_use, file = paste0(path_pred,"/CAU/elnet_1Mb_CAU_to_use_chr_",i,".RData"))
	print(dim(CAU_to_use))
}


#---------------------------
# for HIS 1Mb
#---------------------------

path_MESA = "/net/mulan/home/shanglu/GENOA/data/MESA"
pop = c("AFA", "CAU", "HIS")

popnum = 3
load(paste0(path_MESA,"/",pop[popnum],"_weights_1Mb.RData"))


for(i in 1:22){
	print(i)  
  HIS_elnet_table = weights_1Mb[weights_1Mb$chr==i,]
  HIS_elnet_table$GENE = as.character(HIS_elnet_table$gene)
  HIS_elnet_table$SNP = as.character(HIS_elnet_table$rsid)
  HIS_elnet_table$allele1 = as.character(HIS_elnet_table$eff_allele)
  HIS_chr = HIS_elnet_table
	genes_HIS = unique(as.character(HIS_chr$GENE))
	genes_geu = unique(as.character(gene_exp_ping$ENSG)[which(gene_exp_ping$Chr==i)])
	common_gene = intersect(genes_HIS,genes_geu)
	HIS_keep = HIS_chr[which(as.character(HIS_chr$GENE) %in% common_gene),]
	bim = paste0(path,"/chr",i,".bim")
	Geu_snp = read.table(bim)
	common_snp = intersect(as.character(Geu_snp$V2),as.character(HIS_keep$SNP))
	ind_geu = which(as.character(Geu_snp$V2) %in% common_snp)
	ind_HIS = which(as.character(HIS_keep$SNP) %in% common_snp)
	HIS_to_use = HIS_keep[ind_HIS,]
	geu_to_use = Geu_snp[ind_geu,]
	index = match( as.character(HIS_to_use$SNP),as.character(Geu_snp$V2))
	HIS_to_use$index_in_geu = index
	HIS_to_use$minor_geu = as.character(Geu_snp$V5)[index]
	HIS_to_use$major_geu = as.character(Geu_snp$V6)[index]
	save(HIS_to_use, file = paste0(path_pred,"/HIS/elnet_1Mb_HIS_to_use_chr_",i,".RData"))
	print(dim(HIS_to_use))
}


```

# prediction 

```

library(tidyr)

path = "/net/mulan/home/shanglu/GENOA/data/Geuvadis"
path_pred = "/net/mulan/home/shanglu/GENOA/analysis/prediction"

#-------------------------
# AA
#-------------------------

for(i in 1:22){
print(i)
load(paste0(path_pred,"/AA/elnet_AA_to_use_chr_",i,".RData"))
load(paste0(path,"/chr_",i,"_SnpMatrix.RData"))
fam = paste0(path,"/chr",i,".fam")
Geu_fam = read.table(fam)
	
print("Data loaded")

gene_name = unique(as.character(AA_to_use$GENE))
expr_mat1 = matrix(0,length(gene_name), 465)
rownames(expr_mat1) = gene_name
colnames(expr_mat1) = as.character(Geu_fam$V1)
expr_mat2 = expr_mat1

print("expr_mat created")

for(j in 1:length(gene_name)){
	 print(j)
	subset = AA_to_use[which(as.character(AA_to_use$GENE) %in% gene_name[j]),]  
  subset$weight = as.numeric(subset$weight)
  #!!!!!
	genotype = geno[,subset$index_in_geu]
	if(dim(subset)[1]==1){
	genotypeimpute = genotype
	genotypeimpute[is.na(genotypeimpute)] <- mean(genotypeimpute, na.rm = TRUE)
	}else{
	genotypeimpute =  replace_na(genotype,as.list(colMeans(genotype,na.rm=T)))
	}
	# direction of beta
	inconsistent = which(as.character(subset$allele1)!=as.character(subset$minor_geu)) #!!!!!
	subset[inconsistent,]$weight = -subset[inconsistent,]$weight                       #!!!!!

	genotypeimpute = scale(genotypeimpute)
	expr_pred2 = subset$weight %*% t(2-genotypeimpute)
	expr_mat2[j,]=expr_pred2
}

print("Start saving file")
save(expr_mat2, file = paste0(path_pred,"/AA/AA_elnet_expr_predict_chr_",i,".RData"))
}


summary(rowMeans(expr_mat2))

#-------------------------
# EA
#-------------------------

for(i in 1:22){
print(i)
load(paste0(path_pred,"/EA/elnet_EA_to_use_chr_",i,".RData"))
load(paste0(path,"/chr_",i,"_SnpMatrix.RData"))
fam = paste0(path,"/chr",i,".fam")
Geu_fam = read.table(fam)
	
print("Data loaded")

gene_name = unique(as.character(EA_to_use$GENE))
expr_mat1 = matrix(0,length(gene_name), 465)
rownames(expr_mat1) = gene_name
colnames(expr_mat1) = as.character(Geu_fam$V1)
expr_mat2 = expr_mat1

print("expr_mat created")

for(j in 1:length(gene_name)){
	 print(j)
	subset = EA_to_use[which(as.character(EA_to_use$GENE) %in% gene_name[j]),]  
  subset$weight = as.numeric(subset$weight)
  #!!!!!
	genotype = geno[,subset$index_in_geu]
	if(dim(subset)[1]==1){
	genotypeimpute = genotype
	genotypeimpute[is.na(genotypeimpute)] <- mean(genotypeimpute, na.rm = TRUE)
	}else{
	genotypeimpute =  replace_na(genotype,as.list(colMeans(genotype,na.rm=T)))
	}
	# direction of beta
	inconsistent = which(as.character(subset$allele1)!=as.character(subset$minor_geu)) #!!!!!
	subset[inconsistent,]$weight = -subset[inconsistent,]$weight                       #!!!!!

	genotypeimpute = scale(genotypeimpute)
	expr_pred2 = subset$weight %*% t(2-genotypeimpute)
	expr_mat2[j,]=expr_pred2
}

print("Start saving file")
save(expr_mat2, file = paste0(path_pred,"/EA/EA_elnet_expr_predict_chr_",i,".RData"))
}


summary(rowMeans(expr_mat2))


#-------------------------
# AFA - 100kb
#-------------------------

for(i in 1:22){
print(i)
load(paste0(path_pred,"/AFA/elnet_100kb_AFA_to_use_chr_",i,".RData"))
load(paste0(path,"/chr_",i,"_SnpMatrix.RData"))
fam = paste0(path,"/chr",i,".fam")
Geu_fam = read.table(fam)
	
print("Data loaded")
dat_to_use = AFA_to_use
dat_to_use$GENE = dat_to_use$gene
dat_to_use$SNP = dat_to_use$snps
print("Data loaded")


gene_name = unique(as.character(AFA_to_use$GENE))
expr_mat1 = matrix(0,length(gene_name), 465)
rownames(expr_mat1) = gene_name
colnames(expr_mat1) = as.character(Geu_fam$V1)
expr_mat2 = expr_mat1

print("expr_mat created")

for(j in 1:length(gene_name)){
	 print(j)
	subset = dat_to_use[which(as.character(dat_to_use$GENE) %in% gene_name[j]),]        #!!!!!
  subset$weight = as.numeric(subset$weight)
	genotype = geno[,subset$index_in_geu]
	if(dim(subset)[1]==1){
	genotypeimpute = genotype
	genotypeimpute[is.na(genotypeimpute)] <- mean(genotypeimpute, na.rm = TRUE)
	}else{
	genotypeimpute =  replace_na(genotype,as.list(colMeans(genotype,na.rm=T)))
	}
	# direction of beta
	inconsistent = which(as.character(subset$ref)!=as.character(subset$major_geu)) #!!!!!
	subset[inconsistent,]$weight = -subset[inconsistent,]$weight                       #!!!!!

	genotypeimpute = scale(genotypeimpute)
	expr_pred2 = subset$weight %*% t(2-genotypeimpute)
	expr_mat2[j,]=expr_pred2
}

print("Start saving file")
save(expr_mat2, file = paste0(path_pred,"/AFA/AFA_elnet_expr_predict_chr_",i,".RData"))
}


summary(rowMeans(expr_mat2))

#-------------------------
# CAU - 100kb
#-------------------------

for(i in 1:22){
print(i)
load(paste0(path_pred,"/CAU/elnet_100kb_CAU_to_use_chr_",i,".RData"))
load(paste0(path,"/chr_",i,"_SnpMatrix.RData"))
fam = paste0(path,"/chr",i,".fam")
Geu_fam = read.table(fam)
	
print("Data loaded")
dat_to_use = CAU_to_use
dat_to_use$GENE = dat_to_use$gene
dat_to_use$SNP = dat_to_use$snps
print("Data loaded")


gene_name = unique(as.character(CAU_to_use$GENE))
expr_mat1 = matrix(0,length(gene_name), 465)
rownames(expr_mat1) = gene_name
colnames(expr_mat1) = as.character(Geu_fam$V1)
expr_mat2 = expr_mat1

print("expr_mat created")

for(j in 1:length(gene_name)){
	 print(j)
	subset = dat_to_use[which(as.character(dat_to_use$GENE) %in% gene_name[j]),]        #!!!!!
  subset$weight = as.numeric(subset$weight)
	genotype = geno[,subset$index_in_geu]
	if(dim(subset)[1]==1){
	genotypeimpute = genotype
	genotypeimpute[is.na(genotypeimpute)] <- mean(genotypeimpute, na.rm = TRUE)
	}else{
	genotypeimpute =  replace_na(genotype,as.list(colMeans(genotype,na.rm=T)))
	}
	# direction of beta
	inconsistent = which(as.character(subset$ref)!=as.character(subset$major_geu)) #!!!!!
	subset[inconsistent,]$weight = -subset[inconsistent,]$weight                       #!!!!!

	genotypeimpute = scale(genotypeimpute)
	expr_pred2 = subset$weight %*% t(2-genotypeimpute)
	expr_mat2[j,]=expr_pred2
}

print("Start saving file")
save(expr_mat2, file = paste0(path_pred,"/CAU/CAU_elnet_expr_predict_chr_",i,".RData"))
}

#-------------------------
# HIS - 100kb
#-------------------------

for(i in 1:22){
print(i)
load(paste0(path_pred,"/HIS/elnet_100kb_HIS_to_use_chr_",i,".RData"))
load(paste0(path,"/chr_",i,"_SnpMatrix.RData"))
fam = paste0(path,"/chr",i,".fam")
Geu_fam = read.table(fam)
	
print("Data loaded")
dat_to_use = HIS_to_use
dat_to_use$GENE = dat_to_use$gene
dat_to_use$SNP = dat_to_use$snps
print("Data loaded")


gene_name = unique(as.character(HIS_to_use$GENE))
expr_mat1 = matrix(0,length(gene_name), 465)
rownames(expr_mat1) = gene_name
colnames(expr_mat1) = as.character(Geu_fam$V1)
expr_mat2 = expr_mat1

print("expr_mat created")

for(j in 1:length(gene_name)){
	 print(j)
	subset = dat_to_use[which(as.character(dat_to_use$GENE) %in% gene_name[j]),]        #!!!!!
  subset$weight = as.numeric(subset$weight)
	genotype = geno[,subset$index_in_geu]
	if(dim(subset)[1]==1){
	genotypeimpute = genotype
	genotypeimpute[is.na(genotypeimpute)] <- mean(genotypeimpute, na.rm = TRUE)
	}else{
	genotypeimpute =  replace_na(genotype,as.list(colMeans(genotype,na.rm=T)))
	}
	# direction of beta
	inconsistent = which(as.character(subset$ref)!=as.character(subset$major_geu)) #!!!!!
	subset[inconsistent,]$weight = -subset[inconsistent,]$weight                       #!!!!!

	genotypeimpute = scale(genotypeimpute)
	expr_pred2 = subset$weight %*% t(2-genotypeimpute)
	expr_mat2[j,]=expr_pred2
}

print("Start saving file")
save(expr_mat2, file = paste0(path_pred,"/HIS/HIS_elnet_expr_predict_chr_",i,".RData"))
}


#-------------------------
# AFA - 1Mb
#-------------------------
library(tidyr)

path = "/net/mulan/home/shanglu/GENOA/data/Geuvadis"
path_pred = "/net/mulan/home/shanglu/GENOA/analysis/prediction"


for(i in 1:22){
print(i)
load(paste0(path_pred,"/AFA/elnet_1Mb_AFA_to_use_chr_",i,".RData"))
load(paste0(path,"/chr_",i,"_SnpMatrix.RData"))
fam = paste0(path,"/chr",i,".fam")
Geu_fam = read.table(fam)
	
print("Data loaded")
dat_to_use = AFA_to_use
dat_to_use$GENE = dat_to_use$gene
dat_to_use$SNP = dat_to_use$snps
print("Data loaded")


gene_name = unique(as.character(AFA_to_use$GENE))
expr_mat1 = matrix(0,length(gene_name), 465)
rownames(expr_mat1) = gene_name
colnames(expr_mat1) = as.character(Geu_fam$V1)
expr_mat2 = expr_mat1

print("expr_mat created")

for(j in 1:length(gene_name)){
	 print(j)
	subset = dat_to_use[which(as.character(dat_to_use$GENE) %in% gene_name[j]),]        #!!!!!
  subset$weight = as.numeric(subset$weight)
	genotype = geno[,subset$index_in_geu]
	if(dim(subset)[1]==1){
	genotypeimpute = genotype
	genotypeimpute[is.na(genotypeimpute)] <- mean(genotypeimpute, na.rm = TRUE)
	}else{
	genotypeimpute =  replace_na(genotype,as.list(colMeans(genotype,na.rm=T)))
	}
	# direction of beta
	inconsistent = which(as.character(subset$ref)!=as.character(subset$major_geu)) #!!!!!
	subset[inconsistent,]$weight = -subset[inconsistent,]$weight                       #!!!!!

	genotypeimpute = scale(genotypeimpute)
	expr_pred2 = subset$weight %*% t(2-genotypeimpute)
	expr_mat2[j,]=expr_pred2
}

print("Start saving file")
save(expr_mat2, file = paste0(path_pred,"/AFA/AFA_elnet_1Mb_expr_predict_chr_",i,".RData"))
}

#-------------------------
# CAU - 1Mb
#-------------------------
for(i in 1:22){
print(i)
load(paste0(path_pred,"/CAU/elnet_1Mb_CAU_to_use_chr_",i,".RData"))
load(paste0(path,"/chr_",i,"_SnpMatrix.RData"))
fam = paste0(path,"/chr",i,".fam")
Geu_fam = read.table(fam)
	
print("Data loaded")
dat_to_use = CAU_to_use
dat_to_use$GENE = dat_to_use$gene
dat_to_use$SNP = dat_to_use$snps
print("Data loaded")


gene_name = unique(as.character(CAU_to_use$GENE))
expr_mat1 = matrix(0,length(gene_name), 465)
rownames(expr_mat1) = gene_name
colnames(expr_mat1) = as.character(Geu_fam$V1)
expr_mat2 = expr_mat1

print("expr_mat created")

for(j in 1:length(gene_name)){
	 print(j)
	subset = dat_to_use[which(as.character(dat_to_use$GENE) %in% gene_name[j]),]        #!!!!!
  subset$weight = as.numeric(subset$weight)
	genotype = geno[,subset$index_in_geu]
	if(dim(subset)[1]==1){
	genotypeimpute = genotype
	genotypeimpute[is.na(genotypeimpute)] <- mean(genotypeimpute, na.rm = TRUE)
	}else{
	genotypeimpute =  replace_na(genotype,as.list(colMeans(genotype,na.rm=T)))
	}
	# direction of beta
	inconsistent = which(as.character(subset$ref)!=as.character(subset$major_geu)) #!!!!!
	subset[inconsistent,]$weight = -subset[inconsistent,]$weight                       #!!!!!

	genotypeimpute = scale(genotypeimpute)
	expr_pred2 = subset$weight %*% t(2-genotypeimpute)
	expr_mat2[j,]=expr_pred2
}

print("Start saving file")
save(expr_mat2, file = paste0(path_pred,"/CAU/CAU_elnet_1Mb_expr_predict_chr_",i,".RData"))
}

#-------------------------
# HIS - 1Mb
#-------------------------

for(i in 1:22){
print(i)
load(paste0(path_pred,"/HIS/elnet_1Mb_HIS_to_use_chr_",i,".RData"))
load(paste0(path,"/chr_",i,"_SnpMatrix.RData"))
fam = paste0(path,"/chr",i,".fam")
Geu_fam = read.table(fam)
	
print("Data loaded")
dat_to_use = HIS_to_use
dat_to_use$GENE = dat_to_use$gene
dat_to_use$SNP = dat_to_use$snps
print("Data loaded")


gene_name = unique(as.character(HIS_to_use$GENE))
expr_mat1 = matrix(0,length(gene_name), 465)
rownames(expr_mat1) = gene_name
colnames(expr_mat1) = as.character(Geu_fam$V1)
expr_mat2 = expr_mat1

print("expr_mat created")

for(j in 1:length(gene_name)){
	 print(j)
	subset = dat_to_use[which(as.character(dat_to_use$GENE) %in% gene_name[j]),]        #!!!!!
  subset$weight = as.numeric(subset$weight)
	genotype = geno[,subset$index_in_geu]
	if(dim(subset)[1]==1){
	genotypeimpute = genotype
	genotypeimpute[is.na(genotypeimpute)] <- mean(genotypeimpute, na.rm = TRUE)
	}else{
	genotypeimpute =  replace_na(genotype,as.list(colMeans(genotype,na.rm=T)))
	}
	# direction of beta
	inconsistent = which(as.character(subset$ref)!=as.character(subset$major_geu)) #!!!!!
	subset[inconsistent,]$weight = -subset[inconsistent,]$weight                       #!!!!!

	genotypeimpute = scale(genotypeimpute)
	expr_pred2 = subset$weight %*% t(2-genotypeimpute)
	expr_mat2[j,]=expr_pred2
}

print("Start saving file")
save(expr_mat2, file = paste0(path_pred,"/HIS/HIS_elnet_1Mb_expr_predict_chr_",i,".RData"))
}







```

# collect result
```


library(tidyr)
path = "/net/mulan/home/shanglu/GENOA/data/Geuvadis"


load(paste0(path,"/pop.RData"))
load(paste0(path,"/expr_geu_462.RData"))
popu = c("GBR", "FIN", "CEU" ,"YRI" ,"TSI")
mesapop = c("AFA","CAU","HIS") 

for(popnum in 1:3){
popname = mesapop[popnum]
count = 0
genes = c()
for(i in 1:22){

	load(paste0(path_pred,"/",popname,"/",popname,"_elnet_1Mb_expr_predict_chr_",i,".RData"))
	nrow = length(rownames(expr_mat2))
	count = count+nrow
	genes = c(genes,rownames(expr_mat2))
}
count

pearson_p_v2 = matrix(0,count,5)
pearson_rho_v2 = matrix(0,count,5)

countR = 0
for(i in 1:22){
	print(i)

	load(paste0(path_pred,"/",popname,"/",popname,"_elnet_1Mb_expr_predict_chr_",i,".RData"))
	ind = match(rownames(expr_mat2),rownames(expr_geu_462))
	expr_mat_v2 = expr_mat2[,-c(2, 117 ,222)]
	expr_geu_obs = expr_geu_462[ind,]

	for(j in 1:dim(expr_mat2)[1]){
		countR = countR + 1
		for(k in 1:5){
		res_v2 = cor.test(expr_geu_obs[j,pop[[k]]],expr_mat_v2[j,pop[[k]]],method = "pearson")
		
		pearson_p_v2[countR,k] = res_v2$p.value
		pearson_rho_v2[countR,k] = res_v2$estimate
		}
	}
}


rownames(pearson_p_v2) = genes
rownames(pearson_rho_v2) = genes
colnames(pearson_p_v2) = popu
colnames(pearson_rho_v2) = popu

result = list()
result$pearson_rho_v2 = pearson_rho_v2
result$pearson_p_v2 = pearson_p_v2

save(result, file = paste0(path_pred,"/",popname,"/",popname,"_elnet_1Mb_result.RData"))
}

#-------------------------
# 100kb in MESA
#-------------------------


for(popnum in 1:3){
popname = mesapop[popnum]
count = 0
genes = c()
for(i in 1:22){

	load(paste0(path_pred,"/",popname,"/",popname,"_elnet_expr_predict_chr_",i,".RData"))
	nrow = length(rownames(expr_mat2))
	count = count+nrow
	genes = c(genes,rownames(expr_mat2))
}
count

pearson_p_v2 = matrix(0,count,5)
pearson_rho_v2 = matrix(0,count,5)

countR = 0
for(i in 1:22){
	print(i)

	load(paste0(path_pred,"/",popname,"/",popname,"_elnet_expr_predict_chr_",i,".RData"))
	ind = match(rownames(expr_mat2),rownames(expr_geu_462))
	expr_mat_v2 = expr_mat2[,-c(2, 117 ,222)]
	expr_geu_obs = expr_geu_462[ind,]

	for(j in 1:dim(expr_mat2)[1]){
		countR = countR + 1
		for(k in 1:5){
		res_v2 = cor.test(expr_geu_obs[j,pop[[k]]],expr_mat_v2[j,pop[[k]]],method = "pearson")
		
		pearson_p_v2[countR,k] = res_v2$p.value
		pearson_rho_v2[countR,k] = res_v2$estimate
		}
	}
}


rownames(pearson_p_v2) = genes
rownames(pearson_rho_v2) = genes
colnames(pearson_p_v2) = popu
colnames(pearson_rho_v2) = popu

result = list()
result$pearson_rho_v2 = pearson_rho_v2
result$pearson_p_v2 = pearson_p_v2

save(result, file = paste0(path_pred,"/",popname,"/",popname,"_elnet_100kb_result.RData"))
}

#-------------------------
# 100kb in GENOA
#-------------------------
genoapop = c("AA","EA")
for(popnum in 1:2){
popname = genoapop[popnum]
count = 0
genes = c()
for(i in 1:22){

	load(paste0(path_pred,"/",popname,"/",popname,"_elnet_expr_predict_chr_",i,".RData"))
	nrow = length(rownames(expr_mat2))
	count = count+nrow
	genes = c(genes,rownames(expr_mat2))
}
count

pearson_p_v2 = matrix(0,count,5)
pearson_rho_v2 = matrix(0,count,5)

countR = 0
for(i in 1:22){
	print(i)

	load(paste0(path_pred,"/",popname,"/",popname,"_elnet_expr_predict_chr_",i,".RData"))
	ind = match(rownames(expr_mat2),rownames(expr_geu_462))
	expr_mat_v2 = expr_mat2[,-c(2, 117 ,222)]
	expr_geu_obs = expr_geu_462[ind,]

	for(j in 1:dim(expr_mat2)[1]){
		countR = countR + 1
		for(k in 1:5){
		res_v2 = cor.test(expr_geu_obs[j,pop[[k]]],expr_mat_v2[j,pop[[k]]],method = "pearson")
		
		pearson_p_v2[countR,k] = res_v2$p.value
		pearson_rho_v2[countR,k] = res_v2$estimate
		}
	}
}


rownames(pearson_p_v2) = genes
rownames(pearson_rho_v2) = genes
colnames(pearson_p_v2) = popu
colnames(pearson_rho_v2) = popu

result = list()
result$pearson_rho_v2 = pearson_rho_v2
result$pearson_p_v2 = pearson_p_v2

save(result, file = paste0(path_pred,"/",popname,"/",popname,"_elnet_result.RData"))
}



```


Tables
```R

path_pred = "/net/mulan/home/shanglu/GENOA/analysis/prediction"
popname = c("AA", "EA", "AFA", "CAU", "HIS")
popnames = c("result/result_AA_bslmm_allpairs","result/result_EA_bslmm_allpairs","AA/AA_elnet_result","EA/EA_elnet_result","AFA/AFA_elnet_100kb_result","CAU/CAU_elnet_100kb_result","HIS/HIS_elnet_100kb_result" )


popu = c("GBR", "FIN", "CEU" ,"YRI" ,"TSI")

for(k in 1:5){
res_all = matrix(0,7, 7)
geudat = c()
ourpop = c()
gene = c()
count = 0
for(usepop in c("AAbslmm" , "EAbslmm", "AAelnet", "EAelnet", "AFAelnet","CAUelnet","HISelnet")){
count = count + 1
	load(paste0(path_pred,"/",popnames[count],".RData"))
	R2 = (result$pearson_rho_v2)^2
	geudat = c(geudat, R2[,k])
	ourpop = c(ourpop, rep(usepop,dim(R2)[1]))
	gene = c(gene, rownames(R2))
	dat = data.frame(geudat, ourpop,gene)
	dat$ourpop = factor(c(ourpop),levels = c("AAbslmm" , "EAbslmm", "AAelnet", "EAelnet", "AFAelnet","CAUelnet","HISelnet"),order=T)
	dat = na.omit(dat)
	dat$ourpop = as.character(dat$ourpop)
	pop = dat$geudat[dat$ourpop==usepop]

	res_all[count,1] = sum(pop>=0)
	res_all[count,2] = sum(pop>=0.01)
	res_all[count,3] = sum(pop>=0.02)
	res_all[count,4] = sum(pop>=0.05)
	res_all[count,5] = sum(pop>=0.1)
	res_all[count,6] = sum(pop>=0.2)
	res_all[count,7] = sum(pop>=0.5)
}
colnames(res_all) = c("R2_0","R2_001","R2_002","R2_005","R2_01","R2_02","R2_05")
rownames(res_all) = c("AAbslmm" , "EAbslmm", "AAelnet", "EAelnet", "AFAelnet","CAUelnet","HISelnet")
write.csv(res_all,paste0(path_pred,"/result/res_elnet_predict_",popu[k],".csv"),quote=F)
res_percent = round(res_all/res_all[,1],3)
write.csv(res_percent,paste0(path_pred,"/result/res_percent_elnet_predict_",popu[k],".csv"),quote=F)
}










```

# R2 compare scatter plot
```R

path_pred = "/net/mulan/home/shanglu/GENOA/analysis/prediction"
popname = c("AA", "EA", "AFA", "CAU", "HIS")
popnames = c("result/result_AA_bslmm_allpairs","result/result_EA_bslmm_allpairs","AA/AA_elnet_result","EA/EA_elnet_result","AFA/AFA_elnet_100kb_result","CAU/CAU_elnet_100kb_result","HIS/HIS_elnet_100kb_result" )


popu = c("GBR", "FIN", "CEU" ,"YRI" ,"TSI")

k = 0





# YRI: k=4
geudat = c()
ourpop = c()
gene = c()


count = 0
for(usepop in c("AAbslmm" , "EAbslmm", "AAelnet", "EAelnet", "AFAelnet","CAUelnet","HISelnet")){
count = count + 1
	load(paste0(path_pred,"/",popnames[count],".RData"))
	R2 = (result$pearson_rho_v2)^2
	geudat = c(geudat, R2[,k])
	ourpop = c(ourpop, rep(usepop,dim(R2)[1]))
	gene = c(gene, rownames(R2))
}
dat = data.frame(geudat, ourpop,gene)
dat$ourpop = factor(c(ourpop),levels = c("AAbslmm" , "EAbslmm", "AAelnet", "EAelnet", "AFAelnet","CAUelnet","HISelnet"),order=T)
dat = na.omit(dat)

dat$ourpop = as.character(dat$ourpop)
AAgene = as.character(dat$gene)[which(as.character(dat$ourpop) %in% "AAelnet")] 
AFAgene = as.character(dat$gene)[which(as.character(dat$ourpop) %in% "AFAelnet")] 
AA_AFA_gene = intersect(AAgene,AFAgene )

	pop = dat$geudat[dat$ourpop==usepop]
	pop1 = dat$geudat[dat$ourpop=="AAelnet"][match(AA_AFA_gene, AAgene)]
	pop2 = dat$geudat[dat$ourpop=="AFAelnet"][match(AA_AFA_gene, AFAgene)]

	datt = data.frame(pop1, pop2)

path="/net/mulan/home/shanglu/GENOA/analysis/figure"
pdf(paste0(path, "/",popu[k],"_scatter_R2.pdf"),width=6,height=6)
scatterPlot <- ggplot(datt,aes(pop2, pop1)) + 
  geom_point() + 
  geom_abline(intercept =0 , slope = 1,color="blue")+
  #geom_smooth(method="lm", se=TRUE, fullrange=TRUE)+
  scale_color_manual(values = c('#999999','#E69F00')) + 
  theme(legend.position=c(0,1), legend.justification=c(0,1))+theme_bw(base_size = 22)+
  coord_cartesian(ylim = c(0, 1),xlim = c(0, 1))+ geom_rug()+
  labs(  x="R2 in AFA",
         y="R2 in AA")
scatterPlot
xdensity <- ggplot(datt, aes(x=pop1)) + 
  geom_density(alpha=.5) + 
  theme(legend.position = "none")+theme_bw(base_size = 22)
xdensity
# Marginal density plot of y (right panel)
ydensity <- ggplot(datt, aes(x=pop2)) + 
  geom_density(alpha=.5) + 
  theme(legend.position = "none")+theme_bw(base_size = 22)
ydensity
dev.off() 

sum(pop1>pop2)

sum(pop1<pop2)

> sum(pop1>pop2)
[1] 2135
> 
> sum(pop1<pop2)
[1] 1645


# CEU
k=3
geudat = c()
ourpop = c()
gene = c()

count = 0

for(usepop in c("AAbslmm" , "EAbslmm", "AAelnet", "EAelnet", "AFAelnet","CAUelnet","HISelnet")){
count = count + 1
	load(paste0(path_pred,"/",popnames[count],".RData"))
	R2 = (result$pearson_rho_v2)^2
	geudat = c(geudat, R2[,k])
	ourpop = c(ourpop, rep(usepop,dim(R2)[1]))
	gene = c(gene, rownames(R2))
}
dat = data.frame(geudat, ourpop,gene)
dat$ourpop = factor(c(ourpop),levels = c("AAbslmm" , "EAbslmm", "AAelnet", "EAelnet", "AFAelnet","CAUelnet","HISelnet"),order=T)
dat = na.omit(dat)

dat$ourpop = as.character(dat$ourpop)
EAgene = as.character(dat$gene)[which(as.character(dat$ourpop) %in% "EAelnet")] 
CAUgene = as.character(dat$gene)[which(as.character(dat$ourpop) %in% "CAUelnet")] 
EA_CAU_gene = intersect(EAgene,CAUgene )

	pop = dat$geudat[dat$ourpop==usepop]
	pop1 = dat$geudat[dat$ourpop=="EAelnet"][match(EA_CAU_gene, EAgene)]
	pop2 = dat$geudat[dat$ourpop=="CAUelnet"][match(EA_CAU_gene, CAUgene)]

	datt = data.frame(pop1, pop2)

path="/net/mulan/home/shanglu/GENOA/analysis/figure"
pdf(paste0(path, "/",popu[k],"_scatter_R2.pdf"),width=6,height=6)
scatterPlot <- ggplot(datt,aes(pop2, pop1)) + 
  geom_point() + 
  geom_abline(intercept =0 , slope = 1,color="blue")+
  #geom_smooth(method="lm", se=TRUE, fullrange=TRUE)+
  scale_color_manual(values = c('#999999','#E69F00')) + 
  theme(legend.position=c(0,1), legend.justification=c(0,1))+theme_bw(base_size = 22)+
  coord_cartesian(ylim = c(0, 1),xlim = c(0, 1))+ geom_rug()+
  labs(  x="R2 in CAU",
         y="R2 in EA")
scatterPlot
xdensity <- ggplot(datt, aes(x=pop1)) + 
  geom_density(alpha=.5) + 
  theme(legend.position = "none")+theme_bw(base_size = 22)
xdensity
# Marginal density plot of y (right panel)
ydensity <- ggplot(datt, aes(x=pop2)) + 
  geom_density(alpha=.5) + 
  theme(legend.position = "none")+theme_bw(base_size = 22)
ydensity
dev.off() 


sum(pop1>pop2)
sum(pop1<pop2)

> sum(pop1>pop2)
[1] 2207
> sum(pop1<pop2)
[1] 2130


# GBR
k=1
geudat = c()
ourpop = c()
gene = c()

count = 0

for(usepop in c("AAbslmm" , "EAbslmm", "AAelnet", "EAelnet", "AFAelnet","CAUelnet","HISelnet")){
count = count + 1
	load(paste0(path_pred,"/",popnames[count],".RData"))
	R2 = (result$pearson_rho_v2)^2
	geudat = c(geudat, R2[,k])
	ourpop = c(ourpop, rep(usepop,dim(R2)[1]))
	gene = c(gene, rownames(R2))
}
dat = data.frame(geudat, ourpop,gene)
dat$ourpop = factor(c(ourpop),levels = c("AAbslmm" , "EAbslmm", "AAelnet", "EAelnet", "AFAelnet","CAUelnet","HISelnet"),order=T)
dat = na.omit(dat)

dat$ourpop = as.character(dat$ourpop)
EAgene = as.character(dat$gene)[which(as.character(dat$ourpop) %in% "EAelnet")] 
CAUgene = as.character(dat$gene)[which(as.character(dat$ourpop) %in% "CAUelnet")] 
EA_CAU_gene = intersect(EAgene,CAUgene )

	pop = dat$geudat[dat$ourpop==usepop]
	pop1 = dat$geudat[dat$ourpop=="EAelnet"][match(EA_CAU_gene, EAgene)]
	pop2 = dat$geudat[dat$ourpop=="CAUelnet"][match(EA_CAU_gene, CAUgene)]

	datt = data.frame(pop1, pop2)

path="/net/mulan/home/shanglu/GENOA/analysis/figure"
pdf(paste0(path, "/",popu[k],"_scatter_R2.pdf"),width=6,height=6)
scatterPlot <- ggplot(datt,aes(pop2, pop1)) + 
  geom_point() + 
  geom_abline(intercept =0 , slope = 1,color="blue")+
  #geom_smooth(method="lm", se=TRUE, fullrange=TRUE)+
  scale_color_manual(values = c('#999999','#E69F00')) + 
  theme(legend.position=c(0,1), legend.justification=c(0,1))+theme_bw(base_size = 22)+
  coord_cartesian(ylim = c(0, 1),xlim = c(0, 1))+ geom_rug()+
  labs(  x="R2 in CAU",
         y="R2 in EA")
scatterPlot
xdensity <- ggplot(datt, aes(x=pop1)) + 
  geom_density(alpha=.5) + 
  theme(legend.position = "none")+theme_bw(base_size = 22)
xdensity
# Marginal density plot of y (right panel)
ydensity <- ggplot(datt, aes(x=pop2)) + 
  geom_density(alpha=.5) + 
  theme(legend.position = "none")+theme_bw(base_size = 22)
ydensity
dev.off() 


sum(pop1>pop2)
sum(pop1<pop2)

> sum(pop1>pop2)
[1] 2228
> sum(pop1<pop2)
[1] 2112


 
```






