
```R


path_bslmm = "/net/mulan/home/shanglu/GENOA/analysis/prediction"
path_wtccc="/net/mulan/home/shanglu/GENOA/data/WTCCC"
path_twas = "/net/mulan/home/shanglu/GENOA/analysis/TWAS"

path_predixcan = "/net/mulan/home/shanglu/GENOA/analysis/prediction/predixcan"
path_elnet = "/net/mulan/home/shanglu/GENOA/analysis/prediction/elnet"
path_pred = "/net/mulan/home/shanglu/GENOA/analysis/prediction"
path = "/net/mulan/home/shanglu/GENOA/data/Geuvadis"

traits = c("T1D","T2D","RA","HT","CD","CAD","BD")


for(traitnum in 1:7){
print(traits[traitnum])
for(i in 1:22){
	print(i)
	# AA_chr = bslmm_all_result[which(bslmm_all_result$chr==i),]
	load(paste0(path_predixcan,"/AA_elnet_weight_chr_",i,".RData"))
	AA_elnet_table$GENE = as.character(AA_elnet_table$gene_id)
    AA_elnet_table$SNP = as.character(AA_elnet_table$rsid)
    AA_elnet_table$allele1 = as.character(AA_elnet_table$ref)
    AA_elnet_table$weight = as.character(AA_elnet_table$beta)
    AA_chr = AA_elnet_table

	genes_aa = unique(as.character(AA_chr$GENE))
	bim = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".bim")
	Trait_snp = read.table(bim)
	common_snp = intersect(as.character(Trait_snp$V2),as.character(AA_chr$SNP))
	ind_trait = which(as.character(Trait_snp$V2) %in% common_snp)
	ind_aa = which(as.character(AA_chr$SNP) %in% common_snp)
	AA_to_use = AA_chr[ind_aa,]
	Trait_to_use = Trait_snp[ind_trait,]
	index = match( as.character(AA_to_use$SNP),as.character(Trait_snp$V2))
	AA_to_use$index_in_trait = index
	AA_to_use$minor_geu = as.character(Trait_snp$V5)[index]
	AA_to_use$major_geu = as.character(Trait_snp$V6)[index]
	save(AA_to_use, file = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_AA_to_use_chr_",i,".RData"))
	print(dim(AA_to_use))
}
}

# EA: find common SNPs in each gene 

for(traitnum in 1:7){
print(traits[traitnum])
for(i in 1:22){
	print(i)
	# EA_chr = bslmm_all_result[which(bslmm_all_result$chr==i),]
	load(paste0(path_predixcan,"/EA_elnet_weight_chr_",i,".RData"))
	EA_elnet_table$GENE = as.character(EA_elnet_table$gene_id)
    EA_elnet_table$SNP = as.character(EA_elnet_table$rsid)
    EA_elnet_table$allele1 = as.character(EA_elnet_table$ref)
    EA_elnet_table$weight = as.character(EA_elnet_table$beta)
    EA_chr = EA_elnet_table

	genes_EA = unique(as.character(EA_chr$GENE))
	bim = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".bim")
	Trait_snp = read.table(bim)
	common_snp = intersect(as.character(Trait_snp$V2),as.character(EA_chr$SNP))
	ind_trait = which(as.character(Trait_snp$V2) %in% common_snp)
	ind_EA = which(as.character(EA_chr$SNP) %in% common_snp)
	EA_to_use = EA_chr[ind_EA,]
	Trait_to_use = Trait_snp[ind_trait,]
	index = match( as.character(EA_to_use$SNP),as.character(Trait_snp$V2))
	EA_to_use$index_in_trait = index
	EA_to_use$minor_geu = as.character(Trait_snp$V5)[index]
	EA_to_use$major_geu = as.character(Trait_snp$V6)[index]
	save(EA_to_use, file = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_EA_to_use_chr_",i,".RData"))
	print(dim(EA_to_use))
}
}


#--------------------
# MESA AFA 100kb
#--------------------

path_MESA = "/net/mulan/home/shanglu/GENOA/data/MESA"
pop = c("AFA", "CAU", "HIS")

popnum = 1
load(paste0(path_MESA,"/",pop[popnum],"_weights_100kb.RData"))

traits = c("T1D","T2D","RA","HT","CD","CAD","BD")

for(traitnum in 1:7){
print(traits[traitnum])
for(i in 1:22){
	print(i)
	AFA_elnet_table = weights_100kb[weights_100kb$chr==i,]
  AFA_elnet_table$GENE = as.character(AFA_elnet_table$gene)
  AFA_elnet_table$SNP = as.character(AFA_elnet_table$rsid)
  AFA_elnet_table$allele1 = as.character(AFA_elnet_table$eff_allele)
  dat_chr = AFA_elnet_table


	genes_dat = unique(as.character(dat_chr$GENE))
	bim = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".bim")
	Trait_snp = read.table(bim)
	common_snp = intersect(as.character(Trait_snp$V2),as.character(dat_chr$SNP))
	ind_trait = which(as.character(Trait_snp$V2) %in% common_snp)
	ind_aa = which(as.character(dat_chr$SNP) %in% common_snp)
	dat_to_use = dat_chr[ind_aa,]
	Trait_to_use = Trait_snp[ind_trait,]
	index = match( as.character(dat_to_use$SNP),as.character(Trait_snp$V2))
	dat_to_use$index_in_trait = index
	dat_to_use$minor_geu = as.character(Trait_snp$V5)[index]
	dat_to_use$major_geu = as.character(Trait_snp$V6)[index]
	save(dat_to_use, file = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_AFA_100kb_to_use_chr_",i,".RData"))
	print(dim(dat_to_use))
  
}
}



#--------------------
# MESA CAU 100kb
#--------------------

path_MESA = "/net/mulan/home/shanglu/GENOA/data/MESA"
pop = c("AFA", "CAU", "HIS")
popnum = 2
load(paste0(path_MESA,"/",pop[popnum],"_weights_100kb.RData"))

traits = c("T1D","T2D","RA","HT","CD","CAD","BD")



for(traitnum in 1:7){
print(traits[traitnum])
for(i in 1:22){
	print(i)
	CAU_elnet_table = weights_100kb[weights_100kb$chr==i,]
    CAU_elnet_table$GENE = as.character(CAU_elnet_table$gene)
    CAU_elnet_table$SNP = as.character(CAU_elnet_table$rsid)
    CAU_elnet_table$allele1 = as.character(CAU_elnet_table$eff_allele)
    dat_chr = CAU_elnet_table

	genes_dat = unique(as.character(dat_chr$GENE))
	bim = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".bim")
	Trait_snp = read.table(bim)
	common_snp = intersect(as.character(Trait_snp$V2),as.character(dat_chr$SNP))
	ind_trait = which(as.character(Trait_snp$V2) %in% common_snp)
	ind_aa = which(as.character(dat_chr$SNP) %in% common_snp)
	dat_to_use = dat_chr[ind_aa,]
	Trait_to_use = Trait_snp[ind_trait,]
	index = match( as.character(dat_to_use$SNP),as.character(Trait_snp$V2))
	dat_to_use$index_in_trait = index
	dat_to_use$minor_geu = as.character(Trait_snp$V5)[index]
	dat_to_use$major_geu = as.character(Trait_snp$V6)[index]
	save(dat_to_use, file = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_CAU_100kb_to_use_chr_",i,".RData"))
	print(dim(dat_to_use))
  
}
}


#--------------------
# MESA HIS 100kb
#--------------------

path_MESA = "/net/mulan/home/shanglu/GENOA/data/MESA"
pop = c("AFA", "CAU", "HIS")

popnum = 3
load(paste0(path_MESA,"/",pop[popnum],"_weights_100kb.RData"))

traits = c("T1D","T2D","RA","HT","CD","CAD","BD")



for(traitnum in 1:7){
print(traits[traitnum])
for(i in 1:22){
	print(i)
	HIS_elnet_table = weights_100kb[weights_100kb$chr==i,]
  HIS_elnet_table$GENE = as.character(HIS_elnet_table$gene)
  HIS_elnet_table$SNP = as.character(HIS_elnet_table$rsid)
  HIS_elnet_table$allele1 = as.character(HIS_elnet_table$eff_allele)
  dat_chr = HIS_elnet_table


	genes_dat = unique(as.character(dat_chr$GENE))
	bim = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".bim")
	Trait_snp = read.table(bim)
	common_snp = intersect(as.character(Trait_snp$V2),as.character(dat_chr$SNP))
	ind_trait = which(as.character(Trait_snp$V2) %in% common_snp)
	ind_aa = which(as.character(dat_chr$SNP) %in% common_snp)
	dat_to_use = dat_chr[ind_aa,]
	Trait_to_use = Trait_snp[ind_trait,]
	index = match( as.character(dat_to_use$SNP),as.character(Trait_snp$V2))
	dat_to_use$index_in_trait = index
	dat_to_use$minor_geu = as.character(Trait_snp$V5)[index]
	dat_to_use$major_geu = as.character(Trait_snp$V6)[index]
	save(dat_to_use, file = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_HIS_100kb_to_use_chr_",i,".RData"))
	print(dim(dat_to_use))
  
}
}


#--------------------
# MESA AFA 1Mb
#--------------------

path_MESA = "/net/mulan/home/shanglu/GENOA/data/MESA"
pop = c("AFA", "CAU", "HIS")

popnum = 1
load(paste0(path_MESA,"/",pop[popnum],"_weights_1Mb.RData"))

traits = c("T1D","T2D","RA","HT","CD","CAD","BD")

for(traitnum in 1:7){
print(traits[traitnum])
for(i in 1:22){
	print(i)
	AFA_elnet_table = weights_1Mb[weights_1Mb$chr==i,]
  AFA_elnet_table$GENE = as.character(AFA_elnet_table$gene)
  AFA_elnet_table$SNP = as.character(AFA_elnet_table$rsid)
  AFA_elnet_table$allele1 = as.character(AFA_elnet_table$eff_allele)
  dat_chr = AFA_elnet_table


	genes_dat = unique(as.character(dat_chr$GENE))
	bim = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".bim")
	Trait_snp = read.table(bim)
	common_snp = intersect(as.character(Trait_snp$V2),as.character(dat_chr$SNP))
	ind_trait = which(as.character(Trait_snp$V2) %in% common_snp)
	ind_aa = which(as.character(dat_chr$SNP) %in% common_snp)
	dat_to_use = dat_chr[ind_aa,]
	Trait_to_use = Trait_snp[ind_trait,]
	index = match( as.character(dat_to_use$SNP),as.character(Trait_snp$V2))
	dat_to_use$index_in_trait = index
	dat_to_use$minor_geu = as.character(Trait_snp$V5)[index]
	dat_to_use$major_geu = as.character(Trait_snp$V6)[index]
	save(dat_to_use, file = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_AFA_1Mb_to_use_chr_",i,".RData"))
	print(dim(dat_to_use))
  
}
}



#--------------------
# MESA CAU 1Mb
#--------------------

path_MESA = "/net/mulan/home/shanglu/GENOA/data/MESA"
pop = c("AFA", "CAU", "HIS")
popnum = 2
load(paste0(path_MESA,"/",pop[popnum],"_weights_1Mb.RData"))

traits = c("T1D","T2D","RA","HT","CD","CAD","BD")



for(traitnum in 1:7){
print(traits[traitnum])
for(i in 1:22){
	print(i)
	CAU_elnet_table = weights_1Mb[weights_1Mb$chr==i,]
    CAU_elnet_table$GENE = as.character(CAU_elnet_table$gene)
    CAU_elnet_table$SNP = as.character(CAU_elnet_table$rsid)
    CAU_elnet_table$allele1 = as.character(CAU_elnet_table$eff_allele)
    dat_chr = CAU_elnet_table

	genes_dat = unique(as.character(dat_chr$GENE))
	bim = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".bim")
	Trait_snp = read.table(bim)
	common_snp = intersect(as.character(Trait_snp$V2),as.character(dat_chr$SNP))
	ind_trait = which(as.character(Trait_snp$V2) %in% common_snp)
	ind_aa = which(as.character(dat_chr$SNP) %in% common_snp)
	dat_to_use = dat_chr[ind_aa,]
	Trait_to_use = Trait_snp[ind_trait,]
	index = match( as.character(dat_to_use$SNP),as.character(Trait_snp$V2))
	dat_to_use$index_in_trait = index
	dat_to_use$minor_geu = as.character(Trait_snp$V5)[index]
	dat_to_use$major_geu = as.character(Trait_snp$V6)[index]
	save(dat_to_use, file = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_CAU_1Mb_to_use_chr_",i,".RData"))
	print(dim(dat_to_use))
  
}
}


#--------------------
# MESA HIS 1Mb
#--------------------

path_MESA = "/net/mulan/home/shanglu/GENOA/data/MESA"
pop = c("AFA", "CAU", "HIS")

popnum = 3
load(paste0(path_MESA,"/",pop[popnum],"_weights_1Mb.RData"))

traits = c("T1D","T2D","RA","HT","CD","CAD","BD")



for(traitnum in 1:7){
print(traits[traitnum])
for(i in 1:22){
	print(i)
	HIS_elnet_table = weights_1Mb[weights_1Mb$chr==i,]
  HIS_elnet_table$GENE = as.character(HIS_elnet_table$gene)
  HIS_elnet_table$SNP = as.character(HIS_elnet_table$rsid)
  HIS_elnet_table$allele1 = as.character(HIS_elnet_table$eff_allele)
  dat_chr = HIS_elnet_table


	genes_dat = unique(as.character(dat_chr$GENE))
	bim = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".bim")
	Trait_snp = read.table(bim)
	common_snp = intersect(as.character(Trait_snp$V2),as.character(dat_chr$SNP))
	ind_trait = which(as.character(Trait_snp$V2) %in% common_snp)
	ind_aa = which(as.character(dat_chr$SNP) %in% common_snp)
	dat_to_use = dat_chr[ind_aa,]
	Trait_to_use = Trait_snp[ind_trait,]
	index = match( as.character(dat_to_use$SNP),as.character(Trait_snp$V2))
	dat_to_use$index_in_trait = index
	dat_to_use$minor_geu = as.character(Trait_snp$V5)[index]
	dat_to_use$major_geu = as.character(Trait_snp$V6)[index]
	save(dat_to_use, file = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_HIS_1Mb_to_use_chr_",i,".RData"))
	print(dim(dat_to_use))
  
}
}


#-------------------------------------------------
# Prediction
#-------------------------------------------------




#---------
# AA
#---------

args <- as.numeric(commandArgs(TRUE))
traitnum=args[1]
i=args[2]


path_bslmm = "/net/mulan/home/shanglu/GENOA/analysis/prediction"
path_wtccc="/net/mulan/home/shanglu/GENOA/data/WTCCC"
path_twas = "/net/mulan/home/shanglu/GENOA/analysis/TWAS"

path_predixcan = "/net/mulan/home/shanglu/GENOA/analysis/prediction/predixcan"
path_elnet = "/net/mulan/home/shanglu/GENOA/analysis/prediction/elnet"
path_pred = "/net/mulan/home/shanglu/GENOA/analysis/prediction"
path = "/net/mulan/home/shanglu/GENOA/data/Geuvadis"

traits = c("T1D","T2D","RA","HT","CD","CAD","BD")

library(tidyr)

popugenoa=1


print(traits[traitnum])
print(i)

load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_AA_to_use_chr_",i,".RData"))
load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr_",i,"_SnpMatrix.RData"))
fam = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".fam")
Geu_fam = read.table(fam)
	
print("Data loaded")
dat_to_use = AA_to_use
gene_name = unique(as.character(dat_to_use$GENE))			#!!!!
expr_mat1 = matrix(0,length(gene_name), dim(Geu_fam)[1])
rownames(expr_mat1) = gene_name
colnames(expr_mat1) = as.character(Geu_fam$V1)
expr_mat2 = expr_mat1

print("expr_mat created")

for(j in 1:length(gene_name)){
	print(j)
	subset = dat_to_use[which(as.character(dat_to_use$GENE) %in% gene_name[j]),]
	subset$weight = as.numeric(subset$weight)		 #!!!!
	genotype = genomat[,subset$index_in_trait]
	if(dim(subset)[1]==1){
	genotypeimpute = genotype
	genotypeimpute[is.na(genotypeimpute)] <- mean(genotypeimpute, na.rm = TRUE)
	}else{
	genotypeimpute =  replace_na(genotype,as.list(colMeans(genotype,na.rm=T)))
	}
	# direction of beta
	inconsistent = which(as.character(subset$allele1)!=as.character(subset$minor_geu))	 #!!!!
	subset[inconsistent,]$weight = -subset[inconsistent,]$weight 				 #!!!!

	genotypeimpute = scale(genotypeimpute)
	expr_pred2 = subset$weight %*% t(2-genotypeimpute)
	expr_mat2[j,]=expr_pred2
}

print("Start saving file")
save(expr_mat2, file = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_AA_elnet_v2.RData")) #!!!!


###############################
#!/bin/bash
#SBATCH --job-name=AAelnet
#SBATCH --array=1-22
#SBATCH --cpus-per-task=1
#SBATCH --output=/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping/out/AAelnet%a.out
#SBATCH --error=/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping/err/AAelnet%a.err
#SBATCH --workdir=/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping/code
#SBATCH --partition=mulan

bash

let k=0
for ((traitnum=1;traitnum<=7;traitnum++)); do
for ((chr=1;chr<=22;chr++)); do
  let k=${k}+1
  if [ ${k} -eq ${SLURM_ARRAY_TASK_ID} ]; then
  Rscript --verbose ./WTCCC_elnet_AA.r ${traitnum} ${chr}
  fi
done
done





#---------
# EA
#---------

args <- as.numeric(commandArgs(TRUE))
traitnum=args[1]
i=args[2]


path_bslmm = "/net/mulan/home/shanglu/GENOA/analysis/prediction"
path_wtccc="/net/mulan/home/shanglu/GENOA/data/WTCCC"
path_twas = "/net/mulan/home/shanglu/GENOA/analysis/TWAS"

path_predixcan = "/net/mulan/home/shanglu/GENOA/analysis/prediction/predixcan"
path_elnet = "/net/mulan/home/shanglu/GENOA/analysis/prediction/elnet"
path_pred = "/net/mulan/home/shanglu/GENOA/analysis/prediction"
path = "/net/mulan/home/shanglu/GENOA/data/Geuvadis"

traits = c("T1D","T2D","RA","HT","CD","CAD","BD")

library(tidyr)

popugenoa=2


print(traits[traitnum])
print(i)

load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_EA_to_use_chr_",i,".RData"))
load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr_",i,"_SnpMatrix.RData"))
fam = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".fam")
Geu_fam = read.table(fam)
	
print("Data loaded")
dat_to_use = EA_to_use
gene_name = unique(as.character(dat_to_use$GENE))			#!!!!
expr_mat1 = matrix(0,length(gene_name), dim(Geu_fam)[1])
rownames(expr_mat1) = gene_name
colnames(expr_mat1) = as.character(Geu_fam$V1)
expr_mat2 = expr_mat1

print("expr_mat created")

for(j in 1:length(gene_name)){
	print(j)
	subset = dat_to_use[which(as.character(dat_to_use$GENE) %in% gene_name[j]),]
	subset$weight = as.numeric(subset$weight)		 #!!!!
	genotype = genomat[,subset$index_in_trait]
	if(dim(subset)[1]==1){
	genotypeimpute = genotype
	genotypeimpute[is.na(genotypeimpute)] <- mean(genotypeimpute, na.rm = TRUE)
	}else{
	genotypeimpute =  replace_na(genotype,as.list(colMeans(genotype,na.rm=T)))
	}
	# direction of beta
	inconsistent = which(as.character(subset$allele1)!=as.character(subset$minor_geu))	 #!!!!
	subset[inconsistent,]$weight = -subset[inconsistent,]$weight 				 #!!!!

	genotypeimpute = scale(genotypeimpute)
	expr_pred2 = subset$weight %*% t(2-genotypeimpute)
	expr_mat2[j,]=expr_pred2
}

print("Start saving file")
save(expr_mat2, file = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_EA_elnet_v2.RData")) #!!!!

########################

#!/bin/bash
#SBATCH --job-name=EAelnet
#SBATCH --array=1-22
#SBATCH --cpus-per-task=1
#SBATCH --output=/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping/out/EAelnet%a.out
#SBATCH --error=/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping/err/EAelnet%a.err
#SBATCH --workdir=/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping/code
#SBATCH --partition=mulan

bash

let k=0
for ((traitnum=1;traitnum<=7;traitnum++)); do
for ((chr=1;chr<=22;chr++)); do
  let k=${k}+1
  if [ ${k} -eq ${SLURM_ARRAY_TASK_ID} ]; then
  Rscript --verbose ./WTCCC_elnet_EA.r ${traitnum} ${chr}
  fi
done
done





#---------
# AFA 100kb
#---------

library(tidyr)

for(traitnum in 1:7){
print(traits[traitnum])
for(i in 1:22){
print(i)
load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_AFA_100kb_to_use_chr_",i,".RData"))

load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr_",i,"_SnpMatrix.RData"))
fam = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".fam")
Geu_fam = read.table(fam)
	
print("Data loaded")
gene_name = unique(as.character(dat_to_use$GENE))			#!!!!
expr_mat1 = matrix(0,length(gene_name), dim(Geu_fam)[1])
rownames(expr_mat1) = gene_name
colnames(expr_mat1) = as.character(Geu_fam$V1)
expr_mat2 = expr_mat1

print("expr_mat created")

for(j in 1:length(gene_name)){
	# print(j)
    subset = dat_to_use[which(as.character(dat_to_use$GENE) %in% gene_name[j]),]		 #!!!!
	genotype = genomat[,subset$index_in_trait]
	if(dim(subset)[1]==1){
	genotypeimpute = genotype
	genotypeimpute[is.na(genotypeimpute)] <- mean(genotypeimpute, na.rm = TRUE)
	}else{
	genotypeimpute =  replace_na(genotype,as.list(colMeans(genotype,na.rm=T)))
	}
	# direction of beta
	inconsistent = which(as.character(subset$ref)!=as.character(subset$major_geu)) #!!!!!
	subset[inconsistent,]$weight = -subset[inconsistent,]$weight                       #!!!!!

	genotypeimpute = scale(genotypeimpute)
	expr_pred2 = subset$weight %*% t(2-genotypeimpute)
	expr_mat2[j,]=expr_pred2
}

print("Start saving file")
save(expr_mat2, file = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_elnet_AFA_100kb_v2.RData")) #!!!!
}
}


#---------
# CAU 100kb
#---------

library(tidyr)

for(traitnum in 1:7){
print(traits[traitnum])
for(i in 1:22){
print(i)
load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_CAU_100kb_to_use_chr_",i,".RData"))

load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr_",i,"_SnpMatrix.RData"))
fam = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".fam")
Geu_fam = read.table(fam)
	
print("Data loaded")
gene_name = unique(as.character(dat_to_use$GENE))			#!!!!
expr_mat1 = matrix(0,length(gene_name), dim(Geu_fam)[1])
rownames(expr_mat1) = gene_name
colnames(expr_mat1) = as.character(Geu_fam$V1)
expr_mat2 = expr_mat1

print("expr_mat created")

for(j in 1:length(gene_name)){
	# print(j)
    subset = dat_to_use[which(as.character(dat_to_use$GENE) %in% gene_name[j]),]		 #!!!!
	genotype = genomat[,subset$index_in_trait]
	if(dim(subset)[1]==1){
	genotypeimpute = genotype
	genotypeimpute[is.na(genotypeimpute)] <- mean(genotypeimpute, na.rm = TRUE)
	}else{
	genotypeimpute =  replace_na(genotype,as.list(colMeans(genotype,na.rm=T)))
	}
	# direction of beta
	inconsistent = which(as.character(subset$ref)!=as.character(subset$major_geu)) #!!!!!
	subset[inconsistent,]$weight = -subset[inconsistent,]$weight                       #!!!!!

	genotypeimpute = scale(genotypeimpute)
	expr_pred2 = subset$weight %*% t(2-genotypeimpute)
	expr_mat2[j,]=expr_pred2
}

print("Start saving file")
save(expr_mat2, file = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_elnet_CAU_100kb_v2.RData")) #!!!!
}
}

#---------
# HIS 100kb
#---------

library(tidyr)

for(traitnum in 1:7){
print(traits[traitnum])
for(i in 1:22){
print(i)
load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_HIS_100kb_to_use_chr_",i,".RData"))

load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr_",i,"_SnpMatrix.RData"))
fam = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".fam")
Geu_fam = read.table(fam)
	
print("Data loaded")
gene_name = unique(as.character(dat_to_use$GENE))			#!!!!
expr_mat1 = matrix(0,length(gene_name), dim(Geu_fam)[1])
rownames(expr_mat1) = gene_name
colnames(expr_mat1) = as.character(Geu_fam$V1)
expr_mat2 = expr_mat1

print("expr_mat created")

for(j in 1:length(gene_name)){
	# print(j)
    subset = dat_to_use[which(as.character(dat_to_use$GENE) %in% gene_name[j]),]		 #!!!!
	genotype = genomat[,subset$index_in_trait]
	if(dim(subset)[1]==1){
	genotypeimpute = genotype
	genotypeimpute[is.na(genotypeimpute)] <- mean(genotypeimpute, na.rm = TRUE)
	}else{
	genotypeimpute =  replace_na(genotype,as.list(colMeans(genotype,na.rm=T)))
	}
	# direction of beta
	inconsistent = which(as.character(subset$ref)!=as.character(subset$major_geu)) #!!!!!
	subset[inconsistent,]$weight = -subset[inconsistent,]$weight                       #!!!!!

	genotypeimpute = scale(genotypeimpute)
	expr_pred2 = subset$weight %*% t(2-genotypeimpute)
	expr_mat2[j,]=expr_pred2
}

print("Start saving file")
save(expr_mat2, file = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_elnet_HIS_100kb_v2.RData")) #!!!!
}
}


#---------
# AFA 1Mb
#---------

library(tidyr)

for(traitnum in 1:7){
print(traits[traitnum])
for(i in 1:22){
print(i)
load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_AFA_1Mb_to_use_chr_",i,".RData"))

load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr_",i,"_SnpMatrix.RData"))
fam = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".fam")
Geu_fam = read.table(fam)
	
print("Data loaded")
gene_name = unique(as.character(dat_to_use$GENE))			#!!!!
expr_mat1 = matrix(0,length(gene_name), dim(Geu_fam)[1])
rownames(expr_mat1) = gene_name
colnames(expr_mat1) = as.character(Geu_fam$V1)
expr_mat2 = expr_mat1

print("expr_mat created")

for(j in 1:length(gene_name)){
	# print(j)
    subset = dat_to_use[which(as.character(dat_to_use$GENE) %in% gene_name[j]),]		 #!!!!
	genotype = genomat[,subset$index_in_trait]
	if(dim(subset)[1]==1){
	genotypeimpute = genotype
	genotypeimpute[is.na(genotypeimpute)] <- mean(genotypeimpute, na.rm = TRUE)
	}else{
	genotypeimpute =  replace_na(genotype,as.list(colMeans(genotype,na.rm=T)))
	}
	# direction of beta
	inconsistent = which(as.character(subset$ref)!=as.character(subset$major_geu)) #!!!!!
	subset[inconsistent,]$weight = -subset[inconsistent,]$weight                       #!!!!!

	genotypeimpute = scale(genotypeimpute)
	expr_pred2 = subset$weight %*% t(2-genotypeimpute)
	expr_mat2[j,]=expr_pred2
}

print("Start saving file")
save(expr_mat2, file = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_elnet_AFA_1Mb_v2.RData")) #!!!!
}
}


#---------
# CAU 1Mb
#---------

library(tidyr)

for(traitnum in 1:7){
print(traits[traitnum])
for(i in 1:22){
print(i)
load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_CAU_1Mb_to_use_chr_",i,".RData"))

load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr_",i,"_SnpMatrix.RData"))
fam = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".fam")
Geu_fam = read.table(fam)
	
print("Data loaded")
gene_name = unique(as.character(dat_to_use$GENE))			#!!!!
expr_mat1 = matrix(0,length(gene_name), dim(Geu_fam)[1])
rownames(expr_mat1) = gene_name
colnames(expr_mat1) = as.character(Geu_fam$V1)
expr_mat2 = expr_mat1

print("expr_mat created")

for(j in 1:length(gene_name)){
	# print(j)
    subset = dat_to_use[which(as.character(dat_to_use$GENE) %in% gene_name[j]),]		 #!!!!
	genotype = genomat[,subset$index_in_trait]
	if(dim(subset)[1]==1){
	genotypeimpute = genotype
	genotypeimpute[is.na(genotypeimpute)] <- mean(genotypeimpute, na.rm = TRUE)
	}else{
	genotypeimpute =  replace_na(genotype,as.list(colMeans(genotype,na.rm=T)))
	}
	# direction of beta
	inconsistent = which(as.character(subset$ref)!=as.character(subset$major_geu)) #!!!!!
	subset[inconsistent,]$weight = -subset[inconsistent,]$weight                       #!!!!!

	genotypeimpute = scale(genotypeimpute)
	expr_pred2 = subset$weight %*% t(2-genotypeimpute)
	expr_mat2[j,]=expr_pred2
}

print("Start saving file")
save(expr_mat2, file = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_elnet_CAU_1Mb_v2.RData")) #!!!!
}
}

#---------
# HIS 1Mb
#---------

library(tidyr)

for(traitnum in 1:7){
print(traits[traitnum])
for(i in 1:22){
print(i)
load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_HIS_1Mb_to_use_chr_",i,".RData"))

load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr_",i,"_SnpMatrix.RData"))
fam = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".fam")
Geu_fam = read.table(fam)
	
print("Data loaded")
gene_name = unique(as.character(dat_to_use$GENE))			#!!!!
expr_mat1 = matrix(0,length(gene_name), dim(Geu_fam)[1])
rownames(expr_mat1) = gene_name
colnames(expr_mat1) = as.character(Geu_fam$V1)
expr_mat2 = expr_mat1

print("expr_mat created")

for(j in 1:length(gene_name)){
	# print(j)
    subset = dat_to_use[which(as.character(dat_to_use$GENE) %in% gene_name[j]),]		 #!!!!
	genotype = genomat[,subset$index_in_trait]
	if(dim(subset)[1]==1){
	genotypeimpute = genotype
	genotypeimpute[is.na(genotypeimpute)] <- mean(genotypeimpute, na.rm = TRUE)
	}else{
	genotypeimpute =  replace_na(genotype,as.list(colMeans(genotype,na.rm=T)))
	}
	# direction of beta
	inconsistent = which(as.character(subset$ref)!=as.character(subset$major_geu)) #!!!!!
	subset[inconsistent,]$weight = -subset[inconsistent,]$weight                       #!!!!!

	genotypeimpute = scale(genotypeimpute)
	expr_pred2 = subset$weight %*% t(2-genotypeimpute)
	expr_mat2[j,]=expr_pred2
}

print("Start saving file")
save(expr_mat2, file = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_elnet_HIS_1Mb_v2.RData")) #!!!!
}
}


```


Collect results
```
# WTCCC_elnet_GENOA.r
args <- as.numeric(commandArgs(TRUE))
traitnum=args[1]
i=args[2]
PCnum = args[3]
popnum = args[4]


path_pred = "/net/mulan/home/shanglu/GENOA/analysis/prediction"
path_wtccc="/net/mulan/home/shanglu/GENOA/data/WTCCC"
path = "/net/mulan/home/shanglu/GENOA/data/Geuvadis"

library(tidyr)

traits = c("T1D","T2D","RA","HT","CD","CAD","BD")
popugenoanames = c("AA" , "EA")
print(paste0("Trait: ",traits[traitnum]))
print(paste0("Chr: ",i))
print(paste0("PCnum: ",PCnum))
print(paste0("pop: ",popugenoanames[popnum]))

load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_",popugenoanames[popnum],"_elnet_v2.RData")) 

Gglm_PCs_p = matrix(0,dim(expr_mat2)[1],7)
Gglm_noPCs_p = matrix(0,dim(expr_mat2)[1],7)
Glm_PCs_p = matrix(0,dim(expr_mat2)[1],7)
Glm_noPCs_p = matrix(0,dim(expr_mat2)[1],7)

	fam = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".fam")
	WTCCCfam = read.table(fam)
	outcome = WTCCCfam$V6-1
	eigenvec = read.table(paste0(path_wtccc,"/wtccc",traits[traitnum],".PCA.eigenvec"))
	PCs = as.matrix(eigenvec[,3:(PCnum+2)])
	
	for(j in 1:dim(expr_mat2)[1]){
		Gglm_PCs = glm(formula=outcome ~ expr_mat2[j,] + PCs,  family=binomial)
		Glm_PCs = lm(formula=outcome ~ expr_mat2[j,] + PCs)
		Gglm_noPCs = glm(formula=outcome ~ expr_mat2[j,] ,  family=binomial)
		Glm_noPCs = lm(formula=outcome ~ expr_mat2[j,] )

		Gglm_PCs_p[j,traitnum] = summary(Gglm_PCs)$coefficients[2,4]
		Glm_PCs_p[j,traitnum] = summary(Glm_PCs)$coefficients[2,4]
		Gglm_noPCs_p[j,traitnum] = summary(Gglm_noPCs)$coefficients[2,4]
		Glm_noPCs_p[j,traitnum] = summary(Glm_noPCs)$coefficients[2,4]

	}

rownames(Gglm_PCs_p) = rownames(expr_mat2)
rownames(Glm_PCs_p) = rownames(expr_mat2)
rownames(Gglm_noPCs_p) = rownames(expr_mat2)
rownames(Glm_noPCs_p) = rownames(expr_mat2)

colnames(Gglm_PCs_p) = traits
colnames(Glm_PCs_p) = traits
colnames(Gglm_noPCs_p) = traits
colnames(Glm_noPCs_p) = traits

result = list()
result$Gglm_PCs_p = Gglm_PCs_p
result$Glm_PCs_p = Glm_PCs_p
result$Gglm_noPCs_p = Gglm_noPCs_p
result$Glm_noPCs_p = Glm_noPCs_p

save(result, file = paste0(path_wtccc,"/result/result_",popugenoanames[popnum],"_elnet_trait_",traits[traitnum],"_PC_",PCnum,".RData"))



####################

#!/bin/bash
#SBATCH --job-name=wtcccelnet
#SBATCH --array=1-308
#SBATCH --cpus-per-task=1
#SBATCH --output=/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping/out/wtcccelnet%a.out
#SBATCH --error=/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping/err/wtcccelnet%a.err
#SBATCH --workdir=/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping/code
#SBATCH --partition=mulan

bash

let k=0

for ((traitnum=1;traitnum<=7;traitnum++)); do
for ((chr=1;chr<=22;chr++)); do
for ((popnum=1;popnum<=2;popnum++)); do
let k=${k}+1
if [ ${k} -eq ${SLURM_ARRAY_TASK_ID} ]; then
PCnum=2
Rscript --verbose ./WTCCC_elnet_GENOA.r ${traitnum}  ${chr} ${PCnum} ${popnum}
PCnum=5
Rscript --verbose ./WTCCC_elnet_GENOA.r ${traitnum}  ${chr} ${PCnum} ${popnum}
PCnum=10
Rscript --verbose ./WTCCC_elnet_GENOA.r ${traitnum}  ${chr} ${PCnum} ${popnum}
fi
done
done
done





#----------------
# AFA 100kb
#----------------

path_wtccc="/net/mulan/home/shanglu/GENOA/data/WTCCC"
traits = c("T1D","T2D","RA","HT","CD","CAD","BD")
library(tidyr)


traitnum=1

count = 0
genes = c()
for(i in 1:22){
	load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_elnet_AFA_100kb_v2.RData"))
	nrow = length(rownames(expr_mat2))
	count = count+nrow
	genes = c(genes,rownames(expr_mat2))
}
count



Gglm_PCs_p = matrix(0,count,7)
Gglm_noPCs_p = matrix(0,count,7)
Glm_PCs_p = matrix(0,count,7)
Glm_noPCs_p = matrix(0,count,7)

for(traitnum in 1:7){

countR = 0
for(i in 1:22){
	print(paste0("Trait: ",traitnum))
	print(paste0("Chr: ",i))
	load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_elnet_AFA_100kb_v2.RData"))

	fam = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".fam")
	WTCCCfam = read.table(fam)
	outcome = WTCCCfam$V6-1
	eigenvec = read.table(paste0(path_wtccc,"/wtccc",traits[traitnum],".PCA.eigenvec"))
	PCs = as.matrix(eigenvec[,3:12])
	
	for(j in 1:dim(expr_mat2)[1]){
		countR = countR + 1
		Gglm_PCs = glm(formula=outcome ~ expr_mat2[j,] + PCs,  family=binomial)
		Glm_PCs = lm(formula=outcome ~ expr_mat2[j,] + PCs)
		Gglm_noPCs = glm(formula=outcome ~ expr_mat2[j,] ,  family=binomial)
		Glm_noPCs = lm(formula=outcome ~ expr_mat2[j,] )

		Gglm_PCs_p[countR,traitnum] = summary(Gglm_PCs)$coefficients[2,4]
		Glm_PCs_p[countR,traitnum] = summary(Glm_PCs)$coefficients[2,4]
		Gglm_noPCs_p[countR,traitnum] = summary(Gglm_noPCs)$coefficients[2,4]
		Glm_noPCs_p[countR,traitnum] = summary(Glm_noPCs)$coefficients[2,4]

	}
}
}


rownames(Gglm_PCs_p) = genes
rownames(Glm_PCs_p) = genes
rownames(Gglm_noPCs_p) = genes
rownames(Glm_noPCs_p) = genes

colnames(Gglm_PCs_p) = traits
colnames(Glm_PCs_p) = traits
colnames(Gglm_noPCs_p) = traits
colnames(Glm_noPCs_p) = traits

result = list()
result$Gglm_PCs_p = Gglm_PCs_p
result$Glm_PCs_p = Glm_PCs_p
result$Gglm_noPCs_p = Gglm_noPCs_p
result$Glm_noPCs_p = Glm_noPCs_p

save(result, file = paste0(path_wtccc,"/result/result_AFA_100kb_elnet.RData"))





#----------------
# CAU 100kb
#----------------

path_wtccc="/net/mulan/home/shanglu/GENOA/data/WTCCC"
traits = c("T1D","T2D","RA","HT","CD","CAD","BD")
library(tidyr)


traitnum=1

count = 0
genes = c()
for(i in 1:22){
	load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_elnet_CAU_100kb_v2.RData"))
	nrow = length(rownames(expr_mat2))
	count = count+nrow
	genes = c(genes,rownames(expr_mat2))
}
count



Gglm_PCs_p = matrix(0,count,7)
Gglm_noPCs_p = matrix(0,count,7)
Glm_PCs_p = matrix(0,count,7)
Glm_noPCs_p = matrix(0,count,7)

for(traitnum in 1:7){

countR = 0
for(i in 1:22){
	print(paste0("Trait: ",traitnum))
	print(paste0("Chr: ",i))
	load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_elnet_CAU_100kb_v2.RData"))

	fam = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".fam")
	WTCCCfam = read.table(fam)
	outcome = WTCCCfam$V6-1
	eigenvec = read.table(paste0(path_wtccc,"/wtccc",traits[traitnum],".PCA.eigenvec"))
	PCs = as.matrix(eigenvec[,3:12])
	
	for(j in 1:dim(expr_mat2)[1]){
		countR = countR + 1
		Gglm_PCs = glm(formula=outcome ~ expr_mat2[j,] + PCs,  family=binomial)
		Glm_PCs = lm(formula=outcome ~ expr_mat2[j,] + PCs)
		Gglm_noPCs = glm(formula=outcome ~ expr_mat2[j,] ,  family=binomial)
		Glm_noPCs = lm(formula=outcome ~ expr_mat2[j,] )

		Gglm_PCs_p[countR,traitnum] = summary(Gglm_PCs)$coefficients[2,4]
		Glm_PCs_p[countR,traitnum] = summary(Glm_PCs)$coefficients[2,4]
		Gglm_noPCs_p[countR,traitnum] = summary(Gglm_noPCs)$coefficients[2,4]
		Glm_noPCs_p[countR,traitnum] = summary(Glm_noPCs)$coefficients[2,4]

	}
}
}


rownames(Gglm_PCs_p) = genes
rownames(Glm_PCs_p) = genes
rownames(Gglm_noPCs_p) = genes
rownames(Glm_noPCs_p) = genes

colnames(Gglm_PCs_p) = traits
colnames(Glm_PCs_p) = traits
colnames(Gglm_noPCs_p) = traits
colnames(Glm_noPCs_p) = traits

result = list()
result$Gglm_PCs_p = Gglm_PCs_p
result$Glm_PCs_p = Glm_PCs_p
result$Gglm_noPCs_p = Gglm_noPCs_p
result$Glm_noPCs_p = Glm_noPCs_p

save(result, file = paste0(path_wtccc,"/result/result_CAU_100kb_elnet.RData"))





#----------------
# HIS 100kb
#----------------

path_wtccc="/net/mulan/home/shanglu/GENOA/data/WTCCC"
traits = c("T1D","T2D","RA","HT","CD","CAD","BD")
library(tidyr)


traitnum=1

count = 0
genes = c()
for(i in 1:22){
	load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_elnet_HIS_100kb_v2.RData"))
	nrow = length(rownames(expr_mat2))
	count = count+nrow
	genes = c(genes,rownames(expr_mat2))
}
count



Gglm_PCs_p = matrix(0,count,7)
Gglm_noPCs_p = matrix(0,count,7)
Glm_PCs_p = matrix(0,count,7)
Glm_noPCs_p = matrix(0,count,7)

for(traitnum in 1:7){

countR = 0
for(i in 1:22){
	print(paste0("Trait: ",traitnum))
	print(paste0("Chr: ",i))
	load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_elnet_HIS_100kb_v2.RData"))

	fam = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".fam")
	WTCCCfam = read.table(fam)
	outcome = WTCCCfam$V6-1
	eigenvec = read.table(paste0(path_wtccc,"/wtccc",traits[traitnum],".PCA.eigenvec"))
	PCs = as.matrix(eigenvec[,3:12])
	
	for(j in 1:dim(expr_mat2)[1]){
		countR = countR + 1
		Gglm_PCs = glm(formula=outcome ~ expr_mat2[j,] + PCs,  family=binomial)
		Glm_PCs = lm(formula=outcome ~ expr_mat2[j,] + PCs)
		Gglm_noPCs = glm(formula=outcome ~ expr_mat2[j,] ,  family=binomial)
		Glm_noPCs = lm(formula=outcome ~ expr_mat2[j,] )

		Gglm_PCs_p[countR,traitnum] = summary(Gglm_PCs)$coefficients[2,4]
		Glm_PCs_p[countR,traitnum] = summary(Glm_PCs)$coefficients[2,4]
		Gglm_noPCs_p[countR,traitnum] = summary(Gglm_noPCs)$coefficients[2,4]
		Glm_noPCs_p[countR,traitnum] = summary(Glm_noPCs)$coefficients[2,4]

	}
}
}


rownames(Gglm_PCs_p) = genes
rownames(Glm_PCs_p) = genes
rownames(Gglm_noPCs_p) = genes
rownames(Glm_noPCs_p) = genes

colnames(Gglm_PCs_p) = traits
colnames(Glm_PCs_p) = traits
colnames(Gglm_noPCs_p) = traits
colnames(Glm_noPCs_p) = traits

result = list()
result$Gglm_PCs_p = Gglm_PCs_p
result$Glm_PCs_p = Glm_PCs_p
result$Gglm_noPCs_p = Gglm_noPCs_p
result$Glm_noPCs_p = Glm_noPCs_p

save(result, file = paste0(path_wtccc,"/result/result_HIS_100kb_elnet.RData"))






#----------------
# AFA 1Mb
#----------------

path_wtccc="/net/mulan/home/shanglu/GENOA/data/WTCCC"
traits = c("T1D","T2D","RA","HT","CD","CAD","BD")
library(tidyr)


traitnum=1

count = 0
genes = c()
for(i in 1:22){
	load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_elnet_AFA_1Mb_v2.RData"))
	nrow = length(rownames(expr_mat2))
	count = count+nrow
	genes = c(genes,rownames(expr_mat2))
}
count



Gglm_PCs_p = matrix(0,count,7)
Gglm_noPCs_p = matrix(0,count,7)
Glm_PCs_p = matrix(0,count,7)
Glm_noPCs_p = matrix(0,count,7)

for(traitnum in 1:7){

countR = 0
for(i in 1:22){
	print(paste0("Trait: ",traitnum))
	print(paste0("Chr: ",i))
	load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_elnet_AFA_1Mb_v2.RData"))

	fam = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".fam")
	WTCCCfam = read.table(fam)
	outcome = WTCCCfam$V6-1
	eigenvec = read.table(paste0(path_wtccc,"/wtccc",traits[traitnum],".PCA.eigenvec"))
	PCs = as.matrix(eigenvec[,3:12])
	
	for(j in 1:dim(expr_mat2)[1]){
		countR = countR + 1
		Gglm_PCs = glm(formula=outcome ~ expr_mat2[j,] + PCs,  family=binomial)
		Glm_PCs = lm(formula=outcome ~ expr_mat2[j,] + PCs)
		Gglm_noPCs = glm(formula=outcome ~ expr_mat2[j,] ,  family=binomial)
		Glm_noPCs = lm(formula=outcome ~ expr_mat2[j,] )

		Gglm_PCs_p[countR,traitnum] = summary(Gglm_PCs)$coefficients[2,4]
		Glm_PCs_p[countR,traitnum] = summary(Glm_PCs)$coefficients[2,4]
		Gglm_noPCs_p[countR,traitnum] = summary(Gglm_noPCs)$coefficients[2,4]
		Glm_noPCs_p[countR,traitnum] = summary(Glm_noPCs)$coefficients[2,4]

	}
}
}


rownames(Gglm_PCs_p) = genes
rownames(Glm_PCs_p) = genes
rownames(Gglm_noPCs_p) = genes
rownames(Glm_noPCs_p) = genes

colnames(Gglm_PCs_p) = traits
colnames(Glm_PCs_p) = traits
colnames(Gglm_noPCs_p) = traits
colnames(Glm_noPCs_p) = traits

result = list()
result$Gglm_PCs_p = Gglm_PCs_p
result$Glm_PCs_p = Glm_PCs_p
result$Gglm_noPCs_p = Gglm_noPCs_p
result$Glm_noPCs_p = Glm_noPCs_p

save(result, file = paste0(path_wtccc,"/result/result_AFA_1Mb_elnet.RData"))





#----------------
# CAU 1Mb
#----------------

path_wtccc="/net/mulan/home/shanglu/GENOA/data/WTCCC"
traits = c("T1D","T2D","RA","HT","CD","CAD","BD")
library(tidyr)


traitnum=1

count = 0
genes = c()
for(i in 1:22){
	load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_elnet_CAU_1Mb_v2.RData"))
	nrow = length(rownames(expr_mat2))
	count = count+nrow
	genes = c(genes,rownames(expr_mat2))
}
count



Gglm_PCs_p = matrix(0,count,7)
Gglm_noPCs_p = matrix(0,count,7)
Glm_PCs_p = matrix(0,count,7)
Glm_noPCs_p = matrix(0,count,7)

for(traitnum in 1:7){

countR = 0
for(i in 1:22){
	print(paste0("Trait: ",traitnum))
	print(paste0("Chr: ",i))
	load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_elnet_CAU_1Mb_v2.RData"))

	fam = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".fam")
	WTCCCfam = read.table(fam)
	outcome = WTCCCfam$V6-1
	eigenvec = read.table(paste0(path_wtccc,"/wtccc",traits[traitnum],".PCA.eigenvec"))
	PCs = as.matrix(eigenvec[,3:12])
	
	for(j in 1:dim(expr_mat2)[1]){
		countR = countR + 1
		Gglm_PCs = glm(formula=outcome ~ expr_mat2[j,] + PCs,  family=binomial)
		Glm_PCs = lm(formula=outcome ~ expr_mat2[j,] + PCs)
		Gglm_noPCs = glm(formula=outcome ~ expr_mat2[j,] ,  family=binomial)
		Glm_noPCs = lm(formula=outcome ~ expr_mat2[j,] )

		Gglm_PCs_p[countR,traitnum] = summary(Gglm_PCs)$coefficients[2,4]
		Glm_PCs_p[countR,traitnum] = summary(Glm_PCs)$coefficients[2,4]
		Gglm_noPCs_p[countR,traitnum] = summary(Gglm_noPCs)$coefficients[2,4]
		Glm_noPCs_p[countR,traitnum] = summary(Glm_noPCs)$coefficients[2,4]

	}
}
}


rownames(Gglm_PCs_p) = genes
rownames(Glm_PCs_p) = genes
rownames(Gglm_noPCs_p) = genes
rownames(Glm_noPCs_p) = genes

colnames(Gglm_PCs_p) = traits
colnames(Glm_PCs_p) = traits
colnames(Gglm_noPCs_p) = traits
colnames(Glm_noPCs_p) = traits

result = list()
result$Gglm_PCs_p = Gglm_PCs_p
result$Glm_PCs_p = Glm_PCs_p
result$Gglm_noPCs_p = Gglm_noPCs_p
result$Glm_noPCs_p = Glm_noPCs_p

save(result, file = paste0(path_wtccc,"/result/result_CAU_1Mb_elnet.RData"))





#----------------
# HIS 1Mb
#----------------

path_wtccc="/net/mulan/home/shanglu/GENOA/data/WTCCC"
traits = c("T1D","T2D","RA","HT","CD","CAD","BD")
library(tidyr)


traitnum=1

count = 0
genes = c()
for(i in 1:22){
	load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_elnet_HIS_1Mb_v2.RData"))
	nrow = length(rownames(expr_mat2))
	count = count+nrow
	genes = c(genes,rownames(expr_mat2))
}
count



Gglm_PCs_p = matrix(0,count,7)
Gglm_noPCs_p = matrix(0,count,7)
Glm_PCs_p = matrix(0,count,7)
Glm_noPCs_p = matrix(0,count,7)

for(traitnum in 1:7){

countR = 0
for(i in 1:22){
	print(paste0("Trait: ",traitnum))
	print(paste0("Chr: ",i))
	load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_elnet_HIS_1Mb_v2.RData"))

	fam = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".fam")
	WTCCCfam = read.table(fam)
	outcome = WTCCCfam$V6-1
	eigenvec = read.table(paste0(path_wtccc,"/wtccc",traits[traitnum],".PCA.eigenvec"))
	PCs = as.matrix(eigenvec[,3:12])
	
	for(j in 1:dim(expr_mat2)[1]){
		countR = countR + 1
		Gglm_PCs = glm(formula=outcome ~ expr_mat2[j,] + PCs,  family=binomial)
		Glm_PCs = lm(formula=outcome ~ expr_mat2[j,] + PCs)
		Gglm_noPCs = glm(formula=outcome ~ expr_mat2[j,] ,  family=binomial)
		Glm_noPCs = lm(formula=outcome ~ expr_mat2[j,] )

		Gglm_PCs_p[countR,traitnum] = summary(Gglm_PCs)$coefficients[2,4]
		Glm_PCs_p[countR,traitnum] = summary(Glm_PCs)$coefficients[2,4]
		Gglm_noPCs_p[countR,traitnum] = summary(Gglm_noPCs)$coefficients[2,4]
		Glm_noPCs_p[countR,traitnum] = summary(Glm_noPCs)$coefficients[2,4]

	}
}
}


rownames(Gglm_PCs_p) = genes
rownames(Glm_PCs_p) = genes
rownames(Gglm_noPCs_p) = genes
rownames(Glm_noPCs_p) = genes

colnames(Gglm_PCs_p) = traits
colnames(Glm_PCs_p) = traits
colnames(Gglm_noPCs_p) = traits
colnames(Glm_noPCs_p) = traits

result = list()
result$Gglm_PCs_p = Gglm_PCs_p
result$Glm_PCs_p = Glm_PCs_p
result$Gglm_noPCs_p = Gglm_noPCs_p
result$Glm_noPCs_p = Glm_noPCs_p

save(result, file = paste0(path_wtccc,"/result/result_HIS_1Mb_elnet.RData"))



```

Tables
```
library(tidyr)
library(dplyr) 

path_wtccc="/net/mulan/home/shanglu/GENOA/data/WTCCC"
traits = c("T1D","T2D","RA","HT","CD","CAD","BD")

traitnum = 0

traitnum = traitnum + 1
popugenoanamesCapital = c(paste0("AA_elnet_trait_",traits[traitnum],"_PC_10") , paste0("EA_elnet_trait_",traits[traitnum],"_PC_10"), "AFA_100kb_elnet", "CAU_100kb_elnet", "HIS_100kb_elnet")


ph = c()
phnum = c()
genenum = c()
for(popugenoa in 3:5){
	load(paste0(path_wtccc,"/result/result_",popugenoanamesCapital[popugenoa],"_elnet.RData"))
	ph_tmp = c(result$Gglm_PCs_p[,traitnum])
	ph = c(ph, ph_tmp)
	phnum[popugenoa] = length(ph_tmp)
	genenum = c(genenum, sum(ph_tmp< 0.05/dim(result$Gglm_PCs_p)[1]))
}
[1] 17 25 25

ph = c()
phnum = c()
genenum = c()
for(popugenoa in 3:5){
	load(paste0(path_wtccc,"/result/result_",popugenoanamesCapital[popugenoa],"_elnet.RData"))
	ph_tmp = c(result$Gglm_noPCs_p[,traitnum])
	ph = c(ph, ph_tmp)
	phnum[popugenoa] = length(ph_tmp)
	genenum = c(genenum, sum(ph_tmp< 0.05/dim(result$Gglm_PCs_p)[1]))
}
[1] 22 31 28
aa:36
EA:26

Population = c(rep("AA",phnum[1]),
rep("EA",phnum[2]),
rep("AFA",phnum[3]),
rep("CAU",phnum[4]),
rep("HIS",phnum[5]))
dat = data.frame(ph,Population )
dat$Population = as.character(Population)
mydat = dat

my.pvalue.list<-list("AA"=mydat[mydat$Population=="AA",1], "EA"=mydat[mydat$Population=="EA",1],
"AFA"=mydat[mydat$Population=="AFA",1],"CAU"=mydat[mydat$Population=="CAU",1],
"HIS"=mydat[mydat$Population=="HIS",1])

pdf(paste0("QQ_trait",traits[traitnum],"_100kb.pdf") ,width = 6, height = 6)
qqunif.plot(my.pvalue.list, auto.key=list(corner=c(.95,.05)))
dev.off()







```

More Tables
```
pre_WTCCC = read.table("Pre_WTCCC_genes.txt", header=T)
In_WTCCC = read.table("In_WTCCC_genes.txt", header=T)
pre_WTCCC$Gene = as.character(pre_WTCCC$Gene)
pre_WTCCC$Trait = as.character(pre_WTCCC$Trait)
In_WTCCC$Gene = as.character(In_WTCCC$Gene)
In_WTCCC$Trait = as.character(In_WTCCC$Trait)


all_WTCCC = rbind(pre_WTCCC,In_WTCCC)
total_WTCCC = unique(all_WTCCC)

load("/net/mulan/home/shanglu/GENOA/data/AA/gene_AA_anno_order_correct.RData")
load("/net/mulan/home/shanglu/GENOA/data/EA/gene_EA_anno_order_correct.RData")
path_wtccc="/net/mulan/home/shanglu/GENOA/data/WTCCC"





path_wtccc="/net/mulan/home/shanglu/GENOA/data/WTCCC"


library(tidyr)

traits = c("T1D","T2D","RA","HT","CD","CAD","BD")
popugenoanames = c("AA" , "EA")



#######################
# first consider AA
popnum = 1
load("/net/mulan/home/shanglu/GENOA/data/AA/gene_AA_anno_order_correct.RData")
#######################
for(PCnum in c(2,5,10)){
print("PCnum")
print(PCnum)
for(traitnum in 1:7){
print(traitnum)
i=1
load(paste0(path_wtccc,"/result/result_",popugenoanames[popnum],"_elnet_trait_",traits[traitnum],"_PC_",PCnum,"_chr_",i,".RData"))
Gglm_PCs_p = result$Gglm_PCs_p
#Glm_PCs_p = result$Glm_PCs_p
Gglm_noPCs_p = result$Gglm_noPCs_p
#Glm_noPCs_p = result$Glm_noPCs_p

for(i in 2:22){
load(paste0(path_wtccc,"/result/result_",popugenoanames[popnum],"_elnet_trait_",traits[traitnum],"_PC_",PCnum,"_chr_",i,".RData"))
Gglm_PCs_p = rbind(Gglm_PCs_p,result$Gglm_PCs_p)
#Glm_PCs_p = rbind(Glm_PCs_p, result$Glm_PCs_p)
Gglm_noPCs_p = rbind(Gglm_noPCs_p, result$Gglm_noPCs_p)
#Glm_noPCs_p = rbind(Glm_noPCs_p, result$Glm_noPCs_p)
}

ind1 =  which(Gglm_PCs_p[,traitnum] < 0.05/dim(Gglm_PCs_p)[1])
gene_signif_Gglm_PCs_p = rownames(Gglm_PCs_p)[ind1]
ind = match(gene_signif_Gglm_PCs_p, gene_AA_anno_order$GENE)
gene_symbol1 = as.character(gene_AA_anno_order$gene)[ind]
pvalue_Gglm_PCs_p = Gglm_PCs_p[ind1, traitnum]

#ind2 =  which(Glm_PCs_p[,traitnum] < 0.05/dim(Glm_PCs_p)[1])
#gene_signif_Glm_PCs_p = rownames(Glm_PCs_p)[ind2]
#pvalue_Glm_PCs_p = Glm_PCs_p[ind2, traitnum]
#ind = match(gene_signif_Glm_PCs_p, gene_AA_anno_order$GENE)
#gene_symbol2 = as.character(gene_AA_anno_order$gene)[ind]

ind3 =  which(Gglm_noPCs_p[,traitnum] < 0.05/dim(Gglm_noPCs_p)[1])
gene_signif_Gglm_noPCs_p = rownames(Gglm_noPCs_p)[ind3]
pvalue_Gglm_noPCs_p = Gglm_noPCs_p[ind3, traitnum]
ind = match(gene_signif_Gglm_noPCs_p, gene_AA_anno_order$GENE)
gene_symbol3 = as.character(gene_AA_anno_order$gene)[ind]


#ind4 =  which(Glm_noPCs_p[,traitnum] < 0.05/dim(Glm_noPCs_p)[1])
#gene_signif_Glm_noPCs_p = rownames(Glm_noPCs_p)[ind4]
#pvalue_Glm_noPCs_p = Glm_noPCs_p[ind4, traitnum]
#ind = match(gene_signif_Glm_noPCs_p, gene_AA_anno_order$GENE)
#gene_symbol4 = as.character(gene_AA_anno_order$gene)[ind]



dat = data.frame(gene_signif_Gglm_PCs_p, gene_symbol1,pvalue_Gglm_PCs_p)
dat = dat[order( dat[,3] ),]
write.csv(dat, paste0(path_wtccc,"/result/Dec24_result_",popugenoanames[popnum],"_elnet_trait_",traits[traitnum],"_PC_",PCnum,".csv"))

dat = data.frame(gene_signif_Gglm_noPCs_p, gene_symbol3,pvalue_Gglm_noPCs_p)
dat = dat[order( dat[,3] ),]

write.csv(dat, paste0(path_wtccc,"/result/Dec24_result_",popugenoanames[popnum],"_elnet_trait_",traits[traitnum],"_PC_",0,".csv"))

}

}


#######################
# next consider EA
popnum = 2
load("/net/mulan/home/shanglu/GENOA/data/EA/gene_EA_anno_order_correct.RData")
#######################

for(PCnum in c(2,5,10)){
print("PCnum")
print(PCnum)
for(traitnum in 1:7){
print(traitnum)
i=1
load(paste0(path_wtccc,"/result/result_",popugenoanames[popnum],"_elnet_trait_",traits[traitnum],"_PC_",PCnum,"_chr_",i,".RData"))
Gglm_PCs_p = result$Gglm_PCs_p
#Glm_PCs_p = result$Glm_PCs_p
Gglm_noPCs_p = result$Gglm_noPCs_p
#Glm_noPCs_p = result$Glm_noPCs_p

for(i in 2:22){
load(paste0(path_wtccc,"/result/result_",popugenoanames[popnum],"_elnet_trait_",traits[traitnum],"_PC_",PCnum,"_chr_",i,".RData"))
Gglm_PCs_p = rbind(Gglm_PCs_p,result$Gglm_PCs_p)
#Glm_PCs_p = rbind(Glm_PCs_p, result$Glm_PCs_p)
Gglm_noPCs_p = rbind(Gglm_noPCs_p, result$Gglm_noPCs_p)
#Glm_noPCs_p = rbind(Glm_noPCs_p, result$Glm_noPCs_p)
}

ind1 =  which(Gglm_PCs_p[,traitnum] < 0.05/dim(Gglm_PCs_p)[1])
gene_signif_Gglm_PCs_p = rownames(Gglm_PCs_p)[ind1]
ind = match(gene_signif_Gglm_PCs_p, gene_EA_anno_order$GENE)
gene_symbol1 = as.character(gene_EA_anno_order$gene)[ind]
pvalue_Gglm_PCs_p = Gglm_PCs_p[ind1, traitnum]

#ind2 =  which(Glm_PCs_p[,traitnum] < 0.05/dim(Glm_PCs_p)[1])
#gene_signif_Glm_PCs_p = rownames(Glm_PCs_p)[ind2]
#pvalue_Glm_PCs_p = Glm_PCs_p[ind2, traitnum]
#ind = match(gene_signif_Glm_PCs_p, gene_AA_anno_order$GENE)
#gene_symbol2 = as.character(gene_AA_anno_order$gene)[ind]

ind3 =  which(Gglm_noPCs_p[,traitnum] < 0.05/dim(Gglm_noPCs_p)[1])
gene_signif_Gglm_noPCs_p = rownames(Gglm_noPCs_p)[ind3]
pvalue_Gglm_noPCs_p = Gglm_noPCs_p[ind3, traitnum]
ind = match(gene_signif_Gglm_noPCs_p, gene_EA_anno_order$GENE)
gene_symbol3 = as.character(gene_EA_anno_order$gene)[ind]


#ind4 =  which(Glm_noPCs_p[,traitnum] < 0.05/dim(Glm_noPCs_p)[1])
#gene_signif_Glm_noPCs_p = rownames(Glm_noPCs_p)[ind4]
#pvalue_Glm_noPCs_p = Glm_noPCs_p[ind4, traitnum]
#ind = match(gene_signif_Glm_noPCs_p, gene_AA_anno_order$GENE)
#gene_symbol4 = as.character(gene_AA_anno_order$gene)[ind]



dat = data.frame(gene_signif_Gglm_PCs_p, gene_symbol1,pvalue_Gglm_PCs_p)
dat = dat[order( dat[,3] ),]
write.csv(dat, paste0(path_wtccc,"/result/Dec24_result_",popugenoanames[popnum],"_elnet_trait_",traits[traitnum],"_PC_",PCnum,".csv"))

dat = data.frame(gene_signif_Gglm_noPCs_p, gene_symbol3,pvalue_Gglm_noPCs_p)
dat = dat[order( dat[,3] ),]

write.csv(dat, paste0(path_wtccc,"/result/Dec24_result_",popugenoanames[popnum],"_elnet_trait_",traits[traitnum],"_PC_",0,".csv"))

}

}





# count num:

traits = c("T1D","T2D","RA","HT","CD","CAD","BD")
popugenoanames = c("AA" , "EA")

# AA
countnum = c()
popnum = 1
for(traitnum in 1:7){
dat = read.csv(paste0(path_wtccc,"/result/Dec24_result_AA_elnet_trait_",traits[traitnum],"_PC_10.csv"))
countnum[traitnum] = dim(dat)[1]
}
countnum
[1] 30  0 12  0  5  0  1
# total: 48
# EA
countnum = c()
popnum = 1
for(traitnum in 1:7){
dat = read.csv(paste0(path_wtccc,"/result/Dec24_result_EA_elnet_trait_",traits[traitnum],"_PC_10.csv"))
countnum[traitnum] = dim(dat)[1]
}

[1] 24  0 11  0  1  0  0
# total: 36

######## bslmm
# AA
countnum = c()
popnum = 1
for(traitnum in 1:7){
dat = read.csv(paste0(path_wtccc,"/result/Dec24_result_AA_bslmm_trait_",traits[traitnum],"_PC_10.csv"))
countnum[traitnum] = dim(dat)[1]
}
countnum
[1] 62  0 18  0  8  0  0
# total: 48
# EA
countnum = c()
popnum = 1
for(traitnum in 1:7){
dat = read.csv(paste0(path_wtccc,"/result/Dec24_result_EA_bslmm_trait_",traits[traitnum],"_PC_10.csv"))
countnum[traitnum] = dim(dat)[1]
}

[1] 68  1 24  0  5  1  0


##############################################################

# use all genes downloaded

##############################################################

traits = c("T1D","T2D","RA","HT","CD","CAD","BD")
mat = matrix(0,7,7)

for(traitnum in 1:7){
wtcccgene = read.csv(paste0("/net/mulan/home/shanglu/GENOA/data/WTCCC/result/",traits[traitnum],"GeneCards-SearchResults.csv"))
wtcccgene$Gene.Symbol = as.character(wtcccgene$Gene.Symbol)

dat = read.csv(paste0(path_wtccc,"/result/Dec24_result_AA_bslmm_trait_",traits[traitnum],"_PC_10.csv"))
mat[traitnum,1] = length(intersect(as.character(dat[,3]), wtcccgene$Gene.Symbol))

dat = read.csv(paste0(path_wtccc,"/result/Dec24_result_EA_bslmm_trait_",traits[traitnum],"_PC_10.csv"))
mat[traitnum,2] = length(intersect(as.character(dat[,3]), wtcccgene$Gene.Symbol))

dat = read.csv(paste0(path_wtccc,"/result/Dec24_result_AA_elnet_trait_",traits[traitnum],"_PC_10.csv"))
mat[traitnum,3] = length(intersect( as.character(dat[,3]), wtcccgene$Gene.Symbol))

dat = read.csv(paste0(path_wtccc,"/result/Dec24_result_EA_elnet_trait_",traits[traitnum],"_PC_10.csv"))
mat[traitnum,4] = length(intersect(as.character(dat[,3]), wtcccgene$Gene.Symbol))

load(paste0(path_wtccc,"/result/result_AFA_100kb_elnet.RData"))
genenames = rownames(result$Gglm_PCs_p)[which(result$Gglm_PCs_p[,traitnum]<0.05/dim(result$Gglm_PCs_p)[1])]
ind = match(genenames, gene_AA_anno_order$GENE)
gene_symbol1= as.character(gene_AA_anno_order$gene)[ind]
mat[traitnum,5] = length(intersect(gene_symbol1, wtcccgene$Gene.Symbol))


load(paste0(path_wtccc,"/result/result_CAU_100kb_elnet.RData"))
genenames = rownames(result$Gglm_PCs_p)[which(result$Gglm_PCs_p[,traitnum]<0.05/dim(result$Gglm_PCs_p)[1])]
ind = match(genenames, gene_AA_anno_order$GENE)
gene_symbol1= as.character(gene_AA_anno_order$gene)[ind]
num = length(intersect(gene_symbol1, wtcccgene$Gene.Symbol))
mat[traitnum,6] = num

load(paste0(path_wtccc,"/result/result_HIS_100kb_elnet.RData"))
genenames = rownames(result$Gglm_PCs_p)[which(result$Gglm_PCs_p[,traitnum]<0.05/dim(result$Gglm_PCs_p)[1])]
ind = match(genenames, gene_AA_anno_order$GENE)
gene_symbol1= as.character(gene_AA_anno_order$gene)[ind]
num = length(intersect(gene_symbol1, wtcccgene$Gene.Symbol))
mat[traitnum,7] = num
}

colnames(mat) = c("AAbslmm", "EAbslmm", "AAelnet", "EAelnet", "AFAelnet", "CAUelnet", "HISelnet")
rownames(mat) = traits

mat 
    AAbslmm EAbslmm AAelnet EAelnet AFAelnet CAUelnet HISelnet
T1D      23      27      11       9        6       12       11
T2D       0       1       0       0        0        0        0
RA       10      16      10      10        9       11       11
HT        0       0       0       0        0        0        0
CD        7       5       5       1        0        2        3
CAD       0       1       0       0        0        0        0
BD        0       0       0       0        0        0        0


##############################################################

# use genes downloaded with higher score

##############################################################

score = 0

traits = c("T1D","T2D","RA","HT","CD","CAD","BD")
mat = matrix(0,7,7)

for(traitnum in 1:7){
wtcccgene = read.csv(paste0("/net/mulan/home/shanglu/GENOA/data/WTCCC/result/",traits[traitnum],"GeneCards-SearchResults.csv"))
Gene.Symbol = as.character(wtcccgene$Gene.Symbol)[wtcccgene$Relevance.score>=score]

dat = read.csv(paste0(path_wtccc,"/result/Dec24_result_AA_bslmm_trait_",traits[traitnum],"_PC_10.csv"))
mat[traitnum,1] = length(intersect(as.character(dat[,3]), Gene.Symbol))

dat = read.csv(paste0(path_wtccc,"/result/Dec24_result_EA_bslmm_trait_",traits[traitnum],"_PC_10.csv"))
mat[traitnum,2] = length(intersect(as.character(dat[,3]), Gene.Symbol))

dat = read.csv(paste0(path_wtccc,"/result/Dec24_result_AA_elnet_trait_",traits[traitnum],"_PC_10.csv"))
mat[traitnum,3] = length(intersect( as.character(dat[,3]), Gene.Symbol))

dat = read.csv(paste0(path_wtccc,"/result/Dec24_result_EA_elnet_trait_",traits[traitnum],"_PC_10.csv"))
mat[traitnum,4] = length(intersect(as.character(dat[,3]), Gene.Symbol))

load(paste0(path_wtccc,"/result/result_AFA_100kb_elnet.RData"))
genenames = rownames(result$Gglm_PCs_p)[which(result$Gglm_PCs_p[,traitnum]<0.05/dim(result$Gglm_PCs_p)[1])]
ind = match(genenames, gene_AA_anno_order$GENE)
gene_symbol1= as.character(gene_AA_anno_order$gene)[ind]
mat[traitnum,5] = length(intersect(gene_symbol1, Gene.Symbol))


load(paste0(path_wtccc,"/result/result_CAU_100kb_elnet.RData"))
genenames = rownames(result$Gglm_PCs_p)[which(result$Gglm_PCs_p[,traitnum]<0.05/dim(result$Gglm_PCs_p)[1])]
ind = match(genenames, gene_AA_anno_order$GENE)
gene_symbol1= as.character(gene_AA_anno_order$gene)[ind]
num = length(intersect(gene_symbol1, Gene.Symbol))
mat[traitnum,6] = num

load(paste0(path_wtccc,"/result/result_HIS_100kb_elnet.RData"))
genenames = rownames(result$Gglm_PCs_p)[which(result$Gglm_PCs_p[,traitnum]<0.05/dim(result$Gglm_PCs_p)[1])]
ind = match(genenames, gene_AA_anno_order$GENE)
gene_symbol1= as.character(gene_AA_anno_order$gene)[ind]
num = length(intersect(gene_symbol1, Gene.Symbol))
mat[traitnum,7] = num
}

colnames(mat) = c("AAbslmm", "EAbslmm", "AAelnet", "EAelnet", "AFAelnet", "CAUelnet", "HISelnet")
rownames(mat) = traits
mat


# median score
> mat
    AAbslmm EAbslmm AAelnet EAelnet AFAelnet CAUelnet HISelnet
T1D      13      15       6       5        4        8        8
T2D       0       1       0       0        0        0        0
RA        9      12       9       8        8        9       10
HT        0       0       0       0        0        0        0
CD        7       5       5       1        0        2        3
CAD       0       1       0       0        0        0        0
BD        0       0       0       0        0        0        0

# score = 10
> mat
    AAbslmm EAbslmm AAelnet EAelnet AFAelnet CAUelnet HISelnet
T1D       2       2       1       0        1        2        3
T2D       0       1       0       0        0        0        0
RA        5       5       2       2        5        4        3
HT        0       0       0       0        0        0        0
CD        2       2       4       1        0        0        1
CAD       0       1       0       0        0        0        0
BD        0       0       0       0        0        0        0

# score = 5
> mat
    AAbslmm EAbslmm AAelnet EAelnet AFAelnet CAUelnet HISelnet
T1D       5       5       3       1        1        4        4
T2D       0       1       0       0        0        0        0
RA        8      10       6       5        7        6        8
HT        0       0       0       0        0        0        0
CD        4       3       4       1        0        0        2
CAD       0       1       0       0        0        0        0
BD        0       0       0       0        0        0        0

# score = 1
> mat 
    AAbslmm EAbslmm AAelnet EAelnet AFAelnet CAUelnet HISelnet
T1D      12      11       4       3        1        5        6
T2D       0       1       0       0        0        0        0
RA       10      15       9       9        9       10       11
HT        0       0       0       0        0        0        0
CD        7       5       5       1        0        2        3
CAD       0       1       0       0        0        0        0
BD        0       0       0       0        0        0        0

# score = 0.5
> mat
    AAbslmm EAbslmm AAelnet EAelnet AFAelnet CAUelnet HISelnet
T1D      13      14       6       5        4        8        8
T2D       0       1       0       0        0        0        0
RA       10      15       9       9        9       11       11
HT        0       0       0       0        0        0        0
CD        7       5       5       1        0        2        3
CAD       0       1       0       0        0        0        0
BD        0       0       0       0        0        0        0

# score = 0.1
> mat
    AAbslmm EAbslmm AAelnet EAelnet AFAelnet CAUelnet HISelnet
T1D      23      27      11       9        6       12       11
T2D       0       1       0       0        0        0        0
RA       10      16      10      10        9       11       11
HT        0       0       0       0        0        0        0
CD        7       5       5       1        0        2        3
CAD       0       1       0       0        0        0        0
BD        0       0       0       0        0        0        0
# score >=0
> mat
    AAbslmm EAbslmm AAelnet EAelnet AFAelnet CAUelnet HISelnet
T1D      23      27      11       9        6       12       11
T2D       0       1       0       0        0        0        0
RA       10      16      10      10        9       11       11
HT        0       0       0       0        0        0        0
CD        7       5       5       1        0        2        3
CAD       0       1       0       0        0        0        0
BD        0       0       0       0        0        0        0






```

# FIGURE
```R
############### 
WTCCC bar plots


numbers = c(62 ,30 ,68 ,	24 ,17 ,25 ,	25 ,
0 ,	0 ,	1 ,	0	,0	,0,	0,
18 ,	12 ,24 ,	11,	14 ,	14 ,	14 ,
8 ,	5 ,	5 ,	1,	0	,2 ,	3,
0	,0	,1, 	0,	0	,0,	0,
0	,1,0	,0	,0,	0,	0)

traits = c("T1D","T2D","RA","CD","CAD","BD")

population = c("AA(bslmm)", "AA(elnet)","EA(bslmm)", "EA(elnet)","AFA","CAU","HIS")
mat = matrix(numbers,7,6)
colnames(mat) = traits
rownames(mat) = population



k=0

k = k + 1
counts = mat[,k]
popu = rownames(mat)
dat = data.frame(popu,counts )
dat$popu = factor(dat$popu, levels=popu, order=T)

pdf(paste0("WTCCC_barplot_",colnames(mat)[k],".pdf"),width=8,height=4)
ggplot(data=dat, aes(x=popu, y=counts,fill = popu)) +
#scale_fill_manual(values=c("#56B4E9", "honeydew2", "lightpink","lavender","sandybrown","palegreen","khaki"))+
  geom_bar(stat="identity", position=position_dodge(),width = 0.5)+
  geom_text(aes(label=counts), vjust=1.6, color="black",
            position = position_dodge(0.9), size=8)+
  scale_fill_brewer(palette="Paired")+
  theme_bw(base_size=18)+
  theme(legend.position = "none")+
  labs(title=paste0(colnames(mat)[k]),x="", y = "Count")
dev.off()


k=0

k = k + 1
counts = mat[,k]
popu = rownames(mat)
dat = data.frame(popu,counts )
dat$popu = factor(dat$popu, levels=popu, order=T)

pdf(paste0("WTCCC_barplot_",colnames(mat)[k],"_legend.pdf"),width=8,height=4)
ggplot(data=dat, aes(x=popu, y=counts,fill = popu)) +
#scale_fill_manual(values=c("#56B4E9", "honeydew2", "lightpink","lavender","sandybrown","palegreen","khaki"))+
  geom_bar(stat="identity", position=position_dodge(),width = 0.5)+
  geom_text(aes(label=counts), vjust=1.6, color="black",
            position = position_dodge(0.9), size=8)+
  scale_fill_brewer(palette="Paired")+
  theme_bw(base_size=18)+
  labs(title=paste0(colnames(mat)[k]),x="", y = "Count")
dev.off()
```
