
```
# extract all SNPs that we have mapped to genes
pathEA="/net/mulan/home/shanglu/GENOA/data/EA"

load(paste0(pathEA,"/gene_EA_anno_protein.RData"))

for(i in 1:22){
names = c()
chr.num=read.table(paste0(pathEA,"/Gene_file/chr",i,".num"))$V1
print("chr.num read")

gene_anno_EA_pro = gene_EA_anno
gene_anno_pro_chr=gene_anno_EA_pro[gene_anno_EA_pro$chr==i,]

print("gene_chr_info read")

snp_chr=read.table(paste0(pathEA,"/filtered_info/all_filtered_EU.chr",i))
							
print("snp_chr read")

snp_pos=snp_chr$V4
snp_nameonchr=as.character(snp_chr$V3)	

start=gene_anno_pro_chr$low-100000
end  =gene_anno_pro_chr$up+100000
for( j in 1:chr.num){
temp_index <- which(snp_pos >= start[j] &  snp_pos <= end[j] )
snp_name= as.character(snp_nameonchr[temp_index])
name=snp_name[grep("^rs", snp_name)]
names = c(names, name)
}
names = unique(names)
write.table(names, paste0(pathEA,"/filtered_info/mappedSNPs_chr_",i,".txt"),quote=F, row.names=F, col.names=F)
}


```

```


path_EA="/net/mulan/home/shanglu/GENOA/data/EA/VCFtoBED"
fulllistEA = read.table(paste0(path_EA,"/EA_full_list_mulan.txt"),header=F)

for(i in 1:22){
fulllistEA_chr = fulllistEA[-c(1,i),]
write.table(fulllistEA_chr, file = paste0(path_EA,"/fulllist_EA_outchr_",i,".txt"),quote=F,col.names=F,row.names=F)
}


```
first make a combined version of bed file, then extract one chromosome at a time
```
cd /net/mulan/home/shanglu/GENOA/data/EA/filtered_info

qsub -cwd -b y -N EA  -q mulan "plink --noweb --bfile  mappedSNPs_chr_1 --merge-list EA_full_list_mulan.txt --make-bed --out EAcombined"
```
make plink file for kicking out each chromosome one at a time

```
for i in `seq 1 22`
do qsub -cwd -b y -N EA -q mulan "plink --noweb --bfile EAcombined --exclude mappedSNPs_chr_${i}.txt --make-bed --out EAcombined_outchr_${i}"
done



```

then make relatedness matrix using gemma

two vertions:

1. combined version

2. each chromosome version


```
# in order to avoid error in GEMMA, better to change phenotype column in .fam file to other values
# checked the results when using 1 or 2 to replace -9, same results

for(i in 1:22){
fam=read.table(paste0("EAcombined_outchr_",i,".fam"),header=F)
fam$V6=1
write.table(fam,paste0("EAcombined_outchr_",i,".fam"),col.names=F,row.names=F,quote=F)
}

fam=read.table(paste0("EAcombined.fam"),header=F)
fam$V6=1
write.table(fam,paste0("EAcombined.fam"),col.names=F,row.names=F,quote=F)

qsub -cwd -b y -N EA -q mulan "gemma -bfile EAcombined -gk 2 -o EAcombined_gemma"

for i in `seq 1 22`
do qsub -cwd -b y -N EA -q mulan "gemma -bfile EAcombined_outchr_${i} -gk 2 -o EAcombined_outchr_${i}_gemma"
done


```






