
# EA
```
qsub -cwd -b y -N EA  "plink --noweb --bfile  EAcombined --geno 0.05 --maf 0.01  --make-bed --out EAcombined_geno"

[shanglu@camel4 relatedness]$ wc -l EAcombined.bim  
5420235 EAcombined.bim
[shanglu@camel4 relatedness]$ wc -l  EAcombined_geno.bim 
3818520 EAcombined_geno.bim
```

```
fam = read.table("/home/skardia_lab/clubhouse/research/projects/GENOA_AA_GE/preprocessing_zxu_March2017/Lulu_Mar2018/EUR/Roch_all_801.fam")
fam_geno = read.table("EAcombined_geno.fam")
# check
sum(as.character(fam$V2)==as.character(fam_geno$V1))
# 801
fam$V6=2
write.table(fam, file = "EAcombined_geno.fam",quote=F, col.names=F,row.names=F)
```

```

king -b EAcombined_geno.bed --kinship --prefix EAcombined_geno


```

```
#------------------------------------------------------------
# GENESIS for each trans effect considered as other chromosome
# -- kingship matrix
#------------------------------------------------------------
library(GENESIS)
library(GWASTools)
library(SNPRelate)

bed.fn <- paste0("EAcombined_geno.bed")
fam.fn <- paste0("EAcombined_geno.fam")
bim.fn <- paste0("EAcombined_geno.bim")
# Convert
snpgdsBED2GDS(bed.fn, fam.fn, bim.fn, paste0("EAcombined_geno.gds"))

family.ID <- as.character(read.table("EAcombined_geno.fam",header=F)$V1)
sample.ID <- as.character(read.table("EAcombined_geno.fam",header=F)$V2)

#Before running PCA, we use LD pruning to select a set of independent SNPs for analysis. We use the snpgdsLDpruning in the SNPRelate package, which returns a list of snp IDs.
gds <- snpgdsOpen(paste0("EAcombined_geno.gds"))

snpset <- snpgdsLDpruning(gds, method="corr", slide.max.bp=10e6, 
                          ld.threshold=sqrt(0.1), verbose=FALSE)
pruned <- unlist(snpset, use.names=FALSE)
print("length(pruned)")
print(length(pruned))
snpgdsClose(gds)
#
#> print(length(pruned))
#[1] 125354

save(pruned, file = "prunedEA.RData")
```

```

KINGmat=kingToMatrix(c("EAcombined_geno.kin0","EAcombined_geno.kin"), sample.include = NULL, thresh = 2^(-11/2), verbose = TRUE)

KINGmat[1:5,1:5]


snpgdsClose(gds)

#------------------------------------------------------------------
# We use the GWASTools package to create the 
# GenotypeData object needed by GENESIS.
#------------------------------------------------------------------

library(GWASTools)
current_geno <- GdsGenotypeReader(filename = paste0("EAcombined_geno.gds"))
# create a GenotypeData class object
current_genoData <- GenotypeData(current_geno)

mypcair <- pcair(current_genoData, kinobj = KINGmat, divobj = KINGmat, snp.include = pruned)
/*                 
> summary(mypcair)
Call:
.pcair(gdsobj = gdsobj, kinobj = kinobj, divobj = divobj, kin.thresh = kin.thresh, 
    div.thresh = div.thresh, unrel.set = unrel.set, sample.include = sample.include, 
    snp.include = snp.include, num.cores = num.cores, verbose = verbose)

PCA Method: PC-AiR 

Sample Size: 801 
Unrelated Set: 369 Samples 
Related Set: 432 Samples 

Kinship Threshold: 0.02209709 
Divergence Threshold: -0.02209709 

Principal Components Returned: 32 
Eigenvalues: 1.449 1.177 1.149 1.147 1.144 1.143 1.138 1.134 1.132 1.131 ...
SNPs Used: 125354 

*/
pdf("EA_pcair_PC1_PC2.pdf")
plot(mypcair)
dev.off()
# plot PCs 3 and 4
pdf("EA_pcair_PC3_PC4.pdf")
plot(mypcair, vx = 3, vy = 4)
dev.off()


save(mypcair, file = "mypcairEA.RData")

```

make relatedness matrix, although we do not use this relatedness matrix, we are using relatedness matrix from GEMMA.
```
current_genoData <- GenotypeBlockIterator(current_genoData, snpInclude=pruned)
mypcrelate <- pcrelate(current_genoData, pcs = mypcair$vectors[,1:5], training.set = mypcair$unrels)
                       
iids <- as.character(getScanID(current_genoData))
relatednessmat = pcrelateToMatrix(mypcrelate, sample.include = iids[1:801], thresh = 2^(-11/2), scaleKin = 2)                      
                       
save(relatednessmat, file = "relatednessmat.EA.RData")

```



# AA
```

qsub -cwd -b y -N AA  "plink --noweb --bfile  AAcombined --geno 0.05 --maf 0.01  --make-bed --out AAcombined_geno"

[shanglu@lesperan-n1 relatedness]$ wc -l AAcombined.bim 
9596068 AAcombined.bim
[shanglu@lesperan-n1 relatedness]$ wc -l AAcombined_geno.bim 
6432684 AAcombined_geno.bim

```


```
fam_jack=read.table("/home/skardia_lab/clubhouse/research/projects/GENOA_AA_GE/preprocessing_zxu_March2017/Lulu_Mar2018/Binary_files_for_measured_SNPs/Jack1032_used_for_eQTL/Jack_1032.fam")
fam_geno = read.table("AAcombined_geno.fam")

ind = match(as.character(fam_geno$V2), as.character(fam_jack$V2))
fam_jack = fam_jack[ind,]

fam_geno$V1 = fam_jack$V1
write.table(fam_geno, file = "AAcombined_geno.fam",quote=F, col.names=F,row.names=F)

```


```
king -b AAcombined_geno.bed --kinship --prefix AAcombined_geno
```

```
library(GENESIS)
library(GWASTools)
library(SNPRelate)

bed.fn <- paste0("AAcombined_geno.bed")
fam.fn <- paste0("AAcombined_geno.fam")
bim.fn <- paste0("AAcombined_geno.bim")
# Convert
snpgdsBED2GDS(bed.fn, fam.fn, bim.fn, paste0("AAcombined_geno.gds"))

family.ID <- as.character(read.table("AAcombined_geno.fam",header=F)$V1)
sample.ID <- as.character(read.table("AAcombined_geno.fam",header=F)$V2)

#Before running PCA, we use LD pruning to select a set of independent SNPs for analysis. We use the snpgdsLDpruning in the SNPRelate package, which returns a list of snp IDs.
gds <- snpgdsOpen(paste0("AAcombined_geno.gds"))

snpset <- snpgdsLDpruning(gds, method="corr", slide.max.bp=10e6, 
                          ld.threshold=sqrt(0.1), num.thread=10,verbose=FALSE)
                          
 # this step is too slow...
 
pruned <- unlist(snpset, use.names=FALSE)
print("length(pruned)")
print(length(pruned))
snpgdsClose(gds)

> print(length(pruned))
[1] 315721



save(pruned, file = "prunedAA.RData")

KINGmat=kingToMatrix(c("AAcombined_geno.kin0","AAcombined_geno.kin"), sample.include = NULL, thresh = 2^(-11/2), verbose = TRUE)

KINGmat[1:5,1:5]


snpgdsClose(gds)

#------------------------------------------------------------------
# We use the GWASTools package to create the 
# GenotypeData object needed by GENESIS.
#------------------------------------------------------------------

library(GWASTools)
current_geno <- GdsGenotypeReader(filename = paste0("AAcombined_geno.gds"))
# create a GenotypeData class object
current_genoData <- GenotypeData(current_geno)

mypcair <- pcair(current_genoData, kinobj = KINGmat, divobj = KINGmat, snp.include = pruned)
/*                 
> summary(mypcair)
> summary(mypcair)
Call:
.pcair(gdsobj = gdsobj, kinobj = kinobj, divobj = divobj, kin.thresh = kin.thresh, 
    div.thresh = div.thresh, unrel.set = unrel.set, sample.include = sample.include, 
    snp.include = snp.include, num.cores = num.cores, verbose = verbose)

PCA Method: PC-AiR 

Sample Size: 1032 
Unrelated Set: 461 Samples 
Related Set: 571 Samples 

Kinship Threshold: 0.02209709 
Divergence Threshold: -0.02209709 

Principal Components Returned: 32 
Eigenvalues: 2.411 1.22 1.175 1.168 1.166 1.159 1.156 1.15 1.148 1.146 ...
SNPs Used: 315721 
*/

pdf("AA_pcair_PC1_PC2.pdf")
plot(mypcair)
dev.off()
# plot PCs 3 and 4
pdf("AA_pcair_PC3_PC4.pdf")
plot(mypcair, vx = 3, vy = 4)
dev.off()


save(mypcair, file = "mypcairAA.RData")

current_genoData <- GenotypeBlockIterator(current_genoData, snpInclude=pruned)
mypcrelate <- pcrelate(current_genoData, pcs = mypcair$vectors[,1:5], training.set = mypcair$unrels)
                       
iids <- as.character(getScanID(current_genoData))
relatednessmat = pcrelateToMatrix(mypcrelate, sample.include = iids[1:1032], thresh = 2^(-11/2), scaleKin = 2)                      
         
                
save(relatednessmat, file = "relatednessmatAA.RData")       
```
