
# adjust for covariates and different number of PCs

```
#---------------------------------------------
# load meta:
#---------------------------------------------

pathEA="/net/mulan/home/shanglu/GENOA/data/EA"
load(paste0(pathEA,"/EA_meta_new_order.RData"))

#---------------------------------------------
# load PCs
#---------------------------------------------

load(paste0(pathEA,"/mypcairEA.RData"))


#---------------------------------------------
# load expression
#---------------------------------------------


load(paste0(pathEA,"/expression/expr_EA_progene_combat_qq.RData"))

load(paste0(pathEA,"/expression/expr_EA_proteingene.RData"))

colnames(expr_EA_progene_combat_qq) = rownames(expr_EA_proteingene)
rownames(expr_EA_progene_combat_qq) = colnames(expr_EA_proteingene)

meta=meta_new_order
```


# 0PC
```
#---------------------------------------------------
# adjust for covariates and PCs
#---------------------------------------------------

x=0
X=expr_EA_progene_combat_qq

>  dim(expr_EA_progene_combat_qq)
[1]  1032 17616

PCx_meta = meta
PCx_res=matrix(0,dim(X)[1],dim(X)[2])
PCx_res_qq=matrix(0,dim(X)[1],dim(X)[2])
for(j in 1:dim(X)[2]){
print(j)
PCx_res[,j]=lm(X[,j] ~ factor(gender) + age , data = PCx_meta)$residual
PCx_res_qq[,j]=qqnorm(PCx_res[,j],plot.it=F)$x
}

save(PCx_res_qq, file = paste0("PC",x,"_res_qq.RData"))

#---------------------------------------------------
# add permutation
#---------------------------------------------------



load( paste0("permuteID_EA.RData"))


PCx_res_qq_permute = list()
for(permu in 1:10){
	print(permu)
	PCx_res_qq_permute[[permu]] = PCx_res_qq[new_id[[permu]],]
	}
  
  
#---------------------------------------------------
# write phenotypes to fam file
#---------------------------------------------------

pathEAbed = "/net/mulan/home/shanglu/GENOA/data/EA/gene_bed"

load("gene_EA_anno_protein.RData")
load( "gene_EA_anno_order_correct.RData")
ind=match(gene_EA_anno_order$GENE, gene_EA_anno$GENE)


gene_annotation = gene_EA_anno_order
count=0
for(i in 1:22){
chr.num = sum(gene_annotation$chr==i)
print(paste0("chr ",i," read"))
print(count/length(gene_annotation$chr))
for( j in 1:chr.num){
count=count+1
fn = paste0(pathEAbed,"/chr_", i, "/chr_",i,"_gene_",j,".fam", sep = "")

if(file.exists(fn)){
fam_tmp = read.table(fn)
# col 6 to 17: 10 permutation for PC5
# col 18 to 27: 10 permutation for PC0
fam_tmp[,6+12] = c(expr_EA_progene_combat_qq[,ind[count]])
fam_tmp[,7+12] = c(PCx_res_qq[,ind[count]])
fam_tmp[,8+12] = c(PCx_res_qq_permute[[1]][,ind[count]])
fam_tmp[,9+12] = c(PCx_res_qq_permute[[2]][,ind[count]])
fam_tmp[,10+12] = c(PCx_res_qq_permute[[3]][,ind[count]])
fam_tmp[,11+12] = c(PCx_res_qq_permute[[4]][,ind[count]])
fam_tmp[,12+12] = c(PCx_res_qq_permute[[5]][,ind[count]])
fam_tmp[,13+12] = c(PCx_res_qq_permute[[6]][,ind[count]])
fam_tmp[,14+12] = c(PCx_res_qq_permute[[7]][,ind[count]])
fam_tmp[,15+12] = c(PCx_res_qq_permute[[8]][,ind[count]])
fam_tmp[,16+12] = c(PCx_res_qq_permute[[9]][,ind[count]])
fam_tmp[,17+12] = c(PCx_res_qq_permute[[10]][,ind[count]])
write.table(fam_tmp, file=fn, quote = F, row.names = F, col.names = F)
}
}
}



```

# PC10
```
#---------------------------------------------
# load meta:
#---------------------------------------------

pathEA="/net/mulan/home/shanglu/GENOA/data/EA"
load(paste0(pathEA,"/meta_EA.RData"))

#---------------------------------------------
# load PCs
#---------------------------------------------

load(paste0(pathEA,"/mypcairEA.RData"))

#---------------------------------------------
# load expression
#---------------------------------------------

load(paste0(pathEA,"/expression/expr_EA_progene_combat_qq.RData"))
load(paste0(pathEA,"/expression/expr_EA_proteingene.RData"))

colnames(expr_EA_progene_combat_qq) = rownames(expr_EA_proteingene)
rownames(expr_EA_progene_combat_qq) = colnames(expr_EA_proteingene)

#---------------------------------------------------
# adjust for covariates and PCs
#---------------------------------------------------

x = 10
X=expr_EA_progene_combat_qq
pcair_all_order = mypcair$vectors
pcair_PCx = pcair_all_order[,1:x]
func <- function(x) paste0("PC",x)
colnames(pcair_PCx) = unlist(lapply(c(1:x), func))
PCx_meta=cbind(meta,pcair_PCx)
PCx_res=matrix(0,dim(X)[1],dim(X)[2])
PCx_res_qq=matrix(0,dim(X)[1],dim(X)[2])
for(j in 1:dim(X)[2]){
print(j)
PCx_res[,j]=lm(X[,j] ~ factor(gender) + age + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10, data = PCx_meta)$residual
PCx_res_qq[,j]=qqnorm(PCx_res[,j],plot.it=F)$x
}
save(PCx_res_qq, file = paste0("PC",x,"_res_qq.RData"))

#---------------------------------------------------
# add permutation
#---------------------------------------------------
load( paste0("permuteID_EA.RData"))
PCx_res_qq_permute = list()
for(permu in 1:10){
	print(permu)
	PCx_res_qq_permute[[permu]] = PCx_res_qq[new_id[[permu]],]
	}
#---------------------------------------------------
# write phenotypes to fam file
#---------------------------------------------------

pathEAbed = "/net/mulan/home/shanglu/GENOA/data/EA/gene_bed"
load("gene_EA_anno_protein.RData")
gene_EA_anno_chr = gene_EA_anno[gene_EA_anno$chr==1,]
gene_EA_anno_chr$index_j = c(1:dim(gene_EA_anno_chr)[1])
gene_EA_anno_order = gene_EA_anno_chr
for(i in 2:22){
gene_EA_anno_chr = gene_EA_anno[gene_EA_anno$chr==i,]
gene_EA_anno_chr$index_j = c(1:dim(gene_EA_anno_chr)[1])
gene_EA_anno_order = rbind(gene_EA_anno_order,gene_EA_anno_chr) 
}
ind=match(gene_EA_anno_order$GENE, gene_EA_anno$GENE)
gene_annotation = gene_EA_anno_order

count=0
for(i in 1:22){
chr.num = sum(gene_annotation$chr==i)
print(paste0("chr ",i," read"))
print(count/length(gene_annotation$chr))
for( j in 1:chr.num){
count=count+1
fn = paste0(pathEAbed,"/chr_", i, "/chr_",i,"_gene_",j,".fam", sep = "")
if(file.exists(fn)){
fam_tmp = read.table(fn)
# col 6 to 17: 10 permutation for PC5
# col 18 to 27: 10 permutation for PC0
# col 30 to 41: 10 permutation for PC10
fam_tmp[,6+24] = c(expr_EA_progene_combat_qq[,ind[count]])
fam_tmp[,7+24] = c(PCx_res_qq[,ind[count]])
fam_tmp[,8+24] = c(PCx_res_qq_permute[[1]][,ind[count]])
fam_tmp[,9+24] = c(PCx_res_qq_permute[[2]][,ind[count]])
fam_tmp[,10+24] = c(PCx_res_qq_permute[[3]][,ind[count]])
fam_tmp[,11+24] = c(PCx_res_qq_permute[[4]][,ind[count]])
fam_tmp[,12+24] = c(PCx_res_qq_permute[[5]][,ind[count]])
fam_tmp[,13+24] = c(PCx_res_qq_permute[[6]][,ind[count]])
fam_tmp[,14+24] = c(PCx_res_qq_permute[[7]][,ind[count]])
fam_tmp[,15+24] = c(PCx_res_qq_permute[[8]][,ind[count]])
fam_tmp[,16+24] = c(PCx_res_qq_permute[[9]][,ind[count]])
fam_tmp[,17+24] = c(PCx_res_qq_permute[[10]][,ind[count]])
write.table(fam_tmp, file=fn, quote = F, row.names = F, col.names = F)
}
}
}



```

# PC15
```
#---------------------------------------------
# load meta:
#---------------------------------------------

pathEA="/net/mulan/home/shanglu/GENOA/data/EA"
load(paste0(pathEA,"/meta_EA.RData"))

#---------------------------------------------
# load PCs
#---------------------------------------------

load(paste0(pathEA,"/mypcairEA.RData"))

#---------------------------------------------
# load expression
#---------------------------------------------

load(paste0(pathEA,"/expression/expr_EA_progene_combat_qq.RData"))
load(paste0(pathEA,"/expression/expr_EA_proteingene.RData"))

colnames(expr_EA_progene_combat_qq) = rownames(expr_EA_proteingene)
rownames(expr_EA_progene_combat_qq) = colnames(expr_EA_proteingene)

#---------------------------------------------------
# adjust for covariates and PCs
#---------------------------------------------------

x = 15
X=expr_EA_progene_combat_qq
pcair_all_order = mypcair$vectors
pcair_PCx = pcair_all_order[,1:x]
func <- function(x) paste0("PC",x)
colnames(pcair_PCx) = unlist(lapply(c(1:x), func))
PCx_meta=cbind(meta,pcair_PCx)
PCx_res=matrix(0,dim(X)[1],dim(X)[2])
PCx_res_qq=matrix(0,dim(X)[1],dim(X)[2])
for(j in 1:dim(X)[2]){
print(j)
PCx_res[,j]=lm(X[,j] ~ factor(gender) + age + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + PC11 + PC12 + PC13 + PC14 + PC15, data = PCx_meta)$residual
PCx_res_qq[,j]=qqnorm(PCx_res[,j],plot.it=F)$x
}
save(PCx_res_qq, file = paste0("PC",x,"_res_qq.RData"))

#---------------------------------------------------
# add permutation
#---------------------------------------------------
load( paste0("permuteID_EA.RData"))
PCx_res_qq_permute = list()
for(permu in 1:10){
	print(permu)
	PCx_res_qq_permute[[permu]] = PCx_res_qq[new_id[[permu]],]
	}
#---------------------------------------------------
# write phenotypes to fam file
#---------------------------------------------------

pathEAbed = "/net/mulan/home/shanglu/GENOA/data/EA/gene_bed"
load("gene_EA_anno_protein.RData")
gene_EA_anno_chr = gene_EA_anno[gene_EA_anno$chr==1,]
gene_EA_anno_chr$index_j = c(1:dim(gene_EA_anno_chr)[1])
gene_EA_anno_order = gene_EA_anno_chr
for(i in 2:22){
gene_EA_anno_chr = gene_EA_anno[gene_EA_anno$chr==i,]
gene_EA_anno_chr$index_j = c(1:dim(gene_EA_anno_chr)[1])
gene_EA_anno_order = rbind(gene_EA_anno_order,gene_EA_anno_chr) 
}
ind=match(gene_EA_anno_order$GENE, gene_EA_anno$GENE)
gene_annotation = gene_EA_anno_order

count=0
for(i in 1:22){
chr.num = sum(gene_annotation$chr==i)
print(paste0("chr ",i," read"))
print(count/length(gene_annotation$chr))
for( j in 1:chr.num){
count=count+1
fn = paste0(pathEAbed,"/chr_", i, "/chr_",i,"_gene_",j,".fam", sep = "")
if(file.exists(fn)){
fam_tmp = read.table(fn)
# col 6 to 17: 10 permutation for PC5
# col 18 to 27: 10 permutation for PC0
# col 30 to 41: 10 permutation for PC10
fam_tmp[,6+36] = c(expr_EA_progene_combat_qq[,ind[count]])
fam_tmp[,7+36] = c(PCx_res_qq[,ind[count]])
fam_tmp[,8+36] = c(PCx_res_qq_permute[[1]][,ind[count]])
fam_tmp[,9+36] = c(PCx_res_qq_permute[[2]][,ind[count]])
fam_tmp[,10+36] = c(PCx_res_qq_permute[[3]][,ind[count]])
fam_tmp[,11+36] = c(PCx_res_qq_permute[[4]][,ind[count]])
fam_tmp[,12+36] = c(PCx_res_qq_permute[[5]][,ind[count]])
fam_tmp[,13+36] = c(PCx_res_qq_permute[[6]][,ind[count]])
fam_tmp[,14+36] = c(PCx_res_qq_permute[[7]][,ind[count]])
fam_tmp[,15+36] = c(PCx_res_qq_permute[[8]][,ind[count]])
fam_tmp[,16+36] = c(PCx_res_qq_permute[[9]][,ind[count]])
fam_tmp[,17+36] = c(PCx_res_qq_permute[[10]][,ind[count]])
write.table(fam_tmp, file=fn, quote = F, row.names = F, col.names = F)
}
}
}



```


# PC20
```
#---------------------------------------------
# load meta:
#---------------------------------------------

pathEA="/net/mulan/home/shanglu/GENOA/data/EA"
load(paste0(pathEA,"/meta_EA.RData"))

#---------------------------------------------
# load PCs
#---------------------------------------------

load(paste0(pathEA,"/mypcairEA.RData"))

#---------------------------------------------
# load expression
#---------------------------------------------

load(paste0(pathEA,"/expression/expr_EA_progene_combat_qq.RData"))
load(paste0(pathEA,"/expression/expr_EA_proteingene.RData"))

colnames(expr_EA_progene_combat_qq) = rownames(expr_EA_proteingene)
rownames(expr_EA_progene_combat_qq) = colnames(expr_EA_proteingene)

#---------------------------------------------------
# adjust for covariates and PCs
#---------------------------------------------------

x = 20
X=expr_EA_progene_combat_qq
pcair_all_order = mypcair$vectors
pcair_PCx = pcair_all_order[,1:x]
func <- function(x) paste0("PC",x)
colnames(pcair_PCx) = unlist(lapply(c(1:x), func))
PCx_meta=cbind(meta,pcair_PCx)
PCx_res=matrix(0,dim(X)[1],dim(X)[2])
PCx_res_qq=matrix(0,dim(X)[1],dim(X)[2])
for(j in 1:dim(X)[2]){
print(j)
PCx_res[,j]=lm(X[,j] ~ factor(gender) + age + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + PC11 + PC12 + PC13 + PC14 + PC15 + PC16 + PC17 + PC18 + PC19 + PC20, data = PCx_meta)$residual
PCx_res_qq[,j]=qqnorm(PCx_res[,j],plot.it=F)$x
}
save(PCx_res_qq, file = paste0("PC",x,"_res_qq.RData"))

#---------------------------------------------------
# add permutation
#---------------------------------------------------
load( paste0("permuteID_EA.RData"))
PCx_res_qq_permute = list()
for(permu in 1:10){
	print(permu)
	PCx_res_qq_permute[[permu]] = PCx_res_qq[new_id[[permu]],]
	}
#---------------------------------------------------
# write phenotypes to fam file
#---------------------------------------------------

pathEAbed = "/net/mulan/home/shanglu/GENOA/data/EA/gene_bed"
load("gene_EA_anno_protein.RData")
gene_EA_anno_chr = gene_EA_anno[gene_EA_anno$chr==1,]
gene_EA_anno_chr$index_j = c(1:dim(gene_EA_anno_chr)[1])
gene_EA_anno_order = gene_EA_anno_chr
for(i in 2:22){
gene_EA_anno_chr = gene_EA_anno[gene_EA_anno$chr==i,]
gene_EA_anno_chr$index_j = c(1:dim(gene_EA_anno_chr)[1])
gene_EA_anno_order = rbind(gene_EA_anno_order,gene_EA_anno_chr) 
}
ind=match(gene_EA_anno_order$GENE, gene_EA_anno$GENE)
gene_annotation = gene_EA_anno_order

count=0
for(i in 1:22){
chr.num = sum(gene_annotation$chr==i)
print(paste0("chr ",i," read"))
print(count/length(gene_annotation$chr))
for( j in 1:chr.num){
count=count+1
fn = paste0(pathEAbed,"/chr_", i, "/chr_",i,"_gene_",j,".fam", sep = "")
if(file.exists(fn)){
fam_tmp = read.table(fn)
# col 6 to 17: 10 permutation for PC5
# col 18 to 27: 10 permutation for PC0
# col 30 to 41: 10 permutation for PC10
fam_tmp[,6+48] = c(expr_EA_progene_combat_qq[,ind[count]])
fam_tmp[,7+48] = c(PCx_res_qq[,ind[count]])
fam_tmp[,8+48] = c(PCx_res_qq_permute[[1]][,ind[count]])
fam_tmp[,9+48] = c(PCx_res_qq_permute[[2]][,ind[count]])
fam_tmp[,10+48] = c(PCx_res_qq_permute[[3]][,ind[count]])
fam_tmp[,11+48] = c(PCx_res_qq_permute[[4]][,ind[count]])
fam_tmp[,12+48] = c(PCx_res_qq_permute[[5]][,ind[count]])
fam_tmp[,13+48] = c(PCx_res_qq_permute[[6]][,ind[count]])
fam_tmp[,14+48] = c(PCx_res_qq_permute[[7]][,ind[count]])
fam_tmp[,15+48] = c(PCx_res_qq_permute[[8]][,ind[count]])
fam_tmp[,16+48] = c(PCx_res_qq_permute[[9]][,ind[count]])
fam_tmp[,17+48] = c(PCx_res_qq_permute[[10]][,ind[count]])
write.table(fam_tmp, file=fn, quote = F, row.names = F, col.names = F)
}
}
}



```

# EA_PC.sh
```
#!/bin/bash
#SBATCH --job-name=EA_PC
#SBATCH --array=1-1056%200
#SBATCH --cpus-per-task=2
#SBATCH --output=/net/mulan/home/shanglu/GENOA/data/EA/out/EA_PC%a.out
#SBATCH --error=/net/mulan/home/shanglu/GENOA/data/EA/err/EA_PC%a.err
#SBATCH --workdir=/net/mulan/home/shanglu/GENOA/data/EA
#SBATCH --partition=mulan
#SBATCH --mem-per-cpu=8000MB


bash

let k=0
pathbed=/net/mulan/home/shanglu/GENOA/data/EA/gene_bed
pathGfile=//net/mulan/home/shanglu/GENOA/data/EA/Gene_file
pathRelatedness=/net/mulan/home/shanglu/GENOA/data/EA/filtered_info/output
GEMMA=/net/mulan/home/shanglu/GENOA/src/gemma

for pheno in `seq 13 60`
do
for i in `seq 1 22`
do
let k=${k}+1
if [ ${k} -eq ${SLURM_ARRAY_TASK_ID} ]; then
gnum=`cat ${pathGfile}/chr${i}.num`
for j in `seq 1 ${gnum}`
do
${GEMMA} -bfile ${pathbed}/chr_${i}/chr_${i}_gene_${j} -maf 0.05 -r2 0.9999999 -hwe 0 -n ${pheno} -k ${pathRelatedness}/EAcombined_gemma.sXX.txt  -lmm 1 -o /chr_${i}/chr_${i}_gene_${j}_pheno_${pheno}
done 
fi
done
done




```


```
PCnum = 0

# load gene annotation 
path_EA="/net/mulan/home/shanglu/GENOA/data/EA"
path_EA_result="/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping/result"
path_EA_output="/net/mulan/home/shanglu/GENOA/data/EA/output/"

load(paste0(path_EA,"/gene_EA_anno_protein.RData"))

all_min_p = list()
count = 0
for(pheno in 14:24){
count = count + 1
print(pheno)
min_p_pheno = list()
for(chr in 1:22){

min_p_chr = c()
load(paste0(path_EA_result,"/EA_eqtl_res_chr_",chr,"pheno",pheno,".RData"))

for(g in 1:length(res)){
  if(length(dim(res[[g]])[1])!=0 ){
    if(dim(res[[g]])[1]!=0){
      min_p_chr[g] = res[[g]]$p_wald[which.min(res[[g]]$p_wald)]
      }else{
      min_p_chr[g]=NA
    }
  }else{
      min_p_chr[g]=NA
}
}
min_p_pheno[[chr]] = min_p_chr
}
all_min_p[[count]] = unlist(min_p_pheno)
}

save(all_min_p, file = paste0("/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping/all_min_p_EA_PC",PCnum,".RData"))



EAnull = which(is.na(all_min_p[[1]]))

min_p_EA = all_min_p[[1]][-EAnull]
order_p_EA = min_p_EA[order(min_p_EA)]
min_p_permu_EA = list()
for(i in 1:10){ 
min_p_permu_EA[[i]] = all_min_p[[i+1]][-EAnull]
}
min_p_EA_permu_mat = matrix(unlist(min_p_permu_EA), ncol = length(order_p_EA), byrow = TRUE) 

count = 0 
num_real = NULL 
FDR_EA = NULL 
for(threshold in order_p_EA){ 
count = count + 1 
num_real[count] = sum(min_p_EA<=threshold) 
FDR_EA[count] = mean(apply(min_p_EA_permu_mat, 1, function(x) sum(x<=threshold)/num_real[count])) }


max(which(FDR_EA<0.01))
order_p_EA[ max(which(FDR_EA<0.01))]
max(which(FDR_EA<0.05))
order_p_EA[ max(which(FDR_EA<0.05))]
max(which(FDR_EA<0.1))
order_p_EA[ max(which(FDR_EA<0.1))]
max(which(FDR_EA<0.2))
order_p_EA[ max(which(FDR_EA<0.2))]

> max(which(FDR_EA<0.01))
[1] 3510
> order_p_EA[ max(which(FDR_EA<0.01))]
[1] 1.948518e-05
> max(which(FDR_EA<0.05))
[1] 4481
> order_p_EA[ max(which(FDR_EA<0.05))]
[1] 0.0001436664
> max(which(FDR_EA<0.1))
[1] 5181
> order_p_EA[ max(which(FDR_EA<0.1))]
[1] 0.0003518576
> max(which(FDR_EA<0.2))
[1] 6289
> order_p_EA[ max(which(FDR_EA<0.2))]
[1] 0.0009555247

```

```
PCnum = 10

# load gene annotation 
path_EA="/net/mulan/home/shanglu/GENOA/data/EA"
path_EA_result="/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping/result"
path_EA_output="/net/mulan/home/shanglu/GENOA/data/EA/output/"

load(paste0(path_EA,"/gene_EA_anno_protein.RData"))

all_min_p = list()
count = 0
for(pheno in 26:36){
count = count + 1
print(pheno)
min_p_pheno = list()
for(chr in 1:22){

min_p_chr = c()
load(paste0(path_EA_result,"/EA_eqtl_res_chr_",chr,"pheno",pheno,".RData"))

for(g in 1:length(res)){
  if(length(dim(res[[g]])[1])!=0 ){
    if(dim(res[[g]])[1]!=0){
      min_p_chr[g] = res[[g]]$p_wald[which.min(res[[g]]$p_wald)]
      }else{
      min_p_chr[g]=NA
    }
  }else{
      min_p_chr[g]=NA
}
}
min_p_pheno[[chr]] = min_p_chr
}
all_min_p[[count]] = unlist(min_p_pheno)
}

save(all_min_p, file = paste0("/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping/all_min_p_EA_PC",PCnum,".RData"))



EAnull = which(is.na(all_min_p[[1]]))

min_p_EA = all_min_p[[1]][-EAnull]
order_p_EA = min_p_EA[order(min_p_EA)]
min_p_permu_EA = list()
for(i in 1:10){ 
min_p_permu_EA[[i]] = all_min_p[[i+1]][-EAnull]
}
min_p_EA_permu_mat = matrix(unlist(min_p_permu_EA), ncol = length(order_p_EA), byrow = TRUE) 

count = 0 
num_real = NULL 
FDR_EA = NULL 
for(threshold in order_p_EA){ 
count = count + 1 
num_real[count] = sum(min_p_EA<=threshold) 
FDR_EA[count] = mean(apply(min_p_EA_permu_mat, 1, function(x) sum(x<=threshold)/num_real[count])) }


max(which(FDR_EA<0.01))
order_p_EA[ max(which(FDR_EA<0.01))]
max(which(FDR_EA<0.05))
order_p_EA[ max(which(FDR_EA<0.05))]
max(which(FDR_EA<0.1))
order_p_EA[ max(which(FDR_EA<0.1))]
max(which(FDR_EA<0.2))
order_p_EA[ max(which(FDR_EA<0.2))]


> max(which(FDR_EA<0.01))
[1] 3515
> order_p_EA[ max(which(FDR_EA<0.01))]
[1] 2.165135e-05
> max(which(FDR_EA<0.05))
[1] 4365
> order_p_EA[ max(which(FDR_EA<0.05))]
[1] 0.0001374437
> max(which(FDR_EA<0.1))
[1] 5065
> order_p_EA[ max(which(FDR_EA<0.1))]
[1] 0.000342987
> max(which(FDR_EA<0.2))
[1] 6227
> order_p_EA[ max(which(FDR_EA<0.2))]
[1] 0.0009546973


```


```
PCnum = 15

# load gene annotation 
path_EA="/net/mulan/home/shanglu/GENOA/data/EA"
path_EA_result="/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping/result"
path_EA_output="/net/mulan/home/shanglu/GENOA/data/EA/output/"

load(paste0(path_EA,"/gene_EA_anno_protein.RData"))

all_min_p = list()
count = 0
for(pheno in 38:48){
count = count + 1
print(pheno)
min_p_pheno = list()
for(chr in 1:22){

min_p_chr = c()
load(paste0(path_EA_result,"/EA_eqtl_res_chr_",chr,"pheno",pheno,".RData"))

for(g in 1:length(res)){
  if(length(dim(res[[g]])[1])!=0 ){
    if(dim(res[[g]])[1]!=0){
      min_p_chr[g] = res[[g]]$p_wald[which.min(res[[g]]$p_wald)]
      }else{
      min_p_chr[g]=NA
    }
  }else{
      min_p_chr[g]=NA
}
}
min_p_pheno[[chr]] = min_p_chr
}
all_min_p[[count]] = unlist(min_p_pheno)
}

save(all_min_p, file = paste0("/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping/all_min_p_EA_PC",PCnum,".RData"))



EAnull = which(is.na(all_min_p[[1]]))

min_p_EA = all_min_p[[1]][-EAnull]
order_p_EA = min_p_EA[order(min_p_EA)]
min_p_permu_EA = list()
for(i in 1:10){ 
min_p_permu_EA[[i]] = all_min_p[[i+1]][-EAnull]
}
min_p_EA_permu_mat = matrix(unlist(min_p_permu_EA), ncol = length(order_p_EA), byrow = TRUE) 

count = 0 
num_real = NULL 
FDR_EA = NULL 
for(threshold in order_p_EA){ 
count = count + 1 
num_real[count] = sum(min_p_EA<=threshold) 
FDR_EA[count] = mean(apply(min_p_EA_permu_mat, 1, function(x) sum(x<=threshold)/num_real[count])) }


max(which(FDR_EA<0.01))
order_p_EA[ max(which(FDR_EA<0.01))]
max(which(FDR_EA<0.05))
order_p_EA[ max(which(FDR_EA<0.05))]
max(which(FDR_EA<0.1))
order_p_EA[ max(which(FDR_EA<0.1))]
max(which(FDR_EA<0.2))
order_p_EA[ max(which(FDR_EA<0.2))]

> max(which(FDR_EA<0.01))
[1] 3452
> order_p_EA[ max(which(FDR_EA<0.01))]
[1] 2.009308e-05
> max(which(FDR_EA<0.05))
[1] 4333
> order_p_EA[ max(which(FDR_EA<0.05))]
[1] 0.0001358554
> max(which(FDR_EA<0.1))
[1] 5026
> order_p_EA[ max(which(FDR_EA<0.1))]
[1] 0.0003404796
> max(which(FDR_EA<0.2))
[1] 6153
> order_p_EA[ max(which(FDR_EA<0.2))]
[1] 0.0009505634

```

```
PCnum = 20

# load gene annotation 
path_EA="/net/mulan/home/shanglu/GENOA/data/EA"
path_EA_result="/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping/result"
path_EA_output="/net/mulan/home/shanglu/GENOA/data/EA/output/"

load(paste0(path_EA,"/gene_EA_anno_protein.RData"))

all_min_p = list()
count = 0
for(pheno in 50:60){
count = count + 1
print(pheno)
min_p_pheno = list()
for(chr in 1:22){

min_p_chr = c()
load(paste0(path_EA_result,"/EA_eqtl_res_chr_",chr,"pheno",pheno,".RData"))

for(g in 1:length(res)){
  if(length(dim(res[[g]])[1])!=0 ){
    if(dim(res[[g]])[1]!=0){
      min_p_chr[g] = res[[g]]$p_wald[which.min(res[[g]]$p_wald)]
      }else{
      min_p_chr[g]=NA
    }
  }else{
      min_p_chr[g]=NA
}
}
min_p_pheno[[chr]] = min_p_chr
}
all_min_p[[count]] = unlist(min_p_pheno)
}

save(all_min_p, file = paste0("/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping/all_min_p_EA_PC",PCnum,".RData"))



EAnull = which(is.na(all_min_p[[1]]))

min_p_EA = all_min_p[[1]][-EAnull]
order_p_EA = min_p_EA[order(min_p_EA)]
min_p_permu_EA = list()
for(i in 1:10){ 
min_p_permu_EA[[i]] = all_min_p[[i+1]][-EAnull]
}
min_p_EA_permu_mat = matrix(unlist(min_p_permu_EA), ncol = length(order_p_EA), byrow = TRUE) 

count = 0 
num_real = NULL 
FDR_EA = NULL 
for(threshold in order_p_EA){ 
count = count + 1 
num_real[count] = sum(min_p_EA<=threshold) 
FDR_EA[count] = mean(apply(min_p_EA_permu_mat, 1, function(x) sum(x<=threshold)/num_real[count])) }


max(which(FDR_EA<0.01))
order_p_EA[ max(which(FDR_EA<0.01))]
max(which(FDR_EA<0.05))
order_p_EA[ max(which(FDR_EA<0.05))]
max(which(FDR_EA<0.1))
order_p_EA[ max(which(FDR_EA<0.1))]
max(which(FDR_EA<0.2))
order_p_EA[ max(which(FDR_EA<0.2))]

> max(which(FDR_EA<0.01))
[1] 3416
> order_p_EA[ max(which(FDR_EA<0.01))]
[1] 2.00205e-05
> max(which(FDR_EA<0.05))
[1] 4281
> order_p_EA[ max(which(FDR_EA<0.05))]
[1] 0.0001345381
> max(which(FDR_EA<0.1))
[1] 4967
> order_p_EA[ max(which(FDR_EA<0.1))]
[1] 0.0003355586
> max(which(FDR_EA<0.2))
[1] 6072
> order_p_EA[ max(which(FDR_EA<0.2))]
[1] 0.000930658

```


# make full table
```
pheno=14
pheno=26

pheno=38
pheno=50

for(i in 1:22){
print(i)
chr=NULL
rs=NULL
ps=NULL
n_miss=NULL
allele1=NULL
allele0=NULL
af=NULL
beta=NULL
se=NULL
l_remle=NULL
p_wald=NULL
path_EA_result="/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping/result"
EA_full_table = data.frame(chr,rs ,ps,n_miss,allele1,allele0,af,beta,se,l_remle, p_wald )
load("/net/mulan/home/shanglu/GENOA/data/EA/gene_EA_anno_order_correct.RData")

  print(i)
  load(paste0(path_EA_result,"/EA_eqtl_res_chr_",i,"pheno",pheno,".RData"))
  for(j in 1:length(res)){
    if(sum(dim(res[[j]])[1])!=0){
    gene_EA_anno_order_chr = gene_EA_anno_order[gene_EA_anno_order$chr==i,]
    res[[j]]$GENE = gene_EA_anno_order_chr$GENE[j]
    EA_full_table = rbind(EA_full_table, res[[j]])
    }
   }

save(EA_full_table, file = paste0("/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping/EA_full_table_chr_",i,"_pheno",pheno,".RData"))

}



```

# figure
```
PC_eGene = c(4556,4490,4463,4419,4384)
PC_eSNP = c(376579,372377,368936,364137,360979)


library(RColorBrewer)
COL = brewer.pal(9,"Set1");
PC = c(0,  5, 10, 15, 20)
nG = PC_eGene
nPC = length(PC)
pdf("eGene_vs_PC_EA.pdf", 4, 3)
plot(0, 0, xlim=c(0, 20), ylim=c(0, 10000), type="n", xlab="Number of Genotype PCs", ylab="Number of eGenes", main="",cex.lab=1.2)
abline(v=seq(0, 20, 5), lty=2, col="lightgrey")
abline(h=seq(0, 10000, 1000), lty=2, col="lightgrey")
points(PC, nG, type="o", col=COL[1], pch=20, lwd=2)
#legend("bottomright", legend=c("eGenes vs PCs at 5% FDR in European American"), fill=c(COL[1], "grey"), bg="white", cex=1.2)
dev.off()


#  y: #eQTLs, x: #PCs
PC = c(0,  5, 10, 15, 20)
nG = PC_eSNP
nPC = length(PC)
pdf("eSNP_vs_PC_EA.pdf", 4, 3)
plot(0, 0, xlim=c(0, 20), ylim=c(0, 600000), type="n", xlab="Number of Genotype PCs", ylab="Number of eSNPs", main="",cex.lab=1.2)
abline(v=seq(0, 20, 5), lty=2, col="lightgrey")
abline(h=seq(0, 600000, 50000), lty=2, col="lightgrey")
points(PC, nG, type="o", col=COL[2], pch=20, lwd=2)
#legend("bottomright", legend=c("eQTLs vs PCs at 5% FDR in European American"), fill=c(COL[1], "grey"), bg="white", cex=1.2)
dev.off()
```

