
# adjust for covariates and different number of PCs

```
#---------------------------------------------
# load meta:
#---------------------------------------------

pathAA="/net/mulan/home/shanglu/GENOA/data/AA"
load(paste0(pathAA,"/meta_AA.RData"))

#---------------------------------------------
# load PCs
#---------------------------------------------

load(paste0(pathAA,"/mypcairAA.RData"))


#---------------------------------------------
# load expression
#---------------------------------------------


load(paste0(pathAA,"/expression/expr_AA_progene_combat_qq.RData"))

load(paste0(pathAA,"/expression/expr_AA_proteingene.RData"))

colnames(expr_AA_progene_combat_qq) = rownames(expr_AA_proteingene)
rownames(expr_AA_progene_combat_qq) = colnames(expr_AA_proteingene)

```


# 0PC
```
#---------------------------------------------------
# adjust for covariates and PCs
#---------------------------------------------------

x=0
X=expr_AA_progene_combat_qq

>  dim(expr_AA_progene_combat_qq)
[1]  1032 17616

PCx_meta = meta
PCx_res=matrix(0,dim(X)[1],dim(X)[2])
PCx_res_qq=matrix(0,dim(X)[1],dim(X)[2])
for(j in 1:dim(X)[2]){
print(j)
PCx_res[,j]=lm(X[,j] ~ factor(gender) + age , data = PCx_meta)$residual
PCx_res_qq[,j]=qqnorm(PCx_res[,j],plot.it=F)$x
}

save(PCx_res_qq, file = paste0("PC",x,"_res_qq.RData"))

#---------------------------------------------------
# add permutation
#---------------------------------------------------



load( paste0("permuteID_AA.RData"))


PCx_res_qq_permute = list()
for(permu in 1:10){
	print(permu)
	PCx_res_qq_permute[[permu]] = PCx_res_qq[new_id[[permu]],]
	}
  
  
#---------------------------------------------------
# write phenotypes to fam file
#---------------------------------------------------

pathAAbed = "/net/mulan/home/shanglu/GENOA/data/AA/gene_bed"

load("gene_AA_anno_protein.RData")
load( "gene_AA_anno_order_correct.RData")
ind=match(gene_AA_anno_order$GENE, gene_AA_anno$GENE)


gene_annotation = gene_AA_anno_order
count=0
for(i in 1:22){
chr.num = sum(gene_annotation$chr==i)
print(paste0("chr ",i," read"))
print(count/length(gene_annotation$chr))
for( j in 1:chr.num){
count=count+1
fn = paste0(pathAAbed,"/chr_", i, "/chr_",i,"_gene_",j,".fam", sep = "")

if(file.exists(fn)){
fam_tmp = read.table(fn)
# col 6 to 17: 10 permutation for PC5
# col 18 to 27: 10 permutation for PC0
fam_tmp[,6+12] = c(expr_AA_progene_combat_qq[,ind[count]])
fam_tmp[,7+12] = c(PCx_res_qq[,ind[count]])
fam_tmp[,8+12] = c(PCx_res_qq_permute[[1]][,ind[count]])
fam_tmp[,9+12] = c(PCx_res_qq_permute[[2]][,ind[count]])
fam_tmp[,10+12] = c(PCx_res_qq_permute[[3]][,ind[count]])
fam_tmp[,11+12] = c(PCx_res_qq_permute[[4]][,ind[count]])
fam_tmp[,12+12] = c(PCx_res_qq_permute[[5]][,ind[count]])
fam_tmp[,13+12] = c(PCx_res_qq_permute[[6]][,ind[count]])
fam_tmp[,14+12] = c(PCx_res_qq_permute[[7]][,ind[count]])
fam_tmp[,15+12] = c(PCx_res_qq_permute[[8]][,ind[count]])
fam_tmp[,16+12] = c(PCx_res_qq_permute[[9]][,ind[count]])
fam_tmp[,17+12] = c(PCx_res_qq_permute[[10]][,ind[count]])
write.table(fam_tmp, file=fn, quote = F, row.names = F, col.names = F)
}
}
}



```

# PC10
```
#---------------------------------------------
# load meta:
#---------------------------------------------

pathAA="/net/mulan/home/shanglu/GENOA/data/AA"
load(paste0(pathAA,"/meta_AA.RData"))

#---------------------------------------------
# load PCs
#---------------------------------------------

load(paste0(pathAA,"/mypcairAA.RData"))

#---------------------------------------------
# load expression
#---------------------------------------------

load(paste0(pathAA,"/expression/expr_AA_progene_combat_qq.RData"))
load(paste0(pathAA,"/expression/expr_AA_proteingene.RData"))

colnames(expr_AA_progene_combat_qq) = rownames(expr_AA_proteingene)
rownames(expr_AA_progene_combat_qq) = colnames(expr_AA_proteingene)

#---------------------------------------------------
# adjust for covariates and PCs
#---------------------------------------------------

x = 10
X=expr_AA_progene_combat_qq
pcair_all_order = mypcair$vectors
pcair_PCx = pcair_all_order[,1:x]
func <- function(x) paste0("PC",x)
colnames(pcair_PCx) = unlist(lapply(c(1:x), func))
PCx_meta=cbind(meta,pcair_PCx)
PCx_res=matrix(0,dim(X)[1],dim(X)[2])
PCx_res_qq=matrix(0,dim(X)[1],dim(X)[2])
for(j in 1:dim(X)[2]){
print(j)
PCx_res[,j]=lm(X[,j] ~ factor(gender) + age + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10, data = PCx_meta)$residual
PCx_res_qq[,j]=qqnorm(PCx_res[,j],plot.it=F)$x
}
save(PCx_res_qq, file = paste0("PC",x,"_res_qq.RData"))

#---------------------------------------------------
# add permutation
#---------------------------------------------------
load( paste0("permuteID_AA.RData"))
PCx_res_qq_permute = list()
for(permu in 1:10){
	print(permu)
	PCx_res_qq_permute[[permu]] = PCx_res_qq[new_id[[permu]],]
	}
#---------------------------------------------------
# write phenotypes to fam file
#---------------------------------------------------

pathAAbed = "/net/mulan/home/shanglu/GENOA/data/AA/gene_bed"
load("gene_AA_anno_protein.RData")
gene_AA_anno_chr = gene_AA_anno[gene_AA_anno$chr==1,]
gene_AA_anno_chr$index_j = c(1:dim(gene_AA_anno_chr)[1])
gene_AA_anno_order = gene_AA_anno_chr
for(i in 2:22){
gene_AA_anno_chr = gene_AA_anno[gene_AA_anno$chr==i,]
gene_AA_anno_chr$index_j = c(1:dim(gene_AA_anno_chr)[1])
gene_AA_anno_order = rbind(gene_AA_anno_order,gene_AA_anno_chr) 
}
ind=match(gene_AA_anno_order$GENE, gene_AA_anno$GENE)
gene_annotation = gene_AA_anno_order

count=0
for(i in 1:22){
chr.num = sum(gene_annotation$chr==i)
print(paste0("chr ",i," read"))
print(count/length(gene_annotation$chr))
for( j in 1:chr.num){
count=count+1
fn = paste0(pathAAbed,"/chr_", i, "/chr_",i,"_gene_",j,".fam", sep = "")
if(file.exists(fn)){
fam_tmp = read.table(fn)
# col 6 to 17: 10 permutation for PC5
# col 18 to 27: 10 permutation for PC0
# col 30 to 41: 10 permutation for PC10
fam_tmp[,6+24] = c(expr_AA_progene_combat_qq[,ind[count]])
fam_tmp[,7+24] = c(PCx_res_qq[,ind[count]])
fam_tmp[,8+24] = c(PCx_res_qq_permute[[1]][,ind[count]])
fam_tmp[,9+24] = c(PCx_res_qq_permute[[2]][,ind[count]])
fam_tmp[,10+24] = c(PCx_res_qq_permute[[3]][,ind[count]])
fam_tmp[,11+24] = c(PCx_res_qq_permute[[4]][,ind[count]])
fam_tmp[,12+24] = c(PCx_res_qq_permute[[5]][,ind[count]])
fam_tmp[,13+24] = c(PCx_res_qq_permute[[6]][,ind[count]])
fam_tmp[,14+24] = c(PCx_res_qq_permute[[7]][,ind[count]])
fam_tmp[,15+24] = c(PCx_res_qq_permute[[8]][,ind[count]])
fam_tmp[,16+24] = c(PCx_res_qq_permute[[9]][,ind[count]])
fam_tmp[,17+24] = c(PCx_res_qq_permute[[10]][,ind[count]])
write.table(fam_tmp, file=fn, quote = F, row.names = F, col.names = F)
}
}
}



```

# PC15
```
#---------------------------------------------
# load meta:
#---------------------------------------------

pathAA="/net/mulan/home/shanglu/GENOA/data/AA"
load(paste0(pathAA,"/meta_AA.RData"))

#---------------------------------------------
# load PCs
#---------------------------------------------

load(paste0(pathAA,"/mypcairAA.RData"))

#---------------------------------------------
# load expression
#---------------------------------------------

load(paste0(pathAA,"/expression/expr_AA_progene_combat_qq.RData"))
load(paste0(pathAA,"/expression/expr_AA_proteingene.RData"))

colnames(expr_AA_progene_combat_qq) = rownames(expr_AA_proteingene)
rownames(expr_AA_progene_combat_qq) = colnames(expr_AA_proteingene)

#---------------------------------------------------
# adjust for covariates and PCs
#---------------------------------------------------

x = 15
X=expr_AA_progene_combat_qq
pcair_all_order = mypcair$vectors
pcair_PCx = pcair_all_order[,1:x]
func <- function(x) paste0("PC",x)
colnames(pcair_PCx) = unlist(lapply(c(1:x), func))
PCx_meta=cbind(meta,pcair_PCx)
PCx_res=matrix(0,dim(X)[1],dim(X)[2])
PCx_res_qq=matrix(0,dim(X)[1],dim(X)[2])
for(j in 1:dim(X)[2]){
print(j)
PCx_res[,j]=lm(X[,j] ~ factor(gender) + age + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + PC11 + PC12 + PC13 + PC14 + PC15, data = PCx_meta)$residual
PCx_res_qq[,j]=qqnorm(PCx_res[,j],plot.it=F)$x
}
save(PCx_res_qq, file = paste0("PC",x,"_res_qq.RData"))

#---------------------------------------------------
# add permutation
#---------------------------------------------------
load( paste0("permuteID_AA.RData"))
PCx_res_qq_permute = list()
for(permu in 1:10){
	print(permu)
	PCx_res_qq_permute[[permu]] = PCx_res_qq[new_id[[permu]],]
	}
#---------------------------------------------------
# write phenotypes to fam file
#---------------------------------------------------

pathAAbed = "/net/mulan/home/shanglu/GENOA/data/AA/gene_bed"
load("gene_AA_anno_protein.RData")
gene_AA_anno_chr = gene_AA_anno[gene_AA_anno$chr==1,]
gene_AA_anno_chr$index_j = c(1:dim(gene_AA_anno_chr)[1])
gene_AA_anno_order = gene_AA_anno_chr
for(i in 2:22){
gene_AA_anno_chr = gene_AA_anno[gene_AA_anno$chr==i,]
gene_AA_anno_chr$index_j = c(1:dim(gene_AA_anno_chr)[1])
gene_AA_anno_order = rbind(gene_AA_anno_order,gene_AA_anno_chr) 
}
ind=match(gene_AA_anno_order$GENE, gene_AA_anno$GENE)
gene_annotation = gene_AA_anno_order

count=0
for(i in 1:22){
chr.num = sum(gene_annotation$chr==i)
print(paste0("chr ",i," read"))
print(count/length(gene_annotation$chr))
for( j in 1:chr.num){
count=count+1
fn = paste0(pathAAbed,"/chr_", i, "/chr_",i,"_gene_",j,".fam", sep = "")
if(file.exists(fn)){
fam_tmp = read.table(fn)
# col 6 to 17: 10 permutation for PC5
# col 18 to 27: 10 permutation for PC0
# col 30 to 41: 10 permutation for PC10
fam_tmp[,6+36] = c(expr_AA_progene_combat_qq[,ind[count]])
fam_tmp[,7+36] = c(PCx_res_qq[,ind[count]])
fam_tmp[,8+36] = c(PCx_res_qq_permute[[1]][,ind[count]])
fam_tmp[,9+36] = c(PCx_res_qq_permute[[2]][,ind[count]])
fam_tmp[,10+36] = c(PCx_res_qq_permute[[3]][,ind[count]])
fam_tmp[,11+36] = c(PCx_res_qq_permute[[4]][,ind[count]])
fam_tmp[,12+36] = c(PCx_res_qq_permute[[5]][,ind[count]])
fam_tmp[,13+36] = c(PCx_res_qq_permute[[6]][,ind[count]])
fam_tmp[,14+36] = c(PCx_res_qq_permute[[7]][,ind[count]])
fam_tmp[,15+36] = c(PCx_res_qq_permute[[8]][,ind[count]])
fam_tmp[,16+36] = c(PCx_res_qq_permute[[9]][,ind[count]])
fam_tmp[,17+36] = c(PCx_res_qq_permute[[10]][,ind[count]])
write.table(fam_tmp, file=fn, quote = F, row.names = F, col.names = F)
}
}
}



```


# PC20
```
#---------------------------------------------
# load meta:
#---------------------------------------------

pathAA="/net/mulan/home/shanglu/GENOA/data/AA"
load(paste0(pathAA,"/meta_AA.RData"))

#---------------------------------------------
# load PCs
#---------------------------------------------

load(paste0(pathAA,"/mypcairAA.RData"))

#---------------------------------------------
# load expression
#---------------------------------------------

load(paste0(pathAA,"/expression/expr_AA_progene_combat_qq.RData"))
load(paste0(pathAA,"/expression/expr_AA_proteingene.RData"))

colnames(expr_AA_progene_combat_qq) = rownames(expr_AA_proteingene)
rownames(expr_AA_progene_combat_qq) = colnames(expr_AA_proteingene)

#---------------------------------------------------
# adjust for covariates and PCs
#---------------------------------------------------

x = 20
X=expr_AA_progene_combat_qq
pcair_all_order = mypcair$vectors
pcair_PCx = pcair_all_order[,1:x]
func <- function(x) paste0("PC",x)
colnames(pcair_PCx) = unlist(lapply(c(1:x), func))
PCx_meta=cbind(meta,pcair_PCx)
PCx_res=matrix(0,dim(X)[1],dim(X)[2])
PCx_res_qq=matrix(0,dim(X)[1],dim(X)[2])
for(j in 1:dim(X)[2]){
print(j)
PCx_res[,j]=lm(X[,j] ~ factor(gender) + age + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + PC11 + PC12 + PC13 + PC14 + PC15 + PC16 + PC17 + PC18 + PC19 + PC20, data = PCx_meta)$residual
PCx_res_qq[,j]=qqnorm(PCx_res[,j],plot.it=F)$x
}
save(PCx_res_qq, file = paste0("PC",x,"_res_qq.RData"))

#---------------------------------------------------
# add permutation
#---------------------------------------------------
load( paste0("permuteID_AA.RData"))
PCx_res_qq_permute = list()
for(permu in 1:10){
	print(permu)
	PCx_res_qq_permute[[permu]] = PCx_res_qq[new_id[[permu]],]
	}
#---------------------------------------------------
# write phenotypes to fam file
#---------------------------------------------------

pathAAbed = "/net/mulan/home/shanglu/GENOA/data/AA/gene_bed"
load("gene_AA_anno_protein.RData")
gene_AA_anno_chr = gene_AA_anno[gene_AA_anno$chr==1,]
gene_AA_anno_chr$index_j = c(1:dim(gene_AA_anno_chr)[1])
gene_AA_anno_order = gene_AA_anno_chr
for(i in 2:22){
gene_AA_anno_chr = gene_AA_anno[gene_AA_anno$chr==i,]
gene_AA_anno_chr$index_j = c(1:dim(gene_AA_anno_chr)[1])
gene_AA_anno_order = rbind(gene_AA_anno_order,gene_AA_anno_chr) 
}
ind=match(gene_AA_anno_order$GENE, gene_AA_anno$GENE)
gene_annotation = gene_AA_anno_order

count=0
for(i in 1:22){
chr.num = sum(gene_annotation$chr==i)
print(paste0("chr ",i," read"))
print(count/length(gene_annotation$chr))
for( j in 1:chr.num){
count=count+1
fn = paste0(pathAAbed,"/chr_", i, "/chr_",i,"_gene_",j,".fam", sep = "")
if(file.exists(fn)){
fam_tmp = read.table(fn)
# col 6 to 17: 10 permutation for PC5
# col 18 to 27: 10 permutation for PC0
# col 30 to 41: 10 permutation for PC10
fam_tmp[,6+48] = c(expr_AA_progene_combat_qq[,ind[count]])
fam_tmp[,7+48] = c(PCx_res_qq[,ind[count]])
fam_tmp[,8+48] = c(PCx_res_qq_permute[[1]][,ind[count]])
fam_tmp[,9+48] = c(PCx_res_qq_permute[[2]][,ind[count]])
fam_tmp[,10+48] = c(PCx_res_qq_permute[[3]][,ind[count]])
fam_tmp[,11+48] = c(PCx_res_qq_permute[[4]][,ind[count]])
fam_tmp[,12+48] = c(PCx_res_qq_permute[[5]][,ind[count]])
fam_tmp[,13+48] = c(PCx_res_qq_permute[[6]][,ind[count]])
fam_tmp[,14+48] = c(PCx_res_qq_permute[[7]][,ind[count]])
fam_tmp[,15+48] = c(PCx_res_qq_permute[[8]][,ind[count]])
fam_tmp[,16+48] = c(PCx_res_qq_permute[[9]][,ind[count]])
fam_tmp[,17+48] = c(PCx_res_qq_permute[[10]][,ind[count]])
write.table(fam_tmp, file=fn, quote = F, row.names = F, col.names = F)
}
}
}



```

# AA_PC.sh
```
#!/bin/bash
#SBATCH --job-name=AA_PC
#SBATCH --array=1-10
#SBATCH --cpus-per-task=1
#SBATCH --output=/nfs/sph-mulan/home/shanglu/GENOA/data/AA/out/AA_PC%a.out
#SBATCH --error=/nfs/sph-mulan/home/shanglu/GENOA/data/AA/err/AA_PC%a.err
#SBATCH --workdir=/nfs/sph-mulan/home/shanglu/GENOA/data/AA
#SBATCH --partition=standard
#SBATCH --mem-per-cpu=8000MB
#SBATCH --account=xzhousph

bash

let k=0
pathbed=/nfs/sph-mulan/home/shanglu/GENOA/data/AA/gene_bed
pathGfile=/nfs/sph-mulan/home/shanglu/GENOA/data/AA/Gene_file
pathRelatedness=/nfs/sph-mulan/home/shanglu/GENOA/data/AA/filtered_info/output
GEMMA=/nfs/sph-mulan/home/shanglu/GENOA/src/gemma

for pheno in `seq 13 60`
do
for i in `seq 1 22`
do
let k=${k}+1
if [ ${k} -eq ${SLURM_ARRAY_TASK_ID} ]; then
gnum=`cat ${pathGfile}/chr${i}.num`
for j in `seq 1 ${gnum}`
do
${GEMMA} -bfile ${pathbed}/chr_${i}/chr_${i}_gene_${j} -maf 0.05 -r2 0.9999999 -hwe 0 -n ${pheno} -k ${pathRelatedness}/AAcombined_gemma.sXX.txt  -lmm 1 -o /chr_${i}/chr_${i}_gene_${j}_pheno_${pheno}
done 
fi
done
done




```

# collect results

## PC0
```
PCnum = 0

# load gene annotation 
path_AA="/net/mulan/home/shanglu/GENOA/data/AA"
path_AA_result="/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping/result"
path_AA_output="/net/mulan/home/shanglu/GENOA/data/AA/output/"

load(paste0(path_AA,"/gene_AA_anno_protein.RData"))

all_min_p = list()
count = 0
for(pheno in 14:24){
count = count + 1
print(pheno)
min_p_pheno = list()
for(chr in 1:22){

min_p_chr = c()
load(paste0(path_AA_result,"/AA_eqtl_res_chr_",chr,"pheno",pheno,".RData"))

for(g in 1:length(res)){
  if(length(dim(res[[g]])[1])!=0 ){
    if(dim(res[[g]])[1]!=0){
      min_p_chr[g] = res[[g]]$p_wald[which.min(res[[g]]$p_wald)]
      }else{
      min_p_chr[g]=NA
    }
  }else{
      min_p_chr[g]=NA
}
}
min_p_pheno[[chr]] = min_p_chr
}
all_min_p[[count]] = unlist(min_p_pheno)
}

save(all_min_p, file = paste0("/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping/all_min_p_AA_PC",PCnum,".RData"))



AAnull = which(is.na(all_min_p[[1]]))
min_p_AA = all_min_p[[1]][-AAnull]
order_p_AA = min_p_AA[order(min_p_AA)]
min_p_permu_AA = list()
for(i in 1:10){ 
min_p_permu_AA[[i]] = all_min_p[[i+1]][-AAnull]
}
min_p_AA_permu_mat = matrix(unlist(min_p_permu_AA), ncol = length(order_p_AA), byrow = TRUE) 
count = 0 
num_real = NULL 
FDR_AA = NULL 
for(threshold in order_p_AA){ 
count = count + 1 
num_real[count] = sum(min_p_AA<=threshold) 
FDR_AA[count] = mean(apply(min_p_AA_permu_mat, 1, function(x) sum(x<=threshold)/num_real[count])) 
if(FDR_AA[count]>0.2){
break
}

}

max(which(FDR_AA<0.01))
order_p_AA[ max(which(FDR_AA<0.01))]
max(which(FDR_AA<0.05))
order_p_AA[ max(which(FDR_AA<0.05))]
max(which(FDR_AA<0.1))
order_p_AA[ max(which(FDR_AA<0.1))]
max(which(FDR_AA<0.2))
order_p_AA[ max(which(FDR_AA<0.2))]

> 
> max(which(FDR_AA<0.01))
[1] 4495
> order_p_AA[ max(which(FDR_AA<0.01))]
[1] 9.407656e-06
> max(which(FDR_AA<0.05))
[1] 5527
> order_p_AA[ max(which(FDR_AA<0.05))]
[1] 6.312936e-05
> max(which(FDR_AA<0.1))
[1] 6230
> order_p_AA[ max(which(FDR_AA<0.1))]
[1] 0.0001547304
> max(which(FDR_AA<0.2))
[1] 7329
> order_p_AA[ max(which(FDR_AA<0.2))]
[1] 0.000394044

```

## PC10
```
PCnum = 10

# load gene annotation 
path_AA="/net/mulan/home/shanglu/GENOA/data/AA"
path_AA_result="/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping/result"
path_AA_output="/net/mulan/home/shanglu/GENOA/data/AA/output/"

load(paste0(path_AA,"/gene_AA_anno_protein.RData"))

all_min_p = list()
count = 0
for(pheno in 26:36){
count = count + 1
print(pheno)
min_p_pheno = list()
for(chr in 1:22){

min_p_chr = c()
load(paste0(path_AA_result,"/AA_eqtl_res_chr_",chr,"pheno",pheno,".RData"))

for(g in 1:length(res)){
  if(length(dim(res[[g]])[1])!=0 ){
    if(dim(res[[g]])[1]!=0){
      min_p_chr[g] = res[[g]]$p_wald[which.min(res[[g]]$p_wald)]
      }else{
      min_p_chr[g]=NA
    }
  }else{
      min_p_chr[g]=NA
}
}
min_p_pheno[[chr]] = min_p_chr
}
all_min_p[[count]] = unlist(min_p_pheno)
}

save(all_min_p, file = paste0("/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping/all_min_p_AA_PC",PCnum,".RData"))



AAnull = which(is.na(all_min_p[[1]]))
min_p_AA = all_min_p[[1]][-AAnull]
order_p_AA = min_p_AA[order(min_p_AA)]
min_p_permu_AA = list()
for(i in 1:10){ 
min_p_permu_AA[[i]] = all_min_p[[i+1]][-AAnull]
}
min_p_AA_permu_mat = matrix(unlist(min_p_permu_AA), ncol = length(order_p_AA), byrow = TRUE) 
count = 0 
num_real = NULL 
FDR_AA = NULL 
for(threshold in order_p_AA){ 
count = count + 1 
num_real[count] = sum(min_p_AA<=threshold) 
FDR_AA[count] = mean(apply(min_p_AA_permu_mat, 1, function(x) sum(x<=threshold)/num_real[count])) 
if(FDR_AA[count]>0.2){
break
}

}

max(which(FDR_AA<0.01))
order_p_AA[ max(which(FDR_AA<0.01))]
max(which(FDR_AA<0.05))
order_p_AA[ max(which(FDR_AA<0.05))]
max(which(FDR_AA<0.1))
order_p_AA[ max(which(FDR_AA<0.1))]
max(which(FDR_AA<0.2))
order_p_AA[ max(which(FDR_AA<0.2))]

> max(which(FDR_AA<0.01))
[1] 4388
> order_p_AA[ max(which(FDR_AA<0.01))]
[1] 9.053835e-06
> max(which(FDR_AA<0.05))
[1] 5432
> order_p_AA[ max(which(FDR_AA<0.05))]
[1] 6.262234e-05
> max(which(FDR_AA<0.1))
[1] 6123
> order_p_AA[ max(which(FDR_AA<0.1))]
[1] 0.0001509719
> max(which(FDR_AA<0.2))
[1] 7214
> order_p_AA[ max(which(FDR_AA<0.2))]
[1] 0.0003864891

```

## PC15
```
PCnum = 15

# load gene annotation 
path_AA="/net/mulan/home/shanglu/GENOA/data/AA"
path_AA_result="/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping/result"
path_AA_output="/net/mulan/home/shanglu/GENOA/data/AA/output/"

load(paste0(path_AA,"/gene_AA_anno_protein.RData"))

all_min_p = list()
count = 0
for(pheno in 38:48){
count = count + 1
print(pheno)
min_p_pheno = list()
for(chr in 1:22){

min_p_chr = c()
load(paste0(path_AA_result,"/AA_eqtl_res_chr_",chr,"pheno",pheno,".RData"))

for(g in 1:length(res)){
  if(length(dim(res[[g]])[1])!=0 ){
    if(dim(res[[g]])[1]!=0){
      min_p_chr[g] = res[[g]]$p_wald[which.min(res[[g]]$p_wald)]
      }else{
      min_p_chr[g]=NA
    }
  }else{
      min_p_chr[g]=NA
}
}
min_p_pheno[[chr]] = min_p_chr
}
all_min_p[[count]] = unlist(min_p_pheno)
}

save(all_min_p, file = paste0("/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping/all_min_p_AA_PC",PCnum,".RData"))



AAnull = which(is.na(all_min_p[[1]]))
min_p_AA = all_min_p[[1]][-AAnull]
order_p_AA = min_p_AA[order(min_p_AA)]
min_p_permu_AA = list()
for(i in 1:10){ 
min_p_permu_AA[[i]] = all_min_p[[i+1]][-AAnull]
}
min_p_AA_permu_mat = matrix(unlist(min_p_permu_AA), ncol = length(order_p_AA), byrow = TRUE) 
count = 0 
num_real = NULL 
FDR_AA = NULL 
for(threshold in order_p_AA){ 
count = count + 1 
num_real[count] = sum(min_p_AA<=threshold) 
FDR_AA[count] = mean(apply(min_p_AA_permu_mat, 1, function(x) sum(x<=threshold)/num_real[count])) 
if(FDR_AA[count]>0.2){
break
}

}

max(which(FDR_AA<0.01))
order_p_AA[ max(which(FDR_AA<0.01))]
max(which(FDR_AA<0.05))
order_p_AA[ max(which(FDR_AA<0.05))]
max(which(FDR_AA<0.1))
order_p_AA[ max(which(FDR_AA<0.1))]
max(which(FDR_AA<0.2))
order_p_AA[ max(which(FDR_AA<0.2))]



> max(which(FDR_AA<0.01))
[1] 4376
> order_p_AA[ max(which(FDR_AA<0.01))]
[1] 9.043125e-06
> max(which(FDR_AA<0.05))
[1] 5398
> order_p_AA[ max(which(FDR_AA<0.05))]
[1] 6.191039e-05
> max(which(FDR_AA<0.1))
[1] 6039
> order_p_AA[ max(which(FDR_AA<0.1))]
[1] 0.0001481806
> max(which(FDR_AA<0.2))
[1] 7144
> order_p_AA[ max(which(FDR_AA<0.2))]
[1] 0.0003828382

```


## PC20
```
PCnum = 20

# load gene annotation 
path_AA="/net/mulan/home/shanglu/GENOA/data/AA"
path_AA_result="/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping/result"
path_AA_output="/net/mulan/home/shanglu/GENOA/data/AA/output/"

load(paste0(path_AA,"/gene_AA_anno_protein.RData"))

all_min_p = list()
count = 0
for(pheno in 50:60){
count = count + 1
print(pheno)
min_p_pheno = list()
for(chr in 1:22){

min_p_chr = c()
load(paste0(path_AA_result,"/AA_eqtl_res_chr_",chr,"pheno",pheno,".RData"))

for(g in 1:length(res)){
  if(length(dim(res[[g]])[1])!=0 ){
    if(dim(res[[g]])[1]!=0){
      min_p_chr[g] = res[[g]]$p_wald[which.min(res[[g]]$p_wald)]
      }else{
      min_p_chr[g]=NA
    }
  }else{
      min_p_chr[g]=NA
}
}
min_p_pheno[[chr]] = min_p_chr
}
all_min_p[[count]] = unlist(min_p_pheno)
}

save(all_min_p, file = paste0("/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping/all_min_p_AA_PC",PCnum,".RData"))



AAnull = which(is.na(all_min_p[[1]]))
min_p_AA = all_min_p[[1]][-AAnull]
order_p_AA = min_p_AA[order(min_p_AA)]
min_p_permu_AA = list()
for(i in 1:10){ 
min_p_permu_AA[[i]] = all_min_p[[i+1]][-AAnull]
}
min_p_AA_permu_mat = matrix(unlist(min_p_permu_AA), ncol = length(order_p_AA), byrow = TRUE) 
count = 0 
num_real = NULL 
FDR_AA = NULL 
for(threshold in order_p_AA){ 
count = count + 1 
num_real[count] = sum(min_p_AA<=threshold) 
FDR_AA[count] = mean(apply(min_p_AA_permu_mat, 1, function(x) sum(x<=threshold)/num_real[count])) 
if(FDR_AA[count]>0.2){
break
}

}

max(which(FDR_AA<0.01))
order_p_AA[ max(which(FDR_AA<0.01))]
max(which(FDR_AA<0.05))
order_p_AA[ max(which(FDR_AA<0.05))]
max(which(FDR_AA<0.1))
order_p_AA[ max(which(FDR_AA<0.1))]
max(which(FDR_AA<0.2))
order_p_AA[ max(which(FDR_AA<0.2))]

> max(which(FDR_AA<0.01))
[1] 4355
> order_p_AA[ max(which(FDR_AA<0.01))]
[1] 9.291138e-06
> max(which(FDR_AA<0.05))
[1] 5376
> order_p_AA[ max(which(FDR_AA<0.05))]
[1] 6.228262e-05
> max(which(FDR_AA<0.1))
[1] 5994
> order_p_AA[ max(which(FDR_AA<0.1))]
[1] 0.0001459868
> max(which(FDR_AA<0.2))
[1] 7123
> order_p_AA[ max(which(FDR_AA<0.2))]
[1] 0.0003821787

```


```
pheno=14
pheno=26

pheno=38
pheno=50

for(i in 1:22){
print(i)
chr=NULL
rs=NULL
ps=NULL
n_miss=NULL
allele1=NULL
allele0=NULL
af=NULL
beta=NULL
se=NULL
l_remle=NULL
p_wald=NULL
path_AA_result="/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping/result"
AA_full_table = data.frame(chr,rs ,ps,n_miss,allele1,allele0,af,beta,se,l_remle, p_wald )
load("/net/mulan/home/shanglu/GENOA/data/AA/gene_AA_anno_order_correct.RData")

  print(i)
  load(paste0(path_AA_result,"/AA_eqtl_res_chr_",i,"pheno",pheno,".RData"))
  for(j in 1:length(res)){
    if(sum(dim(res[[j]])[1])!=0){
    gene_AA_anno_order_chr = gene_AA_anno_order[gene_AA_anno_order$chr==i,]
    res[[j]]$GENE = gene_AA_anno_order_chr$GENE[j]
    AA_full_table = rbind(AA_full_table, res[[j]])
    }
   }

save(AA_full_table, file = paste0("/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping/AA_full_table_chr_",i,"_pheno",pheno,".RData"))

}
```


# figure
```
AA_thr = c()
PC_eGene = c(5560, 5508,5464,5430,5393)
PC_eSNP = c(344189,338216,334035,331837,328645)

library(RColorBrewer)
COL = brewer.pal(9,"Set1");
PC = c(0,  5, 10, 15, 20)
nG = PC_eGene
nPC = length(PC)
pdf("eGene_vs_PC_AA.pdf", 4, 3)
plot(0, 0, xlim=c(0, 20), ylim=c(0, 10000), type="n", xlab="Number of Genotype PCs", ylab="Number of eGenes", main="", cex.lab=1.2)
abline(v=seq(0, 20, 5), lty=2, col="lightgrey")
abline(h=seq(0, 10000, 1000), lty=2, col="lightgrey")
points(PC, nG, type="o", col=COL[1], pch=20, lwd=2)
#legend("bottomright", legend=c("eGenes vs PCs at 5% FDR in African American"), fill=c(COL[1], "grey"), bg="white", cex=1.2)
dev.off()


#  y: #eSNPs, x: #PCs
PC = c(0,  5, 10, 15, 20)
nG = PC_eSNP
nPC = length(PC)
pdf("eSNP_vs_PC_AA.pdf", 4,3)
plot(0, 0, xlim=c(0, 20), ylim=c(0, 600000), type="n", xlab="Number of Genotype PCs", ylab="Number of eSNPs", main="",cex.lab=1.2)
abline(v=seq(0, 20, 5), lty=2, col="lightgrey")
abline(h=seq(0, 600000, 50000), lty=2, col="lightgrey")
points(PC, nG, type="o", col=COL[2], pch=20, lwd=2)
#legend("bottomright", legend=c("eSNPs vs PCs at 5% FDR in African American"), fill=c(COL[1], "grey"), bg="white", cex=1.2)
dev.off()




```
