

Clean Geuvadis data

```
#-------------------------------
.bim (PLINK extended MAP file)
Extended variant information file accompanying a .bed binary genotype table. 
(--make-just- bim can be used to update just this file.)
A text file with no header line, and one line per variant with the following six fields:
1. Chromosome code (either an integer, or 'X'/'Y'/'XY'/'MT'; '0' indicates unknown) or name 
2. Variant identifier
3. Position in morgans or centimorgans (safe to use dummy value of '0')
4. Base-pair coordinate (normally 1-based, but 0 ok; limited to 2 31-2)
5. Allele 1 (corresponding to clear bits in .bed; usually minor) 
6. Allele 2 (corresponding to set bits in .bed; usually major)
#-------------------------------

# Prediction step: for each chromosome

# genotype: Geuvadis
	# one file with common in AA, EA, MESA
	# one file with common in AA
	# one file with common in EA
	# one file with common in MESA
		# 

# result want: 
	# 4 sets of predictions
	# for each set, predict five different populations in Geuvadis

#------

path = "/net/mulan/home/shanglu/GENOA/data/Geuvadis"
gene_expr_geuvadis = read.table(paste0(path,"/Geuvadis_gene_expression.txt"),header=T)
gene_exp_ping = read.table(paste0(path,"/Geuvadis_gene_expression_ping.txt"),header=T)

gene_name = as.character(gene_exp_ping$TargetID)
gene_name01 = strsplit(gene_name,"[.]")
gene_name02 = c()
for(i in 1:length(gene_name01)){gene_name02[i]=paste0(gene_name01[[i]][1])}
gene_name02=unlist(gene_name02)
gene_exp_ping$ENSG = gene_name02


# remove duplicate SNPs in chr 22 plink file of Geuvadis data
plink --noweb --bfile chr22 --exclude duplicateSNPinchr22.txt  --make-bed --out chr22
#> bim22$V2[which(duplicated(bim22$V2))]
#[1] rs113940759 rs71904485 


library(snpStats)
for(i in 1:22){
print(i)
fam <- paste0(path,"Geuvadis/chr",i,".fam")
bim <- paste0(path,"Geuvadis/chr",i,".bim")
bed <- paste0(path,"Geuvadis/chr",i,".bed")
m = read.plink(bed, bim, fam)
write.SnpMatrix(m$genotypes, paste0(path,"/chr_",i,"_SnpMatrix.txt"),quote=F,col.names = F,row.names = F)
}

# resave SnpMatrix into R data
for(i in 1:22){
	print(i)
	geno = read.table(paste0(path,"/chr_",i,"_SnpMatrix.txt"),header=F)
	save(geno, file = paste0(path,"/chr_",i,"_SnpMatrix.RData"))
}
```

# Prepare Prediction
```
#------------------
# in AA
#------------------
path_pred = "/net/mulan/home/shanglu/GENOA/analysis/prediction/AA"
path_data = "/net/mulan/home/shanglu/GENOA/data/AA"
path_eqtl = "/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping"

load("/net/mulan/home/shanglu/GENOA/data/AA/pred_combinedAA.RData")
load("/net/mulan/home/shanglu/GENOA/data/AA/gene_AA_anno_order_correct.RData")
load(paste0(path_eqtl,"/AA_full_result.RData"))


len = unlist(lapply(pred_combined$chrAA,length))
GENE = rep(gene_AA_anno_order$GENE,times = c(len))
SNP = unlist(pred_combined$rsAA)
chr = unlist(pred_combined$chrAA)
pos = unlist(pred_combined$psAA)
weight = unlist(pred_combined$weight)

bslmm_all_result = data.frame(GENE,SNP,chr,pos,weight)


/*
> dim(bslmm_all_result)
[1] 14511338        5
> dim(AA_full_result)
[1] 14511338       12
> AA_full_result$GENE = as.character(AA_full_result$GENE)
> sum(AA_full_result$GENE==as.character(GENE))
[1] 14511338
> cor.test(bslmm_all_result$weight, AA_full_result$beta)

        Pearson's product-moment correlation

data:  bslmm_all_result$weight and AA_full_result$beta
t = 1383.6, df = 14511336, p-value < 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.3409368 0.3418459
sample estimates:
      cor 
0.3413915 
*/


bslmm_all_result$allele1 = AA_full_result$allele1
bslmm_all_result$allele2 = AA_full_result$allele2
bslmm_all_result$maf = AA_full_result$maf
save(bslmm_all_result, file  = paste0(path_pred,"/bslmm_all_result_AA.RData"))


```


```
#------------------
# in EA
#------------------
path_pred = "/net/mulan/home/shanglu/GENOA/analysis/prediction/EA"
path_data = "/net/mulan/home/shanglu/GENOA/data/EA"
path_eqtl = "/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping"

load("/net/mulan/home/shanglu/GENOA/data/EA/pred_combinedEA.RData")
load("/net/mulan/home/shanglu/GENOA/data/EA/gene_EA_anno_order_correct.RData")
load(paste0(path_eqtl,"/EA_full_result.RData"))


len = unlist(lapply(pred_combined$chrEA,length))
GENE = rep(gene_EA_anno_order$GENE,times = c(len))
SNP = unlist(pred_combined$rsEA)
chr = unlist(pred_combined$chrEA)
pos = unlist(pred_combined$psEA)
weight = unlist(pred_combined$weight)

bslmm_all_result = data.frame(GENE,SNP,chr,pos,weight)


/*
> dim(bslmm_all_result)
[1] 8521801       5
> dim(EA_full_result)
[1] 8521801      12
> EA_full_result$GENE = as.character(EA_full_result$GENE)
> sum(EA_full_result$GENE==as.character(GENE))
[1] 8521801
> cor.test(bslmm_all_result$weight, EA_full_result$beta)

        Pearson's product-moment correlation

data:  bslmm_all_result$weight and EA_full_result$beta
t = 1418.4, df = 8521799, p-value < 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.4364940 0.4375803
sample estimates:
      cor 
0.4370373 

*/


bslmm_all_result$allele1 = EA_full_result$allele1
bslmm_all_result$allele2 = EA_full_result$allele2
bslmm_all_result$maf = EA_full_result$maf
save(bslmm_all_result, file  = paste0(path_pred,"/bslmm_all_result_EA.RData"))

```

# match gene-SNP pairs
```

path_pred = "/net/mulan/home/shanglu/GENOA/analysis/prediction"



path = "/net/mulan/home/shanglu/GENOA/data/Geuvadis"
gene_expr_geuvadis = read.table(paste0(path,"/Geuvadis_gene_expression.txt"),header=T)
gene_exp_ping = read.table(paste0(path,"/Geuvadis_gene_expression_ping.txt"),header=T)

gene_name = as.character(gene_exp_ping$TargetID)
gene_name01 = strsplit(gene_name,"[.]")
gene_name02 = c()
for(i in 1:length(gene_name01)){gene_name02[i]=paste0(gene_name01[[i]][1])}
gene_name02=unlist(gene_name02)
gene_exp_ping$ENSG = gene_name02
```

# AA and EA intersect with Gevadis
```
#---------------------------
# for EA
#---------------------------

load(paste0(path_pred, "/EA/bslmm_all_result_EA.RData"))

for(i in 1:22){
	print(i)
	EA_chr = bslmm_all_result[which(bslmm_all_result$chr==i),]
	genes_ea = unique(as.character(EA_chr$GENE))
	genes_geu = unique(as.character(gene_exp_ping$ENSG)[which(gene_exp_ping$Chr==i)])
	common_gene = intersect(genes_ea,genes_geu)
	EA_keep = EA_chr[which(as.character(EA_chr$GENE) %in% common_gene),]
	bim = paste0(path,"/chr",i,".bim")
	Geu_snp = read.table(bim)
	common_snp = intersect(as.character(Geu_snp$V2),as.character(EA_keep$SNP))
	ind_geu = which(as.character(Geu_snp$V2) %in% common_snp)
	ind_ea = which(as.character(EA_keep$SNP) %in% common_snp)
	EA_to_use = EA_keep[ind_ea,]
	geu_to_use = Geu_snp[ind_geu,]
	index = match( as.character(EA_to_use$SNP),as.character(Geu_snp$V2))
	EA_to_use$index_in_geu = index
	EA_to_use$minor_geu = as.character(Geu_snp$V5)[index]
	EA_to_use$major_geu = as.character(Geu_snp$V6)[index]
	save(EA_to_use, file = paste0(path_pred,"/EA/EA_to_use_chr_",i,".RData"))
	print(dim(EA_to_use))
}

#---------------------------
# for AA
#---------------------------

load(paste0(path_pred, "/AA/bslmm_all_result_AA.RData"))

for(i in 1:22){
	print(i)
	AA_chr = bslmm_all_result[which(bslmm_all_result$chr==i),]
	genes_aa = unique(as.character(AA_chr$GENE))
	genes_geu = unique(as.character(gene_exp_ping$ENSG)[which(gene_exp_ping$Chr==i)])
	common_gene = intersect(genes_aa,genes_geu)
	AA_keep = AA_chr[which(as.character(AA_chr$GENE) %in% common_gene),]
	bim = paste0(path,"/chr",i,".bim")
	Geu_snp = read.table(bim)
	common_snp = intersect(as.character(Geu_snp$V2),as.character(AA_keep$SNP))
	ind_geu = which(as.character(Geu_snp$V2) %in% common_snp)
	ind_aa = which(as.character(AA_keep$SNP) %in% common_snp)
	AA_to_use = AA_keep[ind_aa,]
	geu_to_use = Geu_snp[ind_geu,]
	index = match( as.character(AA_to_use$SNP),as.character(Geu_snp$V2))
	AA_to_use$index_in_geu = index
	AA_to_use$minor_geu = as.character(Geu_snp$V5)[index]
	AA_to_use$major_geu = as.character(Geu_snp$V6)[index]
	save(AA_to_use, file = paste0(path_pred,"/AA/AA_to_use_chr_",i,".RData"))
	print(dim(AA_to_use))
}


```

# MESA intersect with Gevadis
```
#------
# MESA
#------

path_MESA="/net/mulan/home/shanglu/GENOA/data/MESA"
#AFA_full_result= read.table(paste0(path_MESA,"/AFA_cis_eqtl_summary_statistics.txt"),header=T)
#save(AFA_full_result, file = paste0(path_MESA,"/AFA_full_result.RData"))
#CAU_full_result= read.table(paste0(path_MESA,"/CAU_cis_eqtl_summary_statistics.txt"),header=T)
#save(CAU_full_result, file = paste0(path_MESA,"/CAU_full_result.RData"))
#HIS_full_result= read.table(paste0(path_MESA,"/HIS_cis_eqtl_summary_statistics.txt"),header=T)
#save(HIS_full_result, file = paste0(path_MESA,"/HIS_full_result.RData"))

load(paste0(path_MESA,"/AFA_full_result.RData"))
load(paste0(path_MESA,"/CAU_full_result.RData"))
load(paste0(path_MESA,"/HIS_full_result.RData"))

#------------------------
# AFA_full_result 
#------------------------
# seperate by chromosomes

for(i in 1:22){
	print(i)
	AFA_chr = AFA_full_result[which(AFA_full_result$chr==i),]
	genes_AFA = unique(as.character(AFA_chr$gene))
	genes_geu = unique(as.character(gene_exp_ping$ENSG)[which(gene_exp_ping$Chr==i)])
	common_gene = intersect(genes_AFA,genes_geu)
	AFA_keep = AFA_chr[which(as.character(AFA_chr$gene) %in% common_gene),]
	bim = paste0(path,"/chr",i,".bim")
	Geu_snp = read.table(bim)
	common_snp = intersect(as.character(Geu_snp$V2),as.character(AFA_keep$snps))
	ind_geu = which(as.character(Geu_snp$V2) %in% common_snp)
	ind_AFA = which(as.character(AFA_keep$snps) %in% common_snp)
	AFA_to_use = AFA_keep[ind_AFA,]
	geu_to_use = Geu_snp[ind_geu,]
	index = match( as.character(AFA_to_use$snps),as.character(Geu_snp$V2))
	AFA_to_use$index_in_geu = index
	AFA_to_use$minor_geu = as.character(Geu_snp$V5)[index]
	AFA_to_use$major_geu = as.character(Geu_snp$V6)[index]
	save(AFA_to_use, file = paste0(path_pred,"/AFA/AFA_to_use_chr_",i,".RData"))
	print(dim(AFA_to_use))
}

#------------------------
# CAU_full_result 
#------------------------
for(i in 1:22){
	print(i)
	CAU_chr = CAU_full_result[which(CAU_full_result$chr==i),]
	genes_CAU = unique(as.character(CAU_chr$gene))
	genes_geu = unique(as.character(gene_exp_ping$ENSG)[which(gene_exp_ping$Chr==i)])
	common_gene = intersect(genes_CAU,genes_geu)
	CAU_keep = CAU_chr[which(as.character(CAU_chr$gene) %in% common_gene),]
	bim = paste0(path,"Geuvadis/chr",i,".bim")
	Geu_snp = read.table(bim)
	common_snp = intersect(as.character(Geu_snp$V2),as.character(CAU_keep$snps))
	ind_geu = which(as.character(Geu_snp$V2) %in% common_snp)
	ind_CAU = which(as.character(CAU_keep$snps) %in% common_snp)
	CAU_to_use = CAU_keep[ind_CAU,]
	geu_to_use = Geu_snp[ind_geu,]
	index = match( as.character(CAU_to_use$snps),as.character(Geu_snp$V2))
	CAU_to_use$index_in_geu = index
	CAU_to_use$minor_geu = as.character(Geu_snp$V5)[index]
	CAU_to_use$major_geu = as.character(Geu_snp$V6)[index]
	save(CAU_to_use, file = paste0(path_pred,"/CAU/CAU_to_use_chr_",i,".RData"))
	print(dim(CAU_to_use))
}

#------------------------
# HIS_full_result 
#------------------------
for(i in 1:22){
	print(i)
	HIS_chr = HIS_full_result[which(HIS_full_result$chr==i),]
	genes_HIS = unique(as.character(HIS_chr$gene))
	genes_geu = unique(as.character(gene_exp_ping$ENSG)[which(gene_exp_ping$Chr==i)])
	common_gene = intersect(genes_HIS,genes_geu)
	HIS_keep = HIS_chr[which(as.character(HIS_chr$gene) %in% common_gene),]
	bim = paste0(path,"Geuvadis/chr",i,".bim")
	Geu_snp = read.table(bim)
	common_snp = intersect(as.character(Geu_snp$V2),as.character(HIS_keep$snps))
	ind_geu = which(as.character(Geu_snp$V2) %in% common_snp)
	ind_HIS = which(as.character(HIS_keep$snps) %in% common_snp)
	HIS_to_use = HIS_keep[ind_HIS,]
	geu_to_use = Geu_snp[ind_geu,]
	index = match( as.character(HIS_to_use$snps),as.character(Geu_snp$V2))
	HIS_to_use$index_in_geu = index
	HIS_to_use$minor_geu = as.character(Geu_snp$V5)[index]
	HIS_to_use$major_geu = as.character(Geu_snp$V6)[index]
	save(HIS_to_use, file = paste0(path_pred,"/HIS/HIS_to_use_chr_",i,".RData"))
	print(dim(HIS_to_use))
}



```


Next prepare data for fair comparison, 
here, all 5 prediction populations use same set of gene-SNP pairs, but MESA's weight is from MatrixeQTL, which is not suitable. In the following analysis, we compared with MESA elastic net weights.

```

library(purrr)

for(i in 1:22){
	print(i)
	common_list = list()
	load(paste0(path_pred,"/EA/EA_to_use_chr_",i,".RData"))
	load(paste0(path_pred,"/AA/AA_to_use_chr_",i,".RData"))
	load(paste0(path_pred,"/AFA/AFA_to_use_chr_",i,".RData"))
	load(paste0(path_pred,"/HIS/HIS_to_use_chr_",i,".RData"))
	load(paste0(path_pred,"/CAU/CAU_to_use_chr_",i,".RData"))
	gene_aa = unique(as.character(AA_to_use$GENE))
	gene_ea = unique(as.character(EA_to_use$GENE))
	gene_afa = unique(as.character(AFA_to_use$gene))
	gene_cau = unique(as.character(CAU_to_use$gene))
	gene_his = unique(as.character(HIS_to_use$gene))
	gene_common = Reduce(intersect, list(gene_aa,gene_ea,gene_afa,gene_cau,gene_his))
	ind_aa = which(as.character(AA_to_use$GENE) %in% gene_common)
	ind_ea = which(as.character(EA_to_use$GENE) %in% gene_common)
	ind_afa = which(as.character(AFA_to_use$gene) %in% gene_common)
	ind_cau = which(as.character(CAU_to_use$gene) %in% gene_common)
	ind_his = which(as.character(HIS_to_use$gene) %in% gene_common)
	aa_keep = AA_to_use[ind_aa,]
	ea_keep = EA_to_use[ind_ea,]
	afa_keep = AFA_to_use[ind_afa,]
	cau_keep = CAU_to_use[ind_cau,]
	his_keep = HIS_to_use[ind_his,]
	
	print("keep done!")
	
	bim = paste0(path,"/chr",i,".bim")
	Geu_snp = read.table(bim)
	snp_geu = as.character(Geu_snp$V2)
	snp_aa = unique(as.character(aa_keep$SNP))
	snp_ea = unique(as.character(ea_keep$SNP))
	snp_afa = unique(as.character(afa_keep$snps))
	snp_cau = unique(as.character(cau_keep$snps))
	snp_his = unique(as.character(his_keep$snps))
	
	print("snp done!")
	
	common_snp = Reduce(intersect, list(snp_geu,snp_aa,snp_ea,snp_afa,snp_cau,snp_his))
	ind_geu = which(as.character(Geu_snp$V2) %in% common_snp)
	ind_AA = which(as.character(aa_keep$SNP) %in% common_snp)
	ind_EA = which(as.character(ea_keep$SNP) %in% common_snp)
	ind_AFA = which(as.character(afa_keep$snps) %in% common_snp)
	ind_CAU = which(as.character(cau_keep$snps) %in% common_snp)
	ind_HIS = which(as.character(his_keep$snps) %in% common_snp)
	aa_common = aa_keep[ind_AA,]
	ea_common = ea_keep[ind_EA,]
	afa_common = afa_keep[ind_AFA,]
	cau_common = cau_keep[ind_CAU,]
	his_common = his_keep[ind_HIS,]
	
	print("common 1 done!")
	
	aa_common$GENE = as.character(aa_common$GENE)
	aa_common$SNP = as.character(aa_common$SNP)
	aa_common$pair = pmap(aa_common[1:2], paste, sep = '-')
	
	ea_common$GENE = as.character(ea_common$GENE)
	ea_common$SNP = as.character(ea_common$SNP)
	ea_common$pair = pmap(ea_common[1:2], paste, sep = '-')
	
	afa_common$gene = as.character(afa_common$gene)
	afa_common$snps = as.character(afa_common$snps)
	afa_common$pair = pmap(afa_common[2:1], paste, sep = '-')
	
	cau_common$gene = as.character(cau_common$gene)
	cau_common$snps = as.character(cau_common$snps)
	cau_common$pair = pmap(cau_common[2:1], paste, sep = '-')
	
	his_common$gene = as.character(his_common$gene)
	his_common$snps = as.character(his_common$snps)
	his_common$pair = pmap(his_common[2:1], paste, sep = '-')
	
	common_pair = Reduce(intersect, list(aa_common$pair,ea_common$pair,afa_common$pair,cau_common$pair,his_common$pair))

	print("common pair done!")
	
	aa_common_use = aa_common[match( common_pair,aa_common$pair),]
	ea_common_use = ea_common[match(common_pair,ea_common$pair),]
	afa_common_use = afa_common[match(common_pair,afa_common$pair),]
	cau_common_use = cau_common[match(common_pair,cau_common$pair),]
	his_common_use = his_common[match(common_pair,his_common$pair),]
	
	common_list$aa = aa_common_use
	common_list$ea = ea_common_use
	common_list$afa = afa_common_use
	common_list$cau = cau_common_use
	common_list$his = his_common_use
	
	print("Start saving!")
	
	save(common_list, file = paste0(path_pred,"/allcommon/common_list_chr_",i,".RData"))
}

dimm = c()
for(i in 1:22){
load(paste0(path_pred,"/allcommon/common_list_chr_",i,".RData"))
dimm[i] = dim(common_list[[1]])[1]
}

> dimm
 [1] 157584 117849 107290  69932  83052  95382  72490  63978  62180  78490
[11]  94800  85495  37796  60583  46771  48756  60166  29521  58336  45846
[21]  20051  33186
> sum(dimm)
[1] 1529534

library(tidyr)
path="/net/mulan/home/shanglu/GENOA/data/Geuvadis"


popugenoanames = c("aa" , "ea", "afa", "cau", "his")
# GENOA
for(popugenoa in 1:2){
for(i in 1:22){
print(i)
load(paste0(path_pred,"/allcommon/common_list_chr_",i,".RData"))
load(paste0(path,"/chr_",i,"_SnpMatrix.RData"))
fam = paste0(path,"/chr",i,".fam")
Geu_fam = read.table(fam)
print("Data loaded")

dat_to_use=common_list[[popugenoa]]

gene_name = unique(as.character(dat_to_use$GENE))
expr_mat2 = matrix(0,length(gene_name), 465)
rownames(expr_mat2) = gene_name
colnames(expr_mat2) = as.character(Geu_fam$V1)


print("expr_mat created")

for(j in 1:length(gene_name)){
	#print(j)
	subset = dat_to_use[which(as.character(dat_to_use$GENE) %in% gene_name[j]),]        #!!!!!
	genotype = geno[,subset$index_in_geu]
	if(dim(subset)[1]==1){
	genotypeimpute = genotype
	genotypeimpute[is.na(genotypeimpute)] <- mean(genotypeimpute, na.rm = TRUE)
	}else{
	genotypeimpute =  replace_na(genotype,as.list(colMeans(genotype,na.rm=T)))
	}
	# direction of beta
	inconsistent = which(as.character(subset$allele1)!=as.character(subset$minor_geu)) #!!!!!
	subset[inconsistent,]$weight = -subset[inconsistent,]$weight                       #!!!!!

	genotypeimpute = scale(genotypeimpute)
	expr_pred2 = subset$weight %*% t(2-genotypeimpute)
	expr_mat2[j,]=expr_pred2
}
print("Start saving file")
save(expr_mat2, file = paste0(path_pred,"/allcommon/",popugenoanames[popugenoa],"/expr_predict_chr_",i,"_commonpairs_v2.RData"))
}
}

# common pairs, MESA

# MESA

path_pred = "/net/mulan/home/shanglu/GENOA/analysis/prediction"

for(popugenoa in 3:5){
for(i in 1:22){
print(i)
load(paste0(path_pred,"/allcommon/common_list_chr_",i,".RData"))
load(paste0(path,"/chr_",i,"_SnpMatrix.RData"))
fam = paste0(path,"/chr",i,".fam")
Geu_fam = read.table(fam)
	
print("Data loaded")

##!!!
dat_to_use=common_list[[popugenoa]]
dat_to_use$GENE = dat_to_use$gene
dat_to_use$SNP = dat_to_use$snps
dat_to_use$weight = dat_to_use$beta
print("Data loaded")

gene_name = unique(as.character(dat_to_use$GENE))
expr_mat2 = matrix(0,length(gene_name), 465)
rownames(expr_mat2) = gene_name
colnames(expr_mat2) = as.character(Geu_fam$V1)

print("expr_mat created")

for(j in 1:length(gene_name)){
	 print(j)
	subset = dat_to_use[which(as.character(dat_to_use$GENE) %in% gene_name[j]),]        #!!!!!
	genotype = geno[,subset$index_in_geu]
	if(dim(subset)[1]==1){
	genotypeimpute = genotype
	genotypeimpute[is.na(genotypeimpute)] <- mean(genotypeimpute, na.rm = TRUE)
	}else{
	genotypeimpute =  replace_na(genotype,as.list(colMeans(genotype,na.rm=T)))
	}
	# direction of beta
	inconsistent = which(as.character(subset$ref)!=as.character(subset$major_geu)) #!!!!!
	subset[inconsistent,]$weight = -subset[inconsistent,]$weight                       #!!!!!

	genotypeimpute = scale(genotypeimpute)
	expr_pred2 = subset$weight %*% t(2-genotypeimpute)
	expr_mat2[j,]=expr_pred2
}

print("Start saving file")
save(expr_mat2, file = paste0(path_pred,"/allcommon/",popugenoanames[popugenoa],"/expr_predict_chr_",i,"_commonpairs_v2.RData"))
}
}


```

# collect results, prepare in Geuvadis
```

expr_geu = gene_exp_ping[,7:471]
#> sum(colnames(expr_geu)==colnames(expr_mat2))
#[1] 465
#> which(is.na(expr_geu[1,]))
#[1]   2 117 222

expr_geu_462 = expr_geu[,-c(2, 117 ,222)]
rownames(expr_geu_462)=gene_exp_ping$ENSG

expr_geu_462 = as.matrix(expr_geu_462)
save(expr_geu_462, file =paste0(path,"/expr_geu_462.RData"))

population=read.table(paste0(path,"/population465.txt"),head=T)

#> sum(colnames(expr_geu)==as.character(population$Sample))
#[1] 465
#> dim(population)
#[1] 465   6
#> table(population$Population)
#CEU FIN GBR TSI YRI 
# 92  95  96  93  89 
#> popu = unique(as.character(population$Population))
#> popu
#[1] "GBR" "FIN" "CEU" "YRI" "TSI"

population = population[-c(2, 117 ,222),]
# 2, 117 ,222
# now save 462 people version of population information

popu = unique(as.character(population$Population))
pop = list()
for(k in 1:5){
	pop[[k]] = which(as.character(population$Population)==popu[k])
}

save(pop, file = paste0(path,"/pop.RData"))

```

further normalization:

we quantile normalized the gene expression measurements across individuals in each population to a standard normal distribution, and then quantile normalized the gene expression measurements to a standard normal distribution across individuals from all five populations.



```

#---------------------
# common pairs
	# normalized 
	# non-normalized
		# v1
		# v2
#---------------------

library(tidyr)
path = "/net/mulan/home/shanglu/GENOA/data/Geuvadis"


load(paste0(path,"/pop.RData"))
load(paste0(path,"/expr_geu_462.RData"))
popu = c("GBR", "FIN", "CEU" ,"YRI" ,"TSI")

usepop = "ea"
usepop = "aa"
usepop = "afa"
usepop = "cau"
usepop = "his"

count = 0
genes = c()
for(i in 1:22){

	load(paste0(path_pred,"/allcommon/",usepop,"/expr_predict_chr_",i,"_commonpairs_v2.RData"))
	nrow = length(rownames(expr_mat2))
	count = count+nrow
	genes = c(genes,rownames(expr_mat2))
}
count

pearson_p_v2 = matrix(0,count,5)
pearson_rho_v2 = matrix(0,count,5)

countR = 0
for(i in 1:22){
	print(i)

	load(paste0(path_pred,"/allcommon/",usepop,"/expr_predict_chr_",i,"_commonpairs_v2.RData"))
	ind = match(rownames(expr_mat2),rownames(expr_geu_462))
	expr_mat_v2 = expr_mat2[,-c(2, 117 ,222)]
	expr_geu_obs = expr_geu_462[ind,]

	for(j in 1:dim(expr_mat2)[1]){
		countR = countR + 1
		for(k in 1:5){
		res_v2 = cor.test(expr_geu_obs[j,pop[[k]]],expr_mat_v2[j,pop[[k]]],method = "pearson")
		
		pearson_p_v2[countR,k] = res_v2$p.value
		pearson_rho_v2[countR,k] = res_v2$estimate
		}
	}
}


rownames(pearson_p_v2) = genes
rownames(pearson_rho_v2) = genes
colnames(pearson_p_v2) = popu
colnames(pearson_rho_v2) = popu

result = list()
result$pearson_rho_v2 = pearson_rho_v2
result$pearson_p_v2 = pearson_p_v2

save(result, file = paste0(path_pred,"/result/result_",usepop,"_commonpairs.RData"))




```



```


load(paste0(path_pred,"/result/result_","aa","_commonpairs.RData"))
summary(result$pearson_rho_v2)
load(paste0(path_pred,"/result/result_","ea","_commonpairs.RData"))
summary(result$pearson_rho_v2)
load(paste0(path_pred,"/result/result_","afa","_commonpairs.RData"))
summary(result$pearson_rho_v2)
load(paste0(path_pred,"/result/result_","cau","_commonpairs.RData"))
summary(result$pearson_rho_v2)
load(paste0(path_pred,"/result/result_","his","_commonpairs.RData"))
summary(result$pearson_rho_v2)

load(paste0(path_pred,"/result/result_","aa","_commonpairs.RData"))
summary(result$pearson_rho_v2^2)
load(paste0(path_pred,"/result/result_","ea","_commonpairs.RData"))
summary(result$pearson_rho_v2^2)
load(paste0(path_pred,"/result/result_","afa","_commonpairs.RData"))
summary(result$pearson_rho_v2^2)
load(paste0(path_pred,"/result/result_","cau","_commonpairs.RData"))
summary(result$pearson_rho_v2^2)
load(paste0(path_pred,"/result/result_","his","_commonpairs.RData"))
summary(result$pearson_rho_v2^2)


```   

```

> load(paste0(path_pred,"/result/result_","aa","_commonpairs.RData"))
> summary(result$pearson_rho_v2)
      GBR                FIN                CEU                YRI          
 Min.   :-0.53118   Min.   :-0.67130   Min.   :-0.47122   Min.   :-0.59141  
 1st Qu.:-0.03831   1st Qu.:-0.03834   1st Qu.:-0.03829   1st Qu.:-0.02913  
 Median : 0.05480   Median : 0.05089   Median : 0.04575   Median : 0.05698  
 Mean   : 0.06856   Mean   : 0.06662   Mean   : 0.05707   Mean   : 0.06726  
 3rd Qu.: 0.15308   3rd Qu.: 0.15073   3rd Qu.: 0.13871   3rd Qu.: 0.15053  
 Max.   : 0.91015   Max.   : 0.89929   Max.   : 0.86128   Max.   : 0.85553  
                                                          NA's   :1         
      TSI          
 Min.   :-0.58070  
 1st Qu.:-0.04026  
 Median : 0.05022  
 Mean   : 0.06289  
 3rd Qu.: 0.14540  
 Max.   : 0.89448  
                   
> load(paste0(path_pred,"/result/result_","ea","_commonpairs.RData"))
> summary(result$pearson_rho_v2)
      GBR                FIN                CEU                YRI          
 Min.   :-0.70828   Min.   :-0.75090   Min.   :-0.66333   Min.   :-0.71524  
 1st Qu.:-0.03099   1st Qu.:-0.03223   1st Qu.:-0.03796   1st Qu.:-0.05436  
 Median : 0.05745   Median : 0.05740   Median : 0.04993   Median : 0.03140  
 Mean   : 0.07325   Mean   : 0.07184   Mean   : 0.05933   Mean   : 0.03838  
 3rd Qu.: 0.16003   3rd Qu.: 0.15451   3rd Qu.: 0.14345   3rd Qu.: 0.11998  
 Max.   : 0.90323   Max.   : 0.89260   Max.   : 0.85610   Max.   : 0.90247  
                                                          NA's   :1         
      TSI          
 Min.   :-0.76312  
 1st Qu.:-0.03353  
 Median : 0.05449  
 Mean   : 0.06638  
 3rd Qu.: 0.14825  
 Max.   : 0.91648  
                   
> load(paste0(path_pred,"/result/result_","afa","_commonpairs.RData"))
> summary(result$pearson_rho_v2)
      GBR                FIN                CEU                YRI          
 Min.   :-0.73355   Min.   :-0.75906   Min.   :-0.76408   Min.   :-0.72164  
 1st Qu.:-0.06697   1st Qu.:-0.06260   1st Qu.:-0.06701   1st Qu.:-0.06068  
 Median : 0.02162   Median : 0.02286   Median : 0.02011   Median : 0.02360  
 Mean   : 0.03021   Mean   : 0.03076   Mean   : 0.02470   Mean   : 0.03018  
 3rd Qu.: 0.11404   3rd Qu.: 0.11563   3rd Qu.: 0.10743   3rd Qu.: 0.11313  
 Max.   : 0.90352   Max.   : 0.86800   Max.   : 0.85594   Max.   : 0.89736  
                                                          NA's   :1         
      TSI          
 Min.   :-0.76745  
 1st Qu.:-0.06573  
 Median : 0.02171  
 Mean   : 0.02844  
 3rd Qu.: 0.11217  
 Max.   : 0.89181  
> load(paste0(path_pred,"/result/result_","cau","_commonpairs.RData"))
> summary(result$pearson_rho_v2)
      GBR                FIN                CEU                YRI          
 Min.   :-0.76084   Min.   :-0.75331   Min.   :-0.75020   Min.   :-0.72386  
 1st Qu.:-0.06056   1st Qu.:-0.05572   1st Qu.:-0.05980   1st Qu.:-0.06519  
 Median : 0.03124   Median : 0.03131   Median : 0.02797   Median : 0.01699  
 Mean   : 0.04058   Mean   : 0.04038   Mean   : 0.03387   Mean   : 0.02204  
 3rd Qu.: 0.12773   3rd Qu.: 0.12734   3rd Qu.: 0.11843   3rd Qu.: 0.10477  
 Max.   : 0.90803   Max.   : 0.86069   Max.   : 0.85913   Max.   : 0.86216  
                                                          NA's   :1         
      TSI          
 Min.   :-0.77947  
 1st Qu.:-0.05866  
 Median : 0.03115  
 Mean   : 0.03849  
 3rd Qu.: 0.12180  
 Max.   : 0.90407  
                   
> load(paste0(path_pred,"/result/result_","his","_commonpairs.RData"))
> summary(result$pearson_rho_v2)
      GBR                FIN                CEU                YRI          
 Min.   :-0.76745   Min.   :-0.75797   Min.   :-0.74606   Min.   :-0.72229  
 1st Qu.:-0.06369   1st Qu.:-0.05655   1st Qu.:-0.05868   1st Qu.:-0.06236  
 Median : 0.02876   Median : 0.03043   Median : 0.02899   Median : 0.02146  
 Mean   : 0.03814   Mean   : 0.03955   Mean   : 0.03340   Mean   : 0.02749  
 3rd Qu.: 0.12469   3rd Qu.: 0.12467   3rd Qu.: 0.11614   3rd Qu.: 0.11109  
 Max.   : 0.91108   Max.   : 0.86572   Max.   : 0.85659   Max.   : 0.87035  
                                                          NA's   :1         
      TSI          
 Min.   :-0.78173  
 1st Qu.:-0.05669  
 Median : 0.02889  
 Mean   : 0.03672  
 3rd Qu.: 0.11847  
 Max.   : 0.89830  

```

```
> load(paste0(path_pred,"/result/result_","aa","_commonpairs.RData"))
> summary(result$pearson_rho_v2^2)
      GBR                FIN                CEU                YRI          
 Min.   :0.000000   Min.   :0.000000   Min.   :0.000000   Min.   :0.000000  
 1st Qu.:0.002199   1st Qu.:0.002065   1st Qu.:0.001783   1st Qu.:0.002021  
 Median :0.009643   Median :0.009335   Median :0.008558   Median :0.009118  
 Mean   :0.029790   Mean   :0.029231   Mean   :0.024654   Mean   :0.025329  
 3rd Qu.:0.029930   3rd Qu.:0.029849   3rd Qu.:0.025668   3rd Qu.:0.028470  
 Max.   :0.828376   Max.   :0.808716   Max.   :0.741811   Max.   :0.731932  
                                                          NA's   :1         
      TSI          
 Min.   :0.000000  
 1st Qu.:0.002077  
 Median :0.009015  
 Mean   :0.027415  
 3rd Qu.:0.028822  
 Max.   :0.800096  
                   
> load(paste0(path_pred,"/result/result_","ea","_commonpairs.RData"))
> summary(result$pearson_rho_v2^2)
      GBR                FIN                CEU                YRI          
 Min.   :0.000000   Min.   :0.000000   Min.   :0.000000   Min.   :0.000000  
 1st Qu.:0.002022   1st Qu.:0.002206   1st Qu.:0.001959   1st Qu.:0.001695  
 Median :0.009587   Median :0.009899   Median :0.009025   Median :0.007736  
 Mean   :0.030301   Mean   :0.030145   Mean   :0.025386   Mean   :0.020414  
 3rd Qu.:0.031568   3rd Qu.:0.030169   3rd Qu.:0.027750   3rd Qu.:0.023543  
 Max.   :0.815826   Max.   :0.796729   Max.   :0.732914   Max.   :0.814446  
                                                          NA's   :1         
      TSI          
 Min.   :0.000000  
 1st Qu.:0.002097  
 Median :0.009569  
 Mean   :0.027580  
 3rd Qu.:0.028377  
 Max.   :0.839932  
                   
> load(paste0(path_pred,"/result/result_","afa","_commonpairs.RData"))
> summary(result$pearson_rho_v2^2)
      GBR                FIN                CEU                YRI          
 Min.   :0.000000   Min.   :0.000000   Min.   :0.000000   Min.   :0.000000  
 1st Qu.:0.001856   1st Qu.:0.001804   1st Qu.:0.001755   1st Qu.:0.001702  
 Median :0.008181   Median :0.008270   Median :0.007625   Median :0.007691  
 Mean   :0.023790   Mean   :0.023585   Mean   :0.021207   Mean   :0.019950  
 3rd Qu.:0.024945   3rd Qu.:0.024527   3rd Qu.:0.023436   3rd Qu.:0.022135  
 Max.   :0.816354   Max.   :0.753426   Max.   :0.732635   Max.   :0.805260  
                                                          NA's   :1         
      TSI          
 Min.   :0.000000  
 1st Qu.:0.001692  
 Median :0.008104  
 Mean   :0.022601  
 3rd Qu.:0.024003  
 Max.   :0.795329  
                   
                   
> load(paste0(path_pred,"/result/result_","cau","_commonpairs.RData"))
> summary(result$pearson_rho_v2^2)
      GBR                FIN                CEU                YRI          
 Min.   :0.000000   Min.   :0.000000   Min.   :0.000000   Min.   :0.000000  
 1st Qu.:0.001973   1st Qu.:0.001787   1st Qu.:0.001796   1st Qu.:0.001605  
 Median :0.008946   Median :0.008450   Median :0.008287   Median :0.007353  
 Mean   :0.026161   Mean   :0.025588   Mean   :0.022561   Mean   :0.018293  
 3rd Qu.:0.027093   3rd Qu.:0.026713   3rd Qu.:0.024864   3rd Qu.:0.021158  
 Max.   :0.824525   Max.   :0.740782   Max.   :0.738110   Max.   :0.743320  
                                                          NA's   :1         
      TSI          
 Min.   :0.000000  
 1st Qu.:0.001894  
 Median :0.008414  
 Mean   :0.024559  
 3rd Qu.:0.025601  
 Max.   :0.817339  
                   
> load(paste0(path_pred,"/result/result_","his","_commonpairs.RData"))
> summary(result$pearson_rho_v2^2)
      GBR                FIN                CEU                YRI          
 Min.   :0.000000   Min.   :0.000000   Min.   :0.000000   Min.   :0.000000  
 1st Qu.:0.002066   1st Qu.:0.001801   1st Qu.:0.001796   1st Qu.:0.001670  
 Median :0.008861   Median :0.008445   Median :0.008066   Median :0.007489  
 Mean   :0.025751   Mean   :0.025115   Mean   :0.022334   Mean   :0.019122  
 3rd Qu.:0.026981   3rd Qu.:0.026141   3rd Qu.:0.024763   3rd Qu.:0.021976  
 Max.   :0.830066   Max.   :0.749467   Max.   :0.733748   Max.   :0.757504  
                                                          NA's   :1         
      TSI          
 Min.   :0.000000  
 1st Qu.:0.001709  
 Median :0.008229  
 Mean   :0.024179  
 3rd Qu.:0.024957  
 Max.   :0.806951  
 ```

# tables
```
#----
# common pairs, count R2 table
#----
for(k in 1:5){
res_all = matrix(0,5, 7)
geudat = c()
ourpop = c()
gene = c()
count = 0
ourpopulation  = c("aa" , "ea", "afa", "cau", "his")
for(usepop in c("aa" , "ea", "afa", "cau", "his")){
count = count + 1
	load(paste0(path_pred,"/result/result_",usepop,"_commonpairs.RData"))
	R2 = (result$pearson_rho_v2)^2
	geudat = c(geudat, R2[,k])
	ourpop = c(ourpop, rep(usepop,dim(R2)[1]))
	gene = c(gene, rownames(R2))
	dat = data.frame(geudat, ourpop,gene)
	dat$ourpop = factor(c(ourpop),levels = c("aa" , "ea", "afa", "cau", "his"),order=T)
	dat = na.omit(dat)
	dat$ourpop = as.character(dat$ourpop)
	pop = dat$geudat[dat$ourpop==usepop]

	res_all[count,1] = sum(pop>=0)
	res_all[count,2] = sum(pop>=0.01)
	res_all[count,3] = sum(pop>=0.02)
	res_all[count,4] = sum(pop>=0.05)
	res_all[count,5] = sum(pop>=0.1)
	res_all[count,6] = sum(pop>=0.2)
	res_all[count,7] = sum(pop>=0.5)
}
colnames(res_all) = c("R2_0","R2_001","R2_002","R2_005","R2_01","R2_02","R2_05")
rownames(res_all) = c("AA", "EA", "AFA" ,"CAU" ,"HIS")
write.csv(res_all,paste0(path_pred,"/result/res_commonpairs_predict_",popu[k],".csv"),quote=F)
}





```

# PVE and prediction performance

```
load("/net/mulan/home/shanglu/GENOA/data/AA/gene_AA_anno_order_correct.RData")

 library(purrr)
 library(plyr)
ind_in_aa = match(rownames(pearson_rho_v2),gene_AA_anno_order$GENE  )
geno_aa = gene_AA_anno_order[ind_in_aa,]



usepop = "aa"
load(paste0(path_pred,"/result/result_",usepop,"_commonpairs.RData"))
R2 = c(result$pearson_rho_v2^2)
p=c(result$pearson_p_v2)
pve = rep(geno_aa$PVE_bslmm_combined,5)
pop = rep(colnames(pearson_rho_v2 ),each = dim(pearson_rho_v2)[1] )
pval = c(p)
dat = data.frame(pop, pve, R2,pval)


library(ggplot2)
#-----------
# all points in 5 populations, use AA
#-----------


dat = na.omit(dat)
datt = dat
pdf(paste0(path_pred, "/result/",usepop,"_scatter_pearson_pve_allpoints.pdf"))
p <- ggplot(datt, aes(x=pve, y=R2,color=pop)) +
  geom_point(alpha = 0.8) + 
  geom_smooth(method="lm", se=TRUE, fullrange=TRUE)+
  labs(title=paste0("AA: prediction R2 and PVE"),
         subtitle="using common gene-SNP pairs ",
         caption="",
         x="pve",
         y="R2")+
         theme_bw(base_size = 22)+coord_cartesian(ylim = c(0, 1),xlim = c(0, 1))
p
dev.off()  

/*
> cor.test(dat$R2,dat$pve)

        Pearson's product-moment correlation

data:  dat$R2 and dat$pve
t = 86.047, df = 43477, p-value < 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.3734073 0.3894711
sample estimates:
     cor 
0.381468 
*/

#-----------
#  points with R2>=0.1, in 5 populations, use AA
#-----------

dat = na.omit(dat)
datt = dat[dat$R2>=0.1,]
pdf(paste0(path_pred, "/result/",usepop,"_scatter_pearson_pve_R2_0.1.pdf"))
p <- ggplot(datt, aes(x=pve, y=R2,color=pop)) +
  geom_point(alpha = 0.8) + 
  geom_smooth(method="lm", se=TRUE, fullrange=TRUE)+
  labs(title=paste0(usepop,": prediction R2 and PVE"),
         subtitle="using common gene-SNP pairs ",
         caption="",
         x="pve",
         y="R2")+
         theme_bw(base_size = 22)+coord_cartesian(ylim = c(0, 1),xlim = c(0, 1))
p
dev.off()  

cor.test(datt$R2,datt$pve)

/*
> cor.test(datt$R2,datt$pve)

        Pearson's product-moment correlation

data:  datt$R2 and datt$pve
t = 23.999, df = 2555, p-value < 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.3967321 0.4600147
sample estimates:
      cor 
0.4288994 
*/


#-----------
#  points in YRI, use AA
#-----------
dat = na.omit(dat)
datt = dat[dat$pop=="YRI",]
pdf(paste0(path_pred, "/result/",usepop,"_scatter_pearson_pve_YRI.pdf"))
p <- ggplot(datt, aes(x=pve, y=R2)) +
  geom_point(alpha = 0.8) + 
  geom_smooth(method="lm", se=TRUE, fullrange=TRUE)+
  labs(title=paste0(usepop,": prediction R2 and PVE"),
         subtitle="using common gene-SNP pairs ",
         caption="",
         x="pve",
         y="R2")+
         theme_bw(base_size = 22)+coord_cartesian(ylim = c(0, 1),xlim = c(0, 1))
p
dev.off()  
/*
cor.test(datt$R2,datt$pve)
> cor.test(datt$R2,datt$pve)

        Pearson's product-moment correlation

data:  datt$R2 and datt$pve
t = 41.233, df = 8693, p-value < 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.3867262 0.4218909
sample estimates:
     cor 
0.404458 
*/


#-----------
# only norminal p value significant points in 5 populations, use AA
#-----------

dat = na.omit(dat)
datt = dat
datt = datt[datt$pval<0.05,]
pdf(paste0(path_pred, "/result/",usepop,"_scatter_pearson_pve_pval.pdf"))
p <- ggplot(datt, aes(x=pve, y=R2,color=pop)) +
  geom_point(alpha = 0.8) + 
  geom_smooth(method="lm", se=TRUE, fullrange=TRUE)+
  labs(title=paste0(usepop,": prediction R2 and PVE"),
         subtitle="using common gene-SNP pairs ",
         caption="",
         x="pve",
         y="R2")+
         theme_bw(base_size = 22)+coord_cartesian(ylim = c(0, 1),xlim = c(0, 1))
p
dev.off()  

cor.test(datt$R2,datt$pve)
/*
> cor.test(datt$R2,datt$pve)

        Pearson's product-moment correlation

data:  datt$R2 and datt$pve
t = 44.144, df = 7507, p-value < 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.4358247 0.4717432
sample estimates:
      cor 
0.4539684 
*/


```

# figures

```
#-----------
# all points in 5 populations, sep, use AA
#-----------


#--------------
# AA
#--------------

usepop = "aa"
load(paste0(path_pred,"/result/result_",usepop,"_commonpairs.RData"))
R2 = c(result$pearson_rho_v2^2)
p=c(result$pearson_p_v2)
pve = rep(geno_aa$PVE_bslmm_combined,5)
pop = rep(colnames(pearson_rho_v2 ),each = dim(pearson_rho_v2)[1] )
pval = c(p)
dat = data.frame(pop, pve, R2,pval)

usepop = "African American"
kk=0

kk=kk+1
dat = na.omit(dat)
datt = dat
datt = datt[datt$pop==colnames(pearson_rho_v2)[kk],]
pdf(paste0(path_pred, "/result/",usepop,"_scatter_pearson_pve_pval_",colnames(pearson_rho_v2)[kk],".pdf"))
p <- ggplot(datt, aes(x=pve, y=R2)) +
  geom_point(alpha = 0.8) + 
  geom_smooth(method="lm", se=TRUE, fullrange=TRUE)+
  labs(title=paste0(usepop,": predict ",colnames(pearson_rho_v2)[kk]),
         subtitle="using common gene-SNP pairs",
         caption="",
         x="pve",
         y="R2")+
         theme_bw(base_size = 22)+coord_cartesian(ylim = c(0, 1),xlim = c(0, 1))
p
dev.off()  
cor.test(datt$R2,datt$pve)$estimate
cor.test(datt$R2,datt$pve)$p.value
#

/*
0.3758505 
> cor.test(datt$R2,datt$pve)$p.value
[1] 7.011525e-290

> cor.test(datt$R2,datt$pve)$estimate
      cor 
0.3830161 
> cor.test(datt$R2,datt$pve)$p.value
[1] 7.000772e-302

> cor.test(datt$R2,datt$pve)$estimate
      cor 
0.3736225 
> cor.test(datt$R2,datt$pve)$p.value
[1] 3.279726e-286

> cor.test(datt$R2,datt$pve)$estimate
     cor 
0.404458 
> cor.test(datt$R2,datt$pve)$p.value
[1] 0
> cor.test(datt$R2,datt$pve)$estimate
      cor 
0.3827652 
> cor.test(datt$R2,datt$pve)$p.value
[1] 1.864396e-301
*/



#---------
# EA
#---------
usepop = "ea"
load(paste0(path_pred,"/result/result_",usepop,"_commonpairs.RData"))
R2 = c(result$pearson_rho_v2^2)
p=c(result$pearson_p_v2)
pve = rep(geno_aa$PVE_bslmm_combined,5)
pop = rep(colnames(pearson_rho_v2 ),each = dim(pearson_rho_v2)[1] )
pval = c(p)
dat = data.frame(pop, pve, R2,pval)

usepop = "European American"
kk=0

kk=kk+1
dat = na.omit(dat)
datt = dat
datt = datt[datt$pop==colnames(pearson_rho_v2)[kk],]
pdf(paste0(path_pred, "/result/",usepop,"_scatter_pearson_pve_pval_",colnames(pearson_rho_v2)[kk],".pdf"))
p <- ggplot(datt, aes(x=pve, y=R2)) +
  geom_point(alpha = 0.8) + 
  geom_smooth(method="lm", se=TRUE, fullrange=TRUE)+
  labs(title=paste0(usepop,": predict ",colnames(pearson_rho_v2)[kk]),
         subtitle="using common gene-SNP pairs",
         caption="",
         x="pve",
         y="R2")+
         theme_bw(base_size = 22)+coord_cartesian(ylim = c(0, 1),xlim = c(0, 1))
p
dev.off()  
cor.test(datt$R2,datt$pve)$estimate
cor.test(datt$R2,datt$pve)$p.value

/*
> cor.test(datt$R2,datt$pve)$estimate
      cor 
0.3691742 
> cor.test(datt$R2,datt$pve)$p.value
[1] 5.716892e-279
> cor.test(datt$R2,datt$pve)$estimate
      cor 
0.3739371 
> cor.test(datt$R2,datt$pve)$p.value
[1] 9.985818e-287
> 
> cor.test(datt$R2,datt$pve)$estimate
      cor 
0.3623473 
> cor.test(datt$R2,datt$pve)$p.value
[1] 4.462125e-268

> cor.test(datt$R2,datt$pve)$estimate
      cor 
0.3414125 
> cor.test(datt$R2,datt$pve)$p.value
[1] 2.830844e-236

> cor.test(datt$R2,datt$pve)$estimate
      cor 
0.3761394 
> cor.test(datt$R2,datt$pve)$p.value
[1] 2.332998e-290

#---------
# AFA
#---------
usepop = "afa"
load(paste0(path_pred,"/result/result_",usepop,"_commonpairs.RData"))
R2 = c(result$pearson_rho_v2^2)
p=c(result$pearson_p_v2)
pve = rep(geno_aa$PVE_bslmm_combined,5)
pop = rep(colnames(pearson_rho_v2 ),each = dim(pearson_rho_v2)[1] )
pval = c(p)
dat = data.frame(pop, pve, R2,pval)

usepop = "MESA AFA"
kk=0

kk=kk+1
dat = na.omit(dat)
datt = dat
datt = datt[datt$pop==colnames(pearson_rho_v2)[kk],]
pdf(paste0(path_pred, "/result/",usepop,"_scatter_pearson_pve_pval_",colnames(pearson_rho_v2)[kk],".pdf"))
p <- ggplot(datt, aes(x=pve, y=R2)) +
  geom_point(alpha = 0.8) + 
  geom_smooth(method="lm", se=TRUE, fullrange=TRUE)+
  labs(title=paste0(usepop,": predict ",colnames(pearson_rho_v2)[kk]),
         subtitle="using common gene-SNP pairs",
         caption="",
         x="pve",
         y="R2")+
         theme_bw(base_size = 22)+coord_cartesian(ylim = c(0, 1),xlim = c(0, 1))
p
dev.off()  
cor.test(datt$R2,datt$pve)$estimate
cor.test(datt$R2,datt$pve)$p.value

/*
> cor.test(datt$R2,datt$pve)$estimate
      cor 
0.3139735 
> cor.test(datt$R2,datt$pve)$p.value
[1] 3.197365e-198

> cor.test(datt$R2,datt$pve)$estimate
      cor 
0.3288483 
> cor.test(datt$R2,datt$pve)$p.value
[1] 2.249673e-218
> cor.test(datt$R2,datt$pve)$estimate
      cor 
0.3104701 
> cor.test(datt$R2,datt$pve)$p.value
[1] 1.218736e-193
> cor.test(datt$R2,datt$pve)$estimate
      cor 
0.3286069 
> cor.test(datt$R2,datt$pve)$p.value
[1] 5.166137e-218
> cor.test(datt$R2,datt$pve)$estimate
     cor 
0.318113 
> cor.test(datt$R2,datt$pve)$p.value
[1] 1.026273e-203

*/


#---------
# CAU
#---------
usepop = "cau"
load(paste0(path_pred,"/result/result_",usepop,"_commonpairs.RData"))
R2 = c(result$pearson_rho_v2^2)
p=c(result$pearson_p_v2)
pve = rep(geno_aa$PVE_bslmm_combined,5)
pop = rep(colnames(pearson_rho_v2 ),each = dim(pearson_rho_v2)[1] )
pval = c(p)
dat = data.frame(pop, pve, R2,pval)

usepop = "MESA CAU"
kk=0

kk=kk+1
dat = na.omit(dat)
datt = dat
datt = datt[datt$pop==colnames(pearson_rho_v2)[kk],]
pdf(paste0(path_pred, "/result/",usepop,"_scatter_pearson_pve_pval_",colnames(pearson_rho_v2)[kk],".pdf"))
p <- ggplot(datt, aes(x=pve, y=R2)) +
  geom_point(alpha = 0.8) + 
  geom_smooth(method="lm", se=TRUE, fullrange=TRUE)+
  labs(title=paste0(usepop,": predict ",colnames(pearson_rho_v2)[kk]),
         subtitle="using common gene-SNP pairs",
         caption="",
         x="pve",
         y="R2")+
         theme_bw(base_size = 22)+coord_cartesian(ylim = c(0, 1),xlim = c(0, 1))
p
dev.off()  

cor.test(datt$R2,datt$pve)$estimate
cor.test(datt$R2,datt$pve)$p.value

/*
> cor.test(datt$R2,datt$pve)$estimate
      cor 
0.3239897 
> cor.test(datt$R2,datt$pve)$p.value
[1] 1.150288e-211
> cor.test(datt$R2,datt$pve)$estimate
      cor 
0.3331832 
> cor.test(datt$R2,datt$pve)$p.value
[1] 1.826981e-224
> cor.test(datt$R2,datt$pve)$estimate
      cor 
0.3166922 
> cor.test(datt$R2,datt$pve)$p.value
[1] 8.066589e-202
> cor.test(datt$R2,datt$pve)$estimate
      cor 
0.2887653 
> cor.test(datt$R2,datt$pve)$p.value
[1] 1.306377e-166
> cor.test(datt$R2,datt$pve)$estimate
      cor 
0.3250174 
> cor.test(datt$R2,datt$pve)$p.value
[1] 4.487363e-213


*/


#---------
# HIS
#---------
usepop = "his"
load(paste0(path_pred,"/result/result_",usepop,"_commonpairs.RData"))
R2 = c(result$pearson_rho_v2^2)
p=c(result$pearson_p_v2)
pve = rep(geno_aa$PVE_bslmm_combined,5)
pop = rep(colnames(pearson_rho_v2 ),each = dim(pearson_rho_v2)[1] )
pval = c(p)
dat = data.frame(pop, pve, R2,pval)

usepop = "MESA HIS"
kk=0

kk=kk+1
dat = na.omit(dat)
datt = dat
datt = datt[datt$pop==colnames(pearson_rho_v2)[kk],]
pdf(paste0(path_pred, "/result/",usepop,"_scatter_pearson_pve_pval_",colnames(pearson_rho_v2)[kk],".pdf"))
p <- ggplot(datt, aes(x=pve, y=R2)) +
  geom_point(alpha = 0.8) + 
  geom_smooth(method="lm", se=TRUE, fullrange=TRUE)+
  labs(title=paste0(usepop,": predict ",colnames(pearson_rho_v2)[kk]),
         subtitle="using common gene-SNP pairs",
         caption="",
         x="pve",
         y="R2")+
         theme_bw(base_size = 22)+coord_cartesian(ylim = c(0, 1),xlim = c(0, 1))
p
dev.off()  
cor.test(datt$R2,datt$pve)$estimate
cor.test(datt$R2,datt$pve)$p.value

/*
> cor.test(datt$R2,datt$pve)$estimate
      cor 
0.3249518 
> cor.test(datt$R2,datt$pve)$p.value
[1] 5.52195e-213

> cor.test(datt$R2,datt$pve)$estimate
     cor 
0.332654 
> cor.test(datt$R2,datt$pve)$p.value
[1] 1.024588e-223
> cor.test(datt$R2,datt$pve)$estimate
      cor 
0.3201443 
> cor.test(datt$R2,datt$pve)$p.value
[1] 1.920191e-206
> cor.test(datt$R2,datt$pve)$estimate
      cor 
0.3008242 
> cor.test(datt$R2,datt$pve)$p.value
[1] 2.508441e-181
> cor.test(datt$R2,datt$pve)$estimate
      cor 
0.3255238 
> cor.test(datt$R2,datt$pve)$p.value
[1] 9.033801e-214

*/
```

# AA, use all pairs, bslmm
```R
all pairs, AA and EA are using bslmm

#------------------
# in AA
#------------------

args <- as.numeric(commandArgs(TRUE))
i=args[1]
print(i)


library(tidyr)
path_pred = "/net/mulan/home/shanglu/GENOA/analysis/prediction"
path = "/net/mulan/home/shanglu/GENOA/data/Geuvadis"

print(i)
load(paste0(path_pred,"/AA/AA_to_use_chr_",i,".RData"))
load(paste0(path,"/chr_",i,"_SnpMatrix.RData"))
fam = paste0(path,"/chr",i,".fam")
Geu_fam = read.table(fam)
	
print("Data loaded")

gene_name = unique(as.character(AA_to_use$GENE))
expr_mat1 = matrix(0,length(gene_name), 465)
rownames(expr_mat1) = gene_name
colnames(expr_mat1) = as.character(Geu_fam$V1)
expr_mat2 = expr_mat1

print("expr_mat created")

for(j in 1:length(gene_name)){
	 print(j)
	subset = AA_to_use[which(as.character(AA_to_use$GENE) %in% gene_name[j]),]        #!!!!!
	genotype = geno[,subset$index_in_geu]
	if(dim(subset)[1]==1){
	genotypeimpute = genotype
	genotypeimpute[is.na(genotypeimpute)] <- mean(genotypeimpute, na.rm = TRUE)
	}else{
	genotypeimpute =  replace_na(genotype,as.list(colMeans(genotype,na.rm=T)))
	}
	# direction of beta
	inconsistent = which(as.character(subset$allele1)!=as.character(subset$minor_geu)) #!!!!! allele1 in GEMMA is alternative allele, which is minor allele in geu
	subset[inconsistent,]$weight = -subset[inconsistent,]$weight                       #!!!!!

	genotypeimpute = scale(genotypeimpute)
	expr_pred2 = subset$weight %*% t(2-genotypeimpute)
	expr_mat2[j,]=expr_pred2
}
print("Start saving file")
save(expr_mat2, file = paste0(path_pred,"/AA/expr_predict_chr_",i,"_bslmm_allpairs_v2.RData"))

#------------------------


#!/bin/bash
#SBATCH --job-name=AAbslmmall
#SBATCH --array=1-22
#SBATCH --cpus-per-task=1
#SBATCH --output=/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping/out/AAresult%a.out
#SBATCH --error=/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping/err/AAresult%a.err
#SBATCH --workdir=/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping/code
#SBATCH --partition=mulan
#SBATCH --mem-per-cpu=15000MB

bash

let k=0


for ((chr=1;chr<=22;chr++)); do
  let k=${k}+1
  if [ ${k} -eq ${SLURM_ARRAY_TASK_ID} ]; then
  Rscript --verbose ./AA_predict_bslmm_allpairs.r ${chr} 
  fi
done

#------------------
# in EA
#------------------
args <- as.numeric(commandArgs(TRUE))
i=args[1]
print(i)


library(tidyr)
path_pred = "/net/mulan/home/shanglu/GENOA/analysis/prediction"
path = "/net/mulan/home/shanglu/GENOA/data/Geuvadis"


print(i)
load(paste0(path_pred,"/EA/EA_to_use_chr_",i,".RData"))
load(paste0(path,"/chr_",i,"_SnpMatrix.RData"))
fam = paste0(path,"/chr",i,".fam")
Geu_fam = read.table(fam)
	
print("Data loaded")

gene_name = unique(as.character(EA_to_use$GENE))
expr_mat1 = matrix(0,length(gene_name), 465)
rownames(expr_mat1) = gene_name
colnames(expr_mat1) = as.character(Geu_fam$V1)
expr_mat2 = expr_mat1

print("expr_mat created")

for(j in 1:length(gene_name)){
	 print(j)
	subset = EA_to_use[which(as.character(EA_to_use$GENE) %in% gene_name[j]),]        #!!!!!
	genotype = geno[,subset$index_in_geu]
	if(dim(subset)[1]==1){
	genotypeimpute = genotype
	genotypeimpute[is.na(genotypeimpute)] <- mean(genotypeimpute, na.rm = TRUE)
	}else{
	genotypeimpute =  replace_na(genotype,as.list(colMeans(genotype,na.rm=T)))
	}
	# direction of beta
	inconsistent = which(as.character(subset$allele1)!=as.character(subset$minor_geu)) #!!!!! allele1 in GEMMA is alternative allele, which is minor allele in geu
	subset[inconsistent,]$weight = -subset[inconsistent,]$weight                       #!!!!!

	genotypeimpute = scale(genotypeimpute)
	expr_pred2 = subset$weight %*% t(2-genotypeimpute)
	expr_mat2[j,]=expr_pred2
}
print("Start saving file")
save(expr_mat2, file = paste0(path_pred,"/EA/expr_predict_chr_",i,"_bslmm_allpairs_v2.RData"))

#-------------
#!/bin/bash
#SBATCH --job-name=EAbslmmall
#SBATCH --array=1-22
#SBATCH --cpus-per-task=1
#SBATCH --output=/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping/out/EAresult%a.out
#SBATCH --error=/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping/err/EAresult%a.err
#SBATCH --workdir=/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping/code
#SBATCH --partition=mulan
#SBATCH --mem-per-cpu=8000MB

bash

let k=0


for ((chr=1;chr<=22;chr++)); do
  let k=${k}+1
  if [ ${k} -eq ${SLURM_ARRAY_TASK_ID} ]; then
  Rscript --verbose ./EA_predict_bslmm_allpairs.r ${chr} 
  fi
done

```

# collect results for GENOA using all pairs
```R
use all pairs

#--------------------------
# Pearson's correlation
# correlation with observed expression
# all pairs
	# normalized 
	# non-normalized
		# v1
		# v2
#--------------------------

library(tidyr)

load(paste0(path,"/pop.RData"))
load(paste0(path,"/expr_geu_462.RData"))
popu = c("GBR", "FIN", "CEU" ,"YRI" ,"TSI")

#----
usepop = "EA"
usepop = "AA"
#----
count = 0
genes = c()
for(i in 1:22){

	load(paste0(path_pred,"/",usepop,"/expr_predict_chr_",i,"_bslmm_allpairs_v2.RData"))
	nrow = length(rownames(expr_mat2))
	count = count+nrow
	genes = c(genes,rownames(expr_mat2))
}
[1] 13554 in EA 
[1] 13710 in AA

pearson_p_v2 = matrix(0,count,5)
pearson_rho_v2 = matrix(0,count,5)
norm_pearson_p_v2 = matrix(0,count,5)
norm_pearson_rho_v2 = matrix(0,count,5)

countR = 0
for(i in 1:22){
	print(i)
	# v2
	load(paste0(path_pred,"/",usepop,"/expr_predict_chr_",i,"_bslmm_allpairs_v2.RData"))
	ind = match(rownames(expr_mat2),rownames(expr_geu_462))
	expr_mat_v2 = expr_mat2[,-c(2, 117 ,222)]
	expr_geu_obs = expr_geu_462[ind,]
	
	for(j in 1:dim(expr_mat2)[1]){
		
		countR = countR + 1
		#print(countR)
		for(k in 1:5){
		res_v2 = cor.test(expr_geu_obs[j,pop[[k]]],expr_mat_v2[j,pop[[k]]],method = "pearson")
		norm0 = qqnorm(expr_geu_obs[j,pop[[k]]],plot=F)$x
		norm2 = qqnorm(expr_mat2[j,pop[[k]]],plot=F)$x
		res_norm_v2 = cor.test(norm0,norm2,method = "pearson")
		pearson_p_v2[countR,k] = res_v2$p.value
		pearson_rho_v2[countR,k] = res_v2$estimate
		norm_pearson_p_v2[countR,k] = res_norm_v2$p.value
		norm_pearson_rho_v2[countR,k] = res_norm_v2$estimate
		}
	}
	
}



rownames(pearson_p_v2) = genes
rownames(pearson_rho_v2) = genes
rownames(norm_pearson_p_v2) = genes
rownames(norm_pearson_rho_v2) = genes

colnames(pearson_p_v2) = popu
colnames(pearson_rho_v2) = popu
colnames(norm_pearson_p_v2) = popu
colnames(norm_pearson_rho_v2) = popu

result = list()
result$pearson_rho_v2 = pearson_rho_v2
result$pearson_p_v2 = pearson_p_v2
result$norm_pearson_p_v2 = norm_pearson_p_v2
result$norm_pearson_rho_v2 = norm_pearson_rho_v2

save(result, file = paste0(path_pred,"/result/result_",usepop,"_bslmm_allpairs.RData"))


load(paste0(path_pred,"/result/result_","AA","_bslmm_allpairs.RData"))
summary(result$pearson_rho_v2)
load(paste0(path_pred,"/result/result_","EA","_bslmm_allpairs.RData"))
summary(result$pearson_rho_v2)

###################################
# EA results
> summary(result$pearson_rho_v2)
      GBR                FIN                CEU                YRI          
 Min.   :-0.66644   Min.   :-0.72840   Min.   :-0.65314   Min.   :-0.61367  
 1st Qu.:-0.02773   1st Qu.:-0.02844   1st Qu.:-0.03399   1st Qu.:-0.04735  
 Median : 0.06394   Median : 0.06050   Median : 0.05103   Median : 0.03390  
 Mean   : 0.08038   Mean   : 0.07719   Mean   : 0.06322   Mean   : 0.04217  
 3rd Qu.: 0.16643   3rd Qu.: 0.16150   3rd Qu.: 0.14682   3rd Qu.: 0.12331  
 Max.   : 0.90716   Max.   : 0.88538   Max.   : 0.85903   Max.   : 0.90818  
                                                          NA's   :6         
      TSI          
 Min.   :-0.74042  
 1st Qu.:-0.03005  
 Median : 0.05862  
 Mean   : 0.07236  
 3rd Qu.: 0.15681  
 Max.   : 0.88127  
                   
> summary(result$pearson_rho_v2>=0)
    GBR             FIN             CEU             YRI         
 Mode :logical   Mode :logical   Mode :logical   Mode :logical  
 FALSE:4357      FALSE:4402      FALSE:4650      FALSE:5295     
 TRUE :9197      TRUE :9152      TRUE :8904      TRUE :8253     
                                                 NA's :6        
    TSI         
 Mode :logical  
 FALSE:4478     
 TRUE :9076
 
###################################
 # AA results
 > summary(result$pearson_rho_v2)
      GBR                FIN                CEU                YRI          
 Min.   :-0.61244   Min.   :-0.71920   Min.   :-0.48508   Min.   :-0.51712  
 1st Qu.:-0.03391   1st Qu.:-0.03544   1st Qu.:-0.03921   1st Qu.:-0.02292  
 Median : 0.05527   Median : 0.05433   Median : 0.04688   Median : 0.06358  
 Mean   : 0.07269   Mean   : 0.07097   Mean   : 0.05858   Mean   : 0.07691  
 3rd Qu.: 0.15664   3rd Qu.: 0.15594   3rd Qu.: 0.13949   3rd Qu.: 0.16251  
 Max.   : 0.91795   Max.   : 0.88058   Max.   : 0.87294   Max.   : 0.89863  
 NA's   :1          NA's   :2                             NA's   :3         
      TSI          
 Min.   :-0.62612  
 1st Qu.:-0.03571  
 Median : 0.05507  
 Mean   : 0.06717  
 3rd Qu.: 0.14939  
 Max.   : 0.89724  
 NA's   :1         
> summary(result$pearson_rho_v2>=0)
    GBR             FIN             CEU             YRI         
 Mode :logical   Mode :logical   Mode :logical   Mode :logical  
 FALSE:4681      FALSE:4736      FALSE:4888      FALSE:4251     
 TRUE :9028      TRUE :8972      TRUE :8822      TRUE :9456     
 NA's :1         NA's :2                         NA's :3        
    TSI         
 Mode :logical  
 FALSE:4744     
 TRUE :8965     
 NA's :1  
 
 pearson_rho = na.omit(result$pearson_rho_v2)
 pve =  gene_AA_anno_order$PVE_bslmm_combined[match(rownames(pearson_rho),gene_AA_anno_order$GENE)]
 cor_rho = c(cor.test(pearson_rho[,1],pve)$estimate,
 cor.test(pearson_rho[,2],pve)$estimate,
 cor.test(pearson_rho[,3],pve)$estimate,
 cor.test(pearson_rho[,4],pve)$estimate,
 cor.test(pearson_rho[,5],pve)$estimate)
 mean(cor_rho)
 sd(cor_rho)
 
 > cor_rho
      cor       cor       cor       cor       cor 
0.3551534 0.3593960 0.3386795 0.3992226 0.3501630 
>  mean(cor_rho)
[1] 0.3605229
>  sd(cor_rho)
[1] 0.02297952

 cor.test(pearson_rho[,1],pve)$p.value
 cor.test(pearson_rho[,2],pve)$p.value
 cor.test(pearson_rho[,3],pve)$p.value
 cor.test(pearson_rho[,4],pve)$p.value
 cor.test(pearson_rho[,5],pve)$p.value
 
 
 
 ###################################
 
 
 
 
```







