
# seperate by chromosome
```
for TRAIT in BD CAD CD HT RA T1D T2D; do
echo ${TRAIT}
for ((i=1; i<=22; i++)); do
echo ${i}
awk '{print $1,$2,$3,$4,$5,$6}'  ${TRAIT}.fam  > ${TRAIT}.fam1
awk '{print $1,$2,$3,$4,$5,$6}'  ${TRAIT}.fam1  > ${TRAIT}.fam
plink --noweb --bfile ${TRAIT} --chr ${i} --make-bed --out ${TRAIT}_chr${i}
done
done



```

# write genotype into tables
```R
library(snpStats)
traits = c("T1D","T2D","RA","HT","CD","CAD","BD")
pathwtccc="/net/mulan/home/shanglu/GENOA/data/WTCCC"

for(traitnum in 6:7){
print(traits[traitnum])
for(i in 1:22){
print(i)
fam <- paste0(pathwtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".fam")
bim <- paste0(pathwtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".bim")
bed <- paste0(pathwtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".bed")
m = read.plink(bed, bim, fam)
write.SnpMatrix(m$genotypes,paste0(pathwtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr_",i,"_SnpMatrix.txt"),quote=F,col.names = F,row.names = F)
genomat = read.table(paste0(pathwtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr_",i,"_SnpMatrix.txt"),header=F)
save(genomat, file = paste0(pathwtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr_",i,"_SnpMatrix.RData"))
system(paste0("rm ",pathwtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr_",i,"_SnpMatrix.txt"))
}
}


```

##### match gene-SNP pairs of AA/EA and T1D
```R
###############################

args <- as.numeric(commandArgs(TRUE))
traitnum=args[1]
i=args[2]
print(traitnum)


path_bslmm = "/net/mulan/home/shanglu/GENOA/analysis/prediction"
path_wtccc="/net/mulan/home/shanglu/GENOA/data/WTCCC"
load(paste0(path_bslmm,"/AA/bslmm_all_result_AA.RData"))
traits = c("T1D","T2D","RA","HT","CD","CAD","BD")




	print(traits[traitnum])
	print(i)
	AA_chr = bslmm_all_result[which(bslmm_all_result$chr==i),]
	genes_aa = unique(as.character(AA_chr$GENE))
	bim = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".bim")
	Trait_snp = read.table(bim)
	common_snp = intersect(as.character(Trait_snp$V2),as.character(AA_chr$SNP))
	ind_trait = which(as.character(Trait_snp$V2) %in% common_snp)
	ind_aa = which(as.character(AA_chr$SNP) %in% common_snp)
	AA_to_use = AA_chr[ind_aa,]
	Trait_to_use = Trait_snp[ind_trait,]
	index = match( as.character(AA_to_use$SNP),as.character(Trait_snp$V2))
	AA_to_use$index_in_trait = index
	AA_to_use$minor_geu = as.character(Trait_snp$V5)[index]
	AA_to_use$major_geu = as.character(Trait_snp$V6)[index]
	save(AA_to_use, file = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_AA_to_use_chr_",i,".RData"))
	print(dim(AA_to_use))

###############################

#!/bin/bash
#SBATCH --job-name=AAwtccc
#SBATCH --array=1-200
#SBATCH --cpus-per-task=1
#SBATCH --output=/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping/out/AAwtccc%a.out
#SBATCH --error=/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping/err/AAwtccc%a.err
#SBATCH --workdir=/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping/code
#SBATCH --partition=mulan

bash

let k=0

for ((traitnum=1;traitnum<=7;traitnum++)); do
for ((chr=1;chr<=22;chr++)); do
  let k=${k}+1
  if [ ${k} -eq ${SLURM_ARRAY_TASK_ID} ]; then
  Rscript --verbose ./AA_bslmm_WTCCC.r ${traitnum}  ${chr}
  fi
done
done

###############################
args <- as.numeric(commandArgs(TRUE))
traitnum=args[1]
i=args[2]
print(traitnum)


path_bslmm = "/net/mulan/home/shanglu/GENOA/analysis/prediction"
path_wtccc="/net/mulan/home/shanglu/GENOA/data/WTCCC"
load(paste0(path_bslmm,"/EA/bslmm_all_result_EA.RData"))
traits = c("T1D","T2D","RA","HT","CD","CAD","BD")




	print(traits[traitnum])
	print(i)
	EA_chr = bslmm_all_result[which(bslmm_all_result$chr==i),]
	genes_EA = unique(as.character(EA_chr$GENE))
	bim = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".bim")
	Trait_snp = read.table(bim)
	common_snp = intersect(as.character(Trait_snp$V2),as.character(EA_chr$SNP))
	ind_trait = which(as.character(Trait_snp$V2) %in% common_snp)
	ind_EA = which(as.character(EA_chr$SNP) %in% common_snp)
	EA_to_use = EA_chr[ind_EA,]
	Trait_to_use = Trait_snp[ind_trait,]
	index = match( as.character(EA_to_use$SNP),as.character(Trait_snp$V2))
	EA_to_use$index_in_trait = index
	EA_to_use$minor_geu = as.character(Trait_snp$V5)[index]
	EA_to_use$major_geu = as.character(Trait_snp$V6)[index]
	save(EA_to_use, file = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_EA_to_use_chr_",i,".RData"))
	print(dim(EA_to_use))

###############################
#!/bin/bash
#SBATCH --job-name=EAwtccc
#SBATCH --array=1-200
#SBATCH --cpus-per-task=1
#SBATCH --output=/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping/out/EAwtccc%a.out
#SBATCH --error=/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping/err/EAwtccc%a.err
#SBATCH --workdir=/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping/code
#SBATCH --partition=mulan

bash

let k=0

for ((traitnum=1;traitnum<=7;traitnum++)); do
for ((chr=1;chr<=22;chr++)); do
  let k=${k}+1
  if [ ${k} -eq ${SLURM_ARRAY_TASK_ID} ]; then
  Rscript --verbose ./EA_bslmm_WTCCC.r ${traitnum}  ${chr}
  fi
done
done




```


# prediction in WTCCC
```
# AA

args <- as.numeric(commandArgs(TRUE))
traitnum=args[1]
i=args[2]
print(traitnum)


path_bslmm = "/net/mulan/home/shanglu/GENOA/analysis/prediction"
path_wtccc="/net/mulan/home/shanglu/GENOA/data/WTCCC"
traits = c("T1D","T2D","RA","HT","CD","CAD","BD")

library(tidyr)

popugenoa=1
print(i)
load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_AA_to_use_chr_",i,".RData"))
load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr_",i,"_SnpMatrix.RData"))
fam = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".fam")
Geu_fam = read.table(fam)
	
print("Data loaded")
dat_to_use = AA_to_use
gene_name = unique(as.character(dat_to_use$GENE))			#!!!!
expr_mat1 = matrix(0,length(gene_name), dim(Geu_fam)[1])
rownames(expr_mat1) = gene_name
colnames(expr_mat1) = as.character(Geu_fam$V1)
expr_mat2 = expr_mat1

print("expr_mat created")

for(j in 1:length(gene_name)){
	print(j)
	subset = dat_to_use[which(as.character(dat_to_use$GENE) %in% gene_name[j]),]		 #!!!!
	genotype = genomat[,subset$index_in_trait]
	if(dim(subset)[1]==1){
	genotypeimpute = genotype
	genotypeimpute[is.na(genotypeimpute)] <- mean(genotypeimpute, na.rm = TRUE)
	}else{
	genotypeimpute =  replace_na(genotype,as.list(colMeans(genotype,na.rm=T)))
	}
	# direction of beta
	inconsistent = which(as.character(subset$allele1)!=as.character(subset$minor_geu))	 #!!!!
	subset[inconsistent,]$weight = -subset[inconsistent,]$weight 				 #!!!!

	genotypeimpute = scale(genotypeimpute)
	expr_pred2 = subset$weight %*% t(2-genotypeimpute)
	expr_mat2[j,]=expr_pred2
}

print("Start saving file")
save(expr_mat2, file = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_bslmm_AA_v2.RData")) 
}


#######################
#!/bin/bash
#SBATCH --job-name=AAwtcccpred
#SBATCH --array=1-200
#SBATCH --cpus-per-task=1
#SBATCH --output=/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping/out/AAwtcccpred%a.out
#SBATCH --error=/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping/err/AAwtcccpred%a.err
#SBATCH --workdir=/net/mulan/home/shanglu/GENOA/analysis/AA/eqtlmapping/code
#SBATCH --partition=mulan

bash

let k=0

for ((traitnum=1;traitnum<=7;traitnum++)); do
for ((chr=1;chr<=22;chr++)); do
  let k=${k}+1
  if [ ${k} -eq ${SLURM_ARRAY_TASK_ID} ]; then
  Rscript --verbose ./AA_bslmm_WTCCC_pred.r ${traitnum}  ${chr}
  fi
done
done


#######################
args <- as.numeric(commandArgs(TRUE))
traitnum=args[1]
i=args[2]
print(traitnum)


path_bslmm = "/net/mulan/home/shanglu/GENOA/analysis/prediction"
path_wtccc="/net/mulan/home/shanglu/GENOA/data/WTCCC"
traits = c("T1D","T2D","RA","HT","CD","CAD","BD")

library(tidyr)

popugenoa=1

print(traits[traitnum])

print(i)
load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_EA_to_use_chr_",i,".RData"))
load(paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr_",i,"_SnpMatrix.RData"))
fam = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,".fam")
Geu_fam = read.table(fam)
	
print("Data loaded")
dat_to_use = EA_to_use
gene_name = unique(as.character(dat_to_use$GENE))			#!!!!
expr_mat1 = matrix(0,length(gene_name), dim(Geu_fam)[1])
rownames(expr_mat1) = gene_name
colnames(expr_mat1) = as.character(Geu_fam$V1)
expr_mat2 = expr_mat1

print("expr_mat created")

for(j in 1:length(gene_name)){
	print(j)
	subset = dat_to_use[which(as.character(dat_to_use$GENE) %in% gene_name[j]),]		 #!!!!
	genotype = genomat[,subset$index_in_trait]
	if(dim(subset)[1]==1){
	genotypeimpute = genotype
	genotypeimpute[is.na(genotypeimpute)] <- mean(genotypeimpute, na.rm = TRUE)
	}else{
	genotypeimpute =  replace_na(genotype,as.list(colMeans(genotype,na.rm=T)))
	}
	# direction of beta
	inconsistent = which(as.character(subset$allele1)!=as.character(subset$minor_geu))	 #!!!!
	subset[inconsistent,]$weight = -subset[inconsistent,]$weight 				 #!!!!

	genotypeimpute = scale(genotypeimpute)
	expr_pred2 = subset$weight %*% t(2-genotypeimpute)
	expr_mat2[j,]=expr_pred2
}

print("Start saving file")
save(expr_mat2, file = paste0(path_wtccc,"/",traits[traitnum],"/",traits[traitnum],"_chr",i,"_bslmm_EA_v2.RData")) 

#######################
#!/bin/bash
#SBATCH --job-name=EAwtcccpred
#SBATCH --array=1-200
#SBATCH --cpus-per-task=1
#SBATCH --output=/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping/out/EAwtcccpred%a.out
#SBATCH --error=/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping/err/EAwtcccpred%a.err
#SBATCH --workdir=/net/mulan/home/shanglu/GENOA/analysis/EA/eqtlmapping/code
#SBATCH --partition=mulan

bash

let k=0

for ((traitnum=1;traitnum<=7;traitnum++)); do
for ((chr=1;chr<=22;chr++)); do
  let k=${k}+1
  if [ ${k} -eq ${SLURM_ARRAY_TASK_ID} ]; then
  Rscript --verbose ./EA_bslmm_WTCCC_pred.r ${traitnum}  ${chr}
  fi
done
done




```


# Collect results

```R

path_wtccc="/net/mulan/home/shanglu/GENOA/data/WTCCC"


library(tidyr)

traits = c("T1D","T2D","RA","HT","CD","CAD","BD")
popugenoanames = c("AA" , "EA")
PCs = c(2,5,10)
popnums = c(1,2)
traitnums = c(1:7)
chr = c(1:22)
# repeat:
traitnum = traitnum + 1


#######################
# first consider AA
popnum = 1
load("/net/mulan/home/shanglu/GENOA/data/AA/gene_AA_anno_order_correct.RData")
#######################


#######################
# next consider EA
popnum = 2
load("/net/mulan/home/shanglu/GENOA/data/EA/gene_EA_anno_order_correct.RData")
#######################

for(PCnum in c(2,5,10)){
print("PCnum")
print(PCnum)
for(traitnum in 1:7){
print(traitnum)
i=1
load(paste0(path_wtccc,"/result/result_",popugenoanames[popnum],"_bslmm_trait_",traits[traitnum],"_PC_",PCnum,"_chr_",i,".RData"))
Gglm_PCs_p = result$Gglm_PCs_p
#Glm_PCs_p = result$Glm_PCs_p
Gglm_noPCs_p = result$Gglm_noPCs_p
#Glm_noPCs_p = result$Glm_noPCs_p

for(i in 2:22){
load(paste0(path_wtccc,"/result/result_",popugenoanames[popnum],"_bslmm_trait_",traits[traitnum],"_PC_",PCnum,"_chr_",i,".RData"))
Gglm_PCs_p = rbind(Gglm_PCs_p,result$Gglm_PCs_p)
#Glm_PCs_p = rbind(Glm_PCs_p, result$Glm_PCs_p)
Gglm_noPCs_p = rbind(Gglm_noPCs_p, result$Gglm_noPCs_p)
#Glm_noPCs_p = rbind(Glm_noPCs_p, result$Glm_noPCs_p)
}

ind1 =  which(Gglm_PCs_p[,traitnum] < 0.05/dim(Gglm_PCs_p)[1])
gene_signif_Gglm_PCs_p = rownames(Gglm_PCs_p)[ind1]
ind = match(gene_signif_Gglm_PCs_p, gene_EA_anno_order$GENE)
gene_symbol1 = as.character(gene_EA_anno_order$gene)[ind]
pvalue_Gglm_PCs_p = Gglm_PCs_p[ind1, traitnum]

#ind2 =  which(Glm_PCs_p[,traitnum] < 0.05/dim(Glm_PCs_p)[1])
#gene_signif_Glm_PCs_p = rownames(Glm_PCs_p)[ind2]
#pvalue_Glm_PCs_p = Glm_PCs_p[ind2, traitnum]
#ind = match(gene_signif_Glm_PCs_p, gene_AA_anno_order$GENE)
#gene_symbol2 = as.character(gene_AA_anno_order$gene)[ind]

ind3 =  which(Gglm_noPCs_p[,traitnum] < 0.05/dim(Gglm_noPCs_p)[1])
gene_signif_Gglm_noPCs_p = rownames(Gglm_noPCs_p)[ind3]
pvalue_Gglm_noPCs_p = Gglm_noPCs_p[ind3, traitnum]
ind = match(gene_signif_Gglm_noPCs_p, gene_EA_anno_order$GENE)
gene_symbol3 = as.character(gene_EA_anno_order$gene)[ind]


#ind4 =  which(Glm_noPCs_p[,traitnum] < 0.05/dim(Glm_noPCs_p)[1])
#gene_signif_Glm_noPCs_p = rownames(Glm_noPCs_p)[ind4]
#pvalue_Glm_noPCs_p = Glm_noPCs_p[ind4, traitnum]
#ind = match(gene_signif_Glm_noPCs_p, gene_AA_anno_order$GENE)
#gene_symbol4 = as.character(gene_AA_anno_order$gene)[ind]



dat = data.frame(gene_signif_Gglm_PCs_p, gene_symbol1,pvalue_Gglm_PCs_p)
dat = dat[order( dat[,3] ),]
write.csv(dat, paste0(path_wtccc,"/result/Dec24_result_",popugenoanames[popnum],"_bslmm_trait_",traits[traitnum],"_PC_",PCnum,".csv"))

dat = data.frame(gene_signif_Gglm_noPCs_p, gene_symbol3,pvalue_Gglm_noPCs_p)
dat = dat[order( dat[,3] ),]

write.csv(dat, paste0(path_wtccc,"/result/Dec24_result_",popugenoanames[popnum],"_bslmm_trait_",traits[traitnum],"_PC_",0,".csv"))

}

}

```


```R

#Make the plot
library(VennDiagram)
venn.diagram(
  x = sets,
  category.names = c("AA" , "EA" , "AFA","CAU","HIS"),
  filename = paste0(path_wtccc,"/result/vennTrait",traitnum,".png"),
  output=TRUE)


mat = matrix(0,7,5)
for(traitnum in 1:7){
for(popugenoa in 1:5){
	load(paste0(path_wtccc,"/result/result_",popugenoanames[popugenoa],"_commonpairs.RData"))
	sets[[popugenoa]] = rownames(result$Gglm_PCs_p)[which(result$Gglm_PCs_p[,traitnum] < 0.05/9029)]
	mat[traitnum, popugenoa] = length(sets[[popugenoa]])
}
}
colnames(mat) =  c("AA" , "EA" , "AFA","CAU","HIS")
rownames(mat) = traits
write.csv(mat, paste0(path_wtccc,"/result/Genenum_count_trait.csv"))




```

# Get genes, glm with PC version
```R
library(tidyr)
traits = c("T1D","T2D","RA","HT","CD","CAD","BD")
popugenoanames = c("aa" , "ea", "afa", "cau", "his")
popugenoanamesCapital = c("AA" , "EA", "AFA", "CAU", "HIS")

traitnum = 0

# repeat:
traitnum = traitnum + 1
sets = list()
popp = c()
genepopp = c()
pvall = c()
for(popugenoa in 1:5){
load(paste0(path_wtccc,"/result/result_",popugenoanames[popugenoa],"_commonpairs.RData"))
sets[[popugenoa]] = rownames(result$Gglm_PCs_p)[which(result$Gglm_PCs_p[,traitnum] < 0.05/9029)]
popp = c(popp, rep(popugenoanames[popugenoa],length(sets[[popugenoa]])))
genepopp = c(genepopp, sets[[popugenoa]])
pvall = c(pvall, result$Gglm_PCs_p[which(result$Gglm_PCs_p[,traitnum] < 0.05/9029),traitnum])
}
dat = data.frame(popp, genepopp, pvall)
dat = dat[order( dat[,1], dat[,3] ),]
write.csv(dat, paste0(path_wtccc,"/result/Genelist_Trait",traitnum,".csv"))


#Make the plot
library(VennDiagram)
venn.diagram(
  x = sets,
  category.names = c("AA" , "EA" , "AFA","CAU","HIS"),
  filename = paste0(path_wtccc,"/result/vennTrait",traitnum,".png"),
  output=TRUE)


mat = matrix(0,7,5)
for(traitnum in 1:7){
for(popugenoa in 1:5){
	load(paste0(path_wtccc,"/result/result_",popugenoanames[popugenoa],"_commonpairs.RData"))
	sets[[popugenoa]] = rownames(result$Gglm_PCs_p)[which(result$Gglm_PCs_p[,traitnum] < 0.05/9029)]
	mat[traitnum, popugenoa] = length(sets[[popugenoa]])
}
}
colnames(mat) =  c("AA" , "EA" , "AFA","CAU","HIS")
rownames(mat) = traits
write.csv(mat, paste0(path_wtccc,"/result/Genenum_count_trait.csv"))

```

# Get genes, glm no PC version
```R
library(tidyr)
traits = c("T1D","T2D","RA","HT","CD","CAD","BD")
popugenoanames = c("aa" , "ea", "afa", "cau", "his")
popugenoanamesCapital = c("AA" , "EA", "AFA", "CAU", "HIS")

traitnum = 0

# repeat:
traitnum = traitnum + 1
sets = list()
popp = c()
genepopp = c()
pvall = c()
for(popugenoa in 1:5){
load(paste0(path_wtccc,"/result/result_",popugenoanames[popugenoa],"_commonpairs.RData"))
sets[[popugenoa]] = rownames(result$Gglm_noPCs_p)[which(result$Gglm_noPCs_p[,traitnum] < 0.05/9029)]
popp = c(popp, rep(popugenoanames[popugenoa],length(sets[[popugenoa]])))
genepopp = c(genepopp, sets[[popugenoa]])
pvall = c(pvall, result$Gglm_noPCs_p[which(result$Gglm_noPCs_p[,traitnum] < 0.05/9029),traitnum])
}
dat = data.frame(popp, genepopp, pvall)
dat = dat[order( dat[,1], dat[,3] ),]
write.csv(dat, paste0(path_wtccc,"/result/noPC_Genelist_Trait",traitnum,".csv"))


#Make the plot
library(VennDiagram)
venn.diagram(
  x = sets,
  category.names = c("AA" , "EA" , "AFA","CAU","HIS"),
  filename = paste0(path_wtccc,"/result/noPC_vennTrait",traitnum,".png"),
  output=TRUE)


mat = matrix(0,7,5)
for(traitnum in 1:7){
for(popugenoa in 1:5){
	load(paste0(path_wtccc,"/result/result_",popugenoanames[popugenoa],"_commonpairs.RData"))
	sets[[popugenoa]] = rownames(result$Gglm_noPCs_p)[which(result$Gglm_noPCs_p[,traitnum] < 0.05/9029)]
	mat[traitnum, popugenoa] = length(sets[[popugenoa]])
}
}
colnames(mat) =  c("AA" , "EA" , "AFA","CAU","HIS")
rownames(mat) = traits
write.csv(mat, paste0(path_wtccc,"/result/noPC_Genenum_count_trait.csv"))

```


QQ plot
```
library(lattice)
qqunif.plot<-function(pvalues, 
	should.thin=T, thin.obs.places=2, thin.exp.places=2, 
	xlab=expression(paste("Expected (",-log[10], " p-value)")),
	ylab=expression(paste("Observed (",-log[10], " p-value)")), 
	draw.conf=TRUE, conf.points=1000, conf.col="lightgray", conf.alpha=.05,
	already.transformed=FALSE, pch=20, 
	aspect="iso", 
	prepanel=prepanel.qqunif,
	par.settings=list(superpose.symbol=list(pch=pch)), ...) {
	
	
	#error checking
	if (length(pvalues)==0) stop("pvalue vector is empty, can't draw plot")
	if(!(class(pvalues)=="numeric" || 
		(class(pvalues)=="list" && all(sapply(pvalues, class)=="numeric"))))
		stop("pvalue vector is not numeric, can't draw plot")
	if (any(is.na(unlist(pvalues)))) stop("pvalue vector contains NA values, can't draw plot")
	if (already.transformed==FALSE) {
		if (any(unlist(pvalues)==0)) stop("pvalue vector contains zeros, can't draw plot")
	} else {
		if (any(unlist(pvalues)<0)) stop("-log10 pvalue vector contains negative values, can't draw plot")
	}
	
	
	grp<-NULL
	n<-1
	exp.x<-c()
	if(is.list(pvalues)) {
		nn<-sapply(pvalues, length)
		rs<-cumsum(nn)
		re<-rs-nn+1
		n<-min(nn)
		if (!is.null(names(pvalues))) {
			grp=factor(rep(names(pvalues), nn), levels=names(pvalues))
			names(pvalues)<-NULL
		} else {
			grp=factor(rep(1:length(pvalues), nn))
		}
		pvo<-pvalues
		pvalues<-numeric(sum(nn))
		exp.x<-numeric(sum(nn))
		for(i in 1:length(pvo)) {
			if (!already.transformed) {
				pvalues[rs[i]:re[i]] <- -log10(pvo[[i]])
				exp.x[rs[i]:re[i]] <- -log10((rank(pvo[[i]], ties.method="first")-.5)/nn[i])
			} else {
				pvalues[rs[i]:re[i]] <- pvo[[i]]
				exp.x[rs[i]:re[i]] <- -log10((nn[i]+1-rank(pvo[[i]], ties.method="first")-.5)/(nn[i]+1))
			}
		}
	} else {
		n <- length(pvalues)+1
		if (!already.transformed) {
			exp.x <- -log10((rank(pvalues, ties.method="first")-.5)/n)
			pvalues <- -log10(pvalues)
		} else {
			exp.x <- -log10((n-rank(pvalues, ties.method="first")-.5)/n)
		}
	}


	#this is a helper function to draw the confidence interval
	panel.qqconf<-function(n, conf.points=1000, conf.col="gray", conf.alpha=.05, ...) {
		require(grid)
		conf.points = min(conf.points, n-1);
		mpts<-matrix(nrow=conf.points*2, ncol=2)
        	for(i in seq(from=1, to=conf.points)) {
            		mpts[i,1]<- -log10((i-.5)/n)
            		mpts[i,2]<- -log10(qbeta(1-conf.alpha/2, i, n-i))
            		mpts[conf.points*2+1-i,1]<- -log10((i-.5)/n)
            		mpts[conf.points*2+1-i,2]<- -log10(qbeta(conf.alpha/2, i, n-i))
        	}
        	grid.polygon(x=mpts[,1],y=mpts[,2], gp=gpar(fill=conf.col, lty=0), default.units="native")
    	}

	#reduce number of points to plot
	if (should.thin==T) {
		if (!is.null(grp)) {
			thin <- unique(data.frame(pvalues = round(pvalues, thin.obs.places),
				exp.x = round(exp.x, thin.exp.places),
				grp=grp))
			grp = thin$grp
		} else {
			thin <- unique(data.frame(pvalues = round(pvalues, thin.obs.places),
				exp.x = round(exp.x, thin.exp.places)))
		}
		pvalues <- thin$pvalues
		exp.x <- thin$exp.x
	}
	gc()
	
	prepanel.qqunif= function(x,y,...) {
		A = list()
		A$xlim = range(x)*1.02
		A$xlim[1]=0
		A$ylim = range(y)*1.02
		A$ylim[1]=0
		return(A)
	}

	#draw the plot
	xyplot(pvalues~exp.x, groups=grp, xlab=xlab, ylab=ylab, 
		#aspect=aspect,
		prepanel=prepanel, 
		scales=list(axs="i"), 
		pch=pch,
		panel = function(x, y, ...) {
			if (draw.conf) {
				panel.qqconf(n, conf.points=conf.points, 
					conf.col=conf.col, conf.alpha=conf.alpha)
			};
			panel.xyplot(x,y, ...);
			panel.abline(0,1);
		}, par.settings=par.settings, ...
	)
}


```
	
```R

traitnum = 0

traitnum = traitnum + 1
ph = c()
library(dplyr) 

for(popugenoa in 1:5){
	load(paste0(path_wtccc,"/result/result_",popugenoanames[popugenoa],"_commonpairs.RData"))
	ph_tmp = (c(result$Gglm_PCs_p[,traitnum]))
	ph = c(ph, ph_tmp)
}

Population = c(rep("AA",dim(result$Gglm_PCs_p)[1]),
rep("EA",dim(result$Gglm_PCs_p)[1]),
rep("AFA",dim(result$Gglm_PCs_p)[1]),
rep("CAU",dim(result$Gglm_PCs_p)[1]),
rep("HIS",dim(result$Gglm_PCs_p)[1]))
dat = data.frame(ph,Population )
dat$Population = as.character(Population)
mydat = dat

my.pvalue.list<-list("AA"=mydat[mydat$Population=="AA",1], "EA"=mydat[mydat$Population=="EA",1],
"AFA"=mydat[mydat$Population=="AFA",1],"CAU"=mydat[mydat$Population=="CAU",1],
"HIS"=mydat[mydat$Population=="HIS",1])

pdf(paste0("QQ_trait",traits[traitnum],".pdf") ,width = 6, height = 6)
qqunif.plot(my.pvalue.list, auto.key=list(corner=c(.95,.05)))
dev.off()







```
Ping Zeng's Nature Communication:
The table also lists the disease name, gene id, chromosome number, transcription start
site (TSS), association strength (z score, p value), the number of SNPs in each gene set
test, estimated SNP heritability (h2 , from GEMMA), and references that support the
identified association. T1D: type 1 diabetes, CD: Crohn's disease, RA: rheumatoid
arthritis. H indicates Human leukocyte antigen (HLA) region genes on chromosome 6, M
indicates major histocompatibility complex (MHC) region, G indicates genes previously
identified to be associated with diseases in the NHGRI GWAS catalog, V indicates the
vicinity of a reported gene. h2 is the estimator of heritability using linear mixed models in
GEMMA

table:
Disease Gene Chr TSS z score p value #SNPs h2 Reference

