
#### 1.AA_make_plink_file.md
#### 1.EA_make_plink_file.md
for each gene, we map SNPs within +/-100kb of TSS/TES and extract the plink file for genotype
#### 2.AA_make_relatedness.md
#### 2.EA_make_relatedness.md
estimate relatedness matrix for AA and EA
#### 3.Extract_genetic_PCs.md
calculate genetic PCs for AA and EA using PCair
#### 4.AA_prepare_expression.md
#### 4.EA_prepare_expression.md
prepare gene expression files
#### 4.5.AA_PCs.md
#### 4.5.EA_PCs.md
adjust gene expression
#### 5.AAeqtl.md
#### 5.EAeqtl.md
eQTL mapping
#### 6.AA_collect_result.md
#### 6.EA_collect_result.md
collect results
#### 7.AA_bslmm.md
#### 7.EA_bslmm.md
BSLMM calculates PVE
#### 8.Predict_expression_bslmm.md
predict GEUVADIS using results from BSLMM 
#### 9.TWAS_WTCCC.md
TWAS in WTCCC using results from BSLMM 
#### 10.conditional_analysis.md
conditional analysis, efficient codes and scripts
#### 10.5.conditional_pve.md
conditional analysis, PVE fo primary and independent eQTLs
#### 11.AA_elastic_net.md
#### 11.EA_elastic_net.md
#### 11.Prediction_model_compare_elastic_net.md
predict GEUVADIS using results from elnet 
#### 12.WTCCC_elnet.md
TWAS in WTCCC using results from elnet 
#### 13.pi_statistics.md
pi1 statistics
#### 14.LocalAncestry.md
compare results when adjusting for local ancestry instead of global ancestry
#### Paper_figure.md
codes for figures in paper

